diff -rup SDL_gfx-2.0.26.orig/SDL_framerate.h SDL_gfx-2.0.26/SDL_framerate.h
--- SDL_gfx-2.0.26.orig/SDL_framerate.h	2017-02-21 05:37:29.000000000 +0100
+++ SDL_gfx-2.0.26/SDL_framerate.h	2023-11-15 13:33:27.133442411 +0100
@@ -37,9 +37,9 @@ extern "C" {
 
 	/* --- */
 
-#include <SDL.h>
+#include "SDL.h"
 
-	/* --------- Definitions */
+	/* --------- definitions */
 
 	/*!
 	\brief Highest possible rate supported by framerate controller in Hz (1/s).
diff -rup SDL_gfx-2.0.26.orig/SDL_gfxBlitFunc.h SDL_gfx-2.0.26/SDL_gfxBlitFunc.h
--- SDL_gfx-2.0.26.orig/SDL_gfxBlitFunc.h	2017-02-21 05:37:29.000000000 +0100
+++ SDL_gfx-2.0.26/SDL_gfxBlitFunc.h	2023-11-15 13:33:09.413466464 +0100
@@ -38,8 +38,8 @@ extern    "C" {
 #include <stdio.h>
 #include <stdlib.h>
 
-#include <SDL.h>
-#include <SDL_video.h>
+#include "SDL.h"
+#include "SDL_video.h"
 
 
 	extern const unsigned int GFX_ALPHA_ADJUST_ARRAY[256];
diff -rup SDL_gfx-2.0.26.orig/SDL_gfxPrimitives.h SDL_gfx-2.0.26/SDL_gfxPrimitives.h
--- SDL_gfx-2.0.26.orig/SDL_gfxPrimitives.h	2017-02-21 05:37:29.000000000 +0100
+++ SDL_gfx-2.0.26/SDL_gfxPrimitives.h	2023-11-15 13:33:06.426803851 +0100
@@ -35,7 +35,7 @@ Andreas Schiffler -- aschiffler at ferzk
 #define M_PI	3.1415926535897932384626433832795
 #endif
 
-#include <SDL.h>
+#include "SDL.h"
 
 /* Set up for C function definitions, even when using C++ */
 #ifdef __cplusplus
diff -rup SDL_gfx-2.0.26.orig/SDL_rotozoom.h SDL_gfx-2.0.26/SDL_rotozoom.h
--- SDL_gfx-2.0.26.orig/SDL_rotozoom.h	2017-02-21 05:37:29.000000000 +0100
+++ SDL_gfx-2.0.26/SDL_rotozoom.h	2023-11-15 13:33:17.436788907 +0100
@@ -41,7 +41,7 @@ extern "C" {
 #define M_PI	3.141592654
 #endif
 
-#include <SDL.h>
+#include "SDL.h"
 
 	/* ---- Defines */
 
