diff -rupN blobwars-1.14/src/pak.cpp blobwars-1.14.new/src/pak.cpp
--- blobwars-1.14/src/pak.cpp	2009-02-01 12:34:44.000000000 +0100
+++ blobwars-1.14.new/src/pak.cpp	2023-12-03 14:27:01.892761403 +0100
@@ -20,7 +20,7 @@ Foundation, Inc., 59 Temple Place - Suit
 
 #include "pak.h"
 
-FILE *pak;
+static FILE *pak;
 int dirs = 0, files = 0;
 int totalFiles = 0;
 Bytef *buffer;
@@ -124,7 +124,7 @@ void recurseDirectory(const char *dirNam
 			{
 				printf("Couldn't open %s for reading!\n", filename);
 				closedir(dirp);
-				gzclose(pak);
+				fclose(pak);
 				exit(1);
 			}
 			
@@ -156,7 +156,7 @@ void recurseDirectory(const char *dirNam
 			{
 				printf("Couldn't open %s for reading!\n", filename);
 				closedir(dirp);
-				gzclose(pak);
+				fclose(pak);
 				exit(1);
 			}
 			else
