--- a/config.sub.orig	2019-10-27 18:42:59.528348412 +0100
+++ b/config.sub	2019-10-27 18:46:27.316343891 +0100
@@ -823,6 +823,11 @@
 		cpu=m68000
 		vendor=convergent
 		;;
+	mintelf | mintelf[0-9]*)
+		cpu=m68k
+		vendor=atari
+		os=mintelf
+		;;
 	*mint | mint[0-9]* | *MiNT | *MiNT[0-9]*)
 		cpu=m68k
 		vendor=atari
@@ -1491,6 +1496,9 @@
 	ose*)
 		os=ose
 		;;
+	*mintelf | mintelf[0-9]*)
+		os=mintelf
+		;;
 	*mint | mint[0-9]* | *MiNT | MiNT[0-9]*)
 		os=mint
 		;;
@@ -1869,6 +1877,9 @@
 			*-mpw* | *-macos*)
 				vendor=apple
 				;;
+			*-mintelf | *-mintelf[0-9]*)
+				vendor=atari
+				;;
 			*-*mint | *-mint[0-9]* | *-*MiNT | *-MiNT[0-9]*)
 				vendor=atari
 				;;
