--- libexif-0.6.21.orig/doc/Doxyfile-internals.in
+++ libexif-0.6.21/doc/Doxyfile-internals.in
@@ -1542,6 +1542,7 @@ DOT_CLEANUP            = YES
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
 SEARCHENGINE           = NO
+HTML_TIMESTAMP         = NO
 
 # When the SERVER_BASED_SEARCH tag is enabled the search engine will be
 # implemented using a web server instead of a web client using Javascript. There
--- libexif-0.6.21.orig/doc/Doxyfile.in
+++ libexif-0.6.21/doc/Doxyfile.in
@@ -1541,6 +1541,7 @@ DOT_CLEANUP            = YES
 # This tag requires that the tag GENERATE_HTML is set to YES.
 
 SEARCHENGINE           = NO
+HTML_TIMESTAMP         = NO
 
 # When the SERVER_BASED_SEARCH tag is enabled the search engine will be
 # implemented using a web server instead of a web client using Javascript. There
