--- findutils-4.7.0.orig/build-aux/config.sub	2019-06-23 11:00:48.000000000 +0200
+++ findutils-4.7.0/build-aux/config.sub	2020-06-27 12:35:26.897436360 +0200
@@ -1337,6 +1337,14 @@
 	psos*)
 		os=psos
 		;;
+	mintelf | mintelf[0-9]*)
+		basic_machine=m68k-atari
+		os=mintelf
+		;;
+	mint | mint[0-9]*)
+		basic_machine=m68k-atari
+		os=-mint
+		;;
 	# Now accept the basic system types.
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
@@ -1486,7 +1494,12 @@
 	ose*)
 		os=ose
 		;;
+	*mintelf | -mintelf[0-9]*)
+		basic_machine=m68k-atari
+		os=mintelf
+		;;
 	*mint | mint[0-9]* | *MiNT | MiNT[0-9]*)
+		basic_machine=m68k-atari
 		os=mint
 		;;
 	zvmoe)
@@ -1777,6 +1790,9 @@
 			mpw* | macos*)
 				vendor=apple
 				;;
+			*mintelf | mintelf[0-9]*)
+				vendor=atari
+				;;
 			*mint | mint[0-9]* | *MiNT | MiNT[0-9]*)
 				vendor=atari
 				;;
