--- bsd-finger-0.17/finger/Makefile.orig	Fri Sep 28 16:42:38 2001
+++ bsd-finger-0.17/finger/Makefile	Fri Sep 28 16:46:54 2001
@@ -3,7 +3,7 @@
 include ../MCONFIG
 include ../MRULES
 
-finger: finger.o lprint.o net.o sprint.o util.o display.o
+finger: finger.o lprint.o net.o sprint.o util.o display.o finger-strftime.o
 	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
 
 finger.o lprint.o net.o sprint.o util.o display.o: finger.h
