--- bsd-finger-0.17/finger/Makefile.orig	Sun Dec 12 19:04:52 1999
+++ bsd-finger-0.17/finger/Makefile	Wed Sep 26 17:23:38 2001
@@ -10,7 +10,7 @@
 finger.o: ../version.h
 
 install: finger
-	install -s -m$(BINMODE) finger $(INSTALLROOT)$(BINDIR)
+	install -m$(BINMODE) finger $(INSTALLROOT)$(BINDIR)
 	install -m$(MANMODE) finger.1 $(INSTALLROOT)$(MANDIR)/man1
 
 clean:
--- bsd-finger-0.17/fingerd/Makefile.orig	Sun Dec 12 19:04:56 1999
+++ bsd-finger-0.17/fingerd/Makefile	Wed Sep 26 17:23:46 2001
@@ -9,7 +9,7 @@
 fingerd.o: pathnames.h ../version.h
 
 install: fingerd
-	install -s -m$(DAEMONMODE) fingerd $(INSTALLROOT)$(SBINDIR)/in.fingerd
+	install -m$(DAEMONMODE) fingerd $(INSTALLROOT)$(SBINDIR)/in.fingerd
 	install -m$(MANMODE) fingerd.8 $(INSTALLROOT)$(MANDIR)/man8/in.fingerd.8
 	ln -sf in.fingerd.8 $(INSTALLROOT)$(MANDIR)/man8/fingerd.8
 
