--- ccache-2.4/ccache.h	2004-09-13 12:38:30.000000000 +0200
+++ ccache-2.4-mint/ccache.h	2010-06-09 11:23:53.642464500 +0200
@@ -2,6 +2,10 @@
 
 #include "config.h"
 
+#ifndef __MINT__
+#define HAVE_MMAP
+#endif
+
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
@@ -9,7 +13,9 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/wait.h>
+#ifdef HAVE_MMAP
 #include <sys/mman.h>
+#endif
 #include <sys/file.h>
 #include <fcntl.h>
 #include <time.h>
@@ -23,6 +29,8 @@
 #include <pwd.h>
 #endif
 
+
+
 #define STATUS_NOTFOUND 3
 #define STATUS_FATAL 4
 #define STATUS_NOCACHE 5
