--- clips-6.10/source/makefile.gfl	Fri Oct 22 18:31:33 1999
+++ clips-6.10/source/makefile	Fri Oct 22 18:31:54 1999
@@ -29,7 +29,7 @@
  	tmpltpsr.o tmpltrhs.o tmpltutl.o userdata.o utility.o watch.o
 
 .c.o :
-	gcc -c -Wall -Wpointer-arith -Wshadow -Wcast-qual \
+	gcc -c $(RPM_OPT_FLAGS) -Wall -Wpointer-arith -Wshadow -Wcast-qual \
 	    -Wcast-align -Wconversion -Wid-clash-32 -Winline \
 	    -Woverloaded-virtual -Wmissing-prototypes -Wconversion \
 	    -Wstrict-prototypes -Waggregate-return -Wno-implicit $<
