--- clips-6.10/source/makefile.x.xgfl	Fri Oct 22 18:23:19 1999
+++ clips-6.10/source/makefile.x	Fri Oct 22 18:27:20 1999
@@ -4,14 +4,16 @@
 # that xclips color are made.
 # -----------------------------------------------------
 
-WHERE_XLIBS_ARE =       /usr/X11R5/lib
-WHERE_INCL_FILES_ARE =  /usr/X11R5/include
+WHERE_XLIBS_ARE =       /usr/X11R6/lib
+WHERE_INCL_FILES_ARE =  /usr/X11R6/include
 
 #--------------------------------------------------------------------
 #  Please! do not have -O on because there is an unresolved bug either
 #  in the interface code or the optimizer that cause xclips to crash
 #--------------------------------------------------------------------
-
+#--------------------------------------------------------------------
+# I think this does not apply to gcc for Linux or MiNT.  Guido.
+#--------------------------------------------------------------------
 LDFLAGS = -L$(WHERE_XLIBS_ARE)
 INCLUDES = -I$(WHERE_INCL_FILES_ARE)
 LIBS = -lXaw -lXmu -lXt -lXext -lX11 -lm
@@ -49,8 +51,10 @@
 	xmenu_file.o xmenu_opt.o xmenu_watch.o xmenu_wind.o
 
 
+#.c.o :
+#	gcc -c $<
 .c.o :
-	gcc -c $<
+	gcc $(RPM_OPT_FLAGS) $(INCLUDES) -c $< 
 
 xclips : $(OBJS)
 	gcc -o xclips $(OBJS) $(LDFLAGS) $(LIBS)   
