--- cvsweb.cgi.orig	Thu Feb 25 14:07:17 1999
+++ cvsweb.cgi	Tue Mar  9 19:08:01 1999
@@ -44,7 +44,7 @@
 ##### Start of Configuration Area ########
 # == EDIT this == 
 # User configuration is stored in
-$config = '/usr/local/web/apache/conf/cvsweb.conf';
+$config = '/etc/httpd/conf/cvsweb.conf';
 
 # == Configuration defaults ==
 # Defaults for configuration variables that shouldn't need
--- cvsweb.conf.orig	Tue Feb 23 17:49:32 1999
+++ cvsweb.conf	Tue Mar  9 19:11:17 1999
@@ -234,7 +234,7 @@
 # cvsweb to guess the correct mime-type on
 # checkout; you can use the mime.types from
 # apache here:
-$mime_types='/usr/local/web/apache/conf/mime.types';
+$mime_types='/etc/httpd/conf/mime.types';
 
 # quick mime-type lookup; maps file-suffices to
 # mime-types for displaying checkouts in the browser.
@@ -293,7 +293,7 @@
 
 # remember to set the path to your
 # rcsutils: rlog, rcsdiff (gzip if you use compression)
-#$ENV{'PATH'}='/usr/local/bin';
+#$ENV{'PATH'}='/usr/bin';
 
 # If you have files which automatically refers to other files
 # (such as HTML) then this allows you to browse the checked
