diff -ur dialog-0.6-orig/src/Makefile dialog-0.6/src/Makefile
--- dialog-0.6-orig/src/Makefile	Mon Jan 16 00:53:42 1995
+++ dialog-0.6/src/Makefile	Tue Sep 14 14:22:58 1999
@@ -28,10 +28,10 @@
 	$(CC) $(LDFLAGS) -o $@ $(OBJS) $(LIBS)
 
 install: dialog
-	$(INSTALL) -m 755 -o root -g root dialog $(BINDIR)
+	$(INSTALL) -m 755 -o 0 -g 0 dialog $(BINDIR)
 
 install.man:
-	$(INSTALL) -m 644 -o root -g root ../man/dialog.man \
+	$(INSTALL) -m 644 -o 0 -g 0 ../man/dialog.man \
 	$(MANDIR)/man$(MANEXT)/dialog.$(MANEXT)
 
 depend:
