--- freeciv-1.11.0/configure.in.desktop	Fri Jun 30 01:01:03 2000
+++ freeciv-1.11.0/configure.in	Mon Jul 10 17:39:59 2000
@@ -516,4 +516,9 @@
 	  intl/Makefile
 	  po/Makefile.in
 	  doc/Makefile
+	  desktop/civclient-wrapper
+	  desktop/civserver-wrapper
+	  desktop/freeciv-engels.desktop
+	  desktop/freeciv-server.desktop
+	  desktop/freeciv-trident.desktop
 	  undep.sh, [ chmod +x undep.sh ; ./undep.sh ])
