--- freemint-1.18.0/CONFIGVARS	Wed Apr  3 12:07:22 2013
+++ freemint-1.18.0.orig/CONFIGVARS	Wed Apr  3 12:01:30 2013
@@ -24,7 +24,7 @@
 
 LANG=C
 NATIVECC = gcc
-NATIVECFLAGS = -Wall
+NATIVECFLAGS = -O -Wall
 CRLF = echo crlf -s
 
 # e.g. when you don't have the mintlib/gemlib/cflib in
@@ -37,7 +37,7 @@
 CROSSPREFIX=
 
 NATIVECC = gcc
-NATIVECFLAGS = -Wall
+NATIVECFLAGS = -O -Wall
 CRLF  = crlf -s
 
 endif
