--- gnubg-0.02-orig/Makefile.in	Fri Dec  3 02:40:56 1999
+++ gnubg-0.02/Makefile.in	Tue Nov 14 20:30:52 2000
@@ -80,7 +80,7 @@
 @NOX_TRUE@gnubg_DEPENDENCIES = $(COMMON_DEPS)
 @NOX_FALSE@INCLUDES = $(COMMON_INCS) -I$(srcdir)/ext @X_CFLAGS@
 @NOX_FALSE@gnubg_SOURCES = $(COMMON_SOURCES) xboard.c xboard.h xgame.c xgame.h
-@NOX_FALSE@gnubg_LDADD = -Lext -lext @X_LIBS@ @X_PRE_LIBS@ -lX11 -lXmu @X_EXTRA_LIBS@ \
+@NOX_FALSE@gnubg_LDADD = -Lext -lext @X_LIBS@ @X_PRE_LIBS@ -lXmu -lX11 @X_EXTRA_LIBS@ \
 @NOX_FALSE@	$(COMMON_LIBS)
 @NOX_FALSE@gnubg_DEPENDENCIES = $(COMMON_DEPS) ext/libext.a
 
