diff --unified --recursive --new-file gnubg-0.02.orig/configure gnubg-0.02/configure
--- gnubg-0.02.orig/configure	Fri Dec  3 02:40:02 1999
+++ gnubg-0.02/configure	Sat Apr  8 13:17:12 2000
@@ -39,6 +39,7 @@
 sbindir='${exec_prefix}/sbin'
 libexecdir='${exec_prefix}/libexec'
 datadir='${prefix}/share'
+pkgdatadir='${prefix}/share/gnubg'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
