--- groff-1.15/xditview/Imakefile.make	Sat Dec  4 20:13:33 1999
+++ groff-1.15/xditview/Imakefile	Wed Dec 29 19:16:56 1999
@@ -1,4 +1,4 @@
-GROFF_LIBDIR = /usr/local/share/groff
+GROFF_LIBDIR = /usr/lib/groff
 GROFF_FONTDIR = $(GROFF_LIBDIR)/font
 GROFF_FONTPATH = .:$(GROFF_FONTDIR):/usr/local/lib/font:/usr/lib/font
 DPIS = 75 100
--- groff-1.15/Makefile.in.make	Sat Dec 18 07:00:01 1999
+++ groff-1.15/Makefile.in	Wed Dec 29 19:16:05 1999
@@ -74,7 +74,7 @@
 bindir=$(exec_prefix)/bin
 
 # datasubdir says where to install data files
-datadir=$(prefix)/share
+datadir=$(prefix)/lib
 datasubdir=$(datadir)/groff
 
 # fontdir says where to install dev*/*.
