diff -ur info2html-1.1-orig/info2html info2html-1.1/info2html
--- info2html-1.1-orig/info2html	Thu Mar 23 01:53:40 2000
+++ info2html-1.1/info2html	Thu Mar 23 01:56:16 2000
@@ -27,7 +27,7 @@
 
 # set here the full path of the info2html.conf
 $0 =~ m!(.*/)[^/]+$!;
-$INFO2HTMLCONF = "/usr/local/etc/info2html.conf";
+$INFO2HTMLCONF = "/etc/info2html.conf";
 require($INFO2HTMLCONF);  #-- configuration settings
 
 #-- patterns
