--- libpcap-0.6.2/configure.in.mint	Wed Jan 17 19:21:54 2001
+++ libpcap-0.6.2/configure.in	Fri Sep 28 03:48:24 2001
@@ -25,9 +25,9 @@
 dnl test for it and set "HAVE_SYS_IOCCOM_H" if we have it, otherwise
 dnl "AC_LBL_FIXINCLUDES" won't work on some platforms such as Solaris.
 dnl
-AC_CHECK_HEADERS(sys/ioccom.h sys/sockio.h ifaddrs.h netinet/if_ether.h)
+dnl AC_CHECK_HEADERS(sys/ioccom.h sys/sockio.h ifaddrs.h netinet/if_ether.h)
 
-AC_LBL_FIXINCLUDES
+dnl AC_LBL_FIXINCLUDES
 
 AC_CHECK_FUNCS(ether_hostton strerror freeifaddrs strlcpy)
 
