--- tiff-v3.4/libtiff/port.h.mint	Sun Jul 18 15:40:19 1999
+++ tiff-v3.4/libtiff/port.h	Sun Jul 18 15:41:53 1999
@@ -0,0 +1,34 @@
+/*
+ * Warning, this file was automatically created by the TIFF configure script
+ * VERSION:	 v3.4033
+ * DATE:	 Sun Jul 18 15:33:08 UCT 1999
+ * TARGET:	 m68k-atari-mint1.15
+ * CCOMPILER:	 /usr/local/bin/gcc-2.8.1.c
+ */
+
+/* The MiNTLib headers seem to outsmart the TIFF configure script.  Thus,
+   we removed all auto-generated prototypes from here.  */
+#ifndef _PORT_
+#define _PORT_ 1
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include <sys/types.h>
+#define HOST_FILLORDER FILLORDER_MSB2LSB
+#define HOST_BIGENDIAN	1
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdlib.h>
+#include <fcntl.h>
+typedef double dblparam_t;
+#ifdef __STRICT_ANSI__
+#define	INLINE	__inline__
+#else
+#define	INLINE	inline
+#endif
+#define GLOBALDATA(TYPE,NAME)	extern TYPE NAME
+#ifdef __cplusplus
+}
+#endif
+#endif
