--- tiff-v3.5.5/libtiff/tif_fax3.c.steve	Thu Jul 13 12:04:26 2000
+++ tiff-v3.5.5/libtiff/tif_fax3.c	Thu Jul 13 12:05:21 2000
@@ -171,7 +171,7 @@
 	sp->bitmap =
 	    TIFFGetBitRevTable(tif->tif_dir.td_fillorder != FILLORDER_LSB2MSB);
 	if (sp->refruns) {		/* init reference line to white */
-		sp->refruns[0] = (uint16) sp->b.rowpixels;
+		sp->refruns[0] = (uint32) sp->b.rowpixels;
 		sp->refruns[1] = 0;
 	}
 	return (1);
@@ -374,7 +374,7 @@
 	for (; runs < erun; runs += 2) {
 	    run = runs[0];
 	    if (x+run > lastx || run > lastx )
-		run = runs[0] = (uint16) (lastx - x);
+		run = runs[0] = (uint32) (lastx - x);
 	    if (run) {
 		cp = buf + (x>>3);
 		bx = x&7;
