--- tiff-v3.5.5/configure	Thu May 18 18:42:43 2000
+++ tiff-v3.5.5/configure	Thu May 18 18:43:05 2000
@@ -1156,7 +1156,7 @@
 	TIFFLIBREF='-L${DEPTH}/libtiff -ltiff'
 	;;
     *-linux*)
-	if [ -r /lib/libc.so.6 || -r /lib/libc.so.5 ]; then
+	if [ -r /lib/libc.so.6 -o -r /lib/libc.so.5 ]; then
 	    DSOSUF=so.${DIST_MAJOR}
 	    DSOSUF_VERSION=${DSOSUF}.${DIST_MINOR}.${DIST_ALPHA}
 	    LIBCOPTS='-fPIC'
