--- logrotate-3.5.9/Makefile.orig	Thu Jun 21 20:17:52 2001
+++ logrotate-3.5.9/Makefile	Fri Sep 28 23:49:22 2001
@@ -45,7 +45,7 @@
 	[ -d $(PREFIX)/$(MANDIR) ] || mkdir -p $(PREFIX)/$(MANDIR)
 	[ -d $(PREFIX)/$(MANDIR)/man8 ] || mkdir -p $(PREFIX)/$(MANDIR)/man8
 
-	install -s -m 755 $(PROG) $(PREFIX)/$(BINDIR)
+	install -m 755 $(PROG) $(PREFIX)/$(BINDIR)
 	install -m 644 $(MAN) $(PREFIX)/$(MANDIR)/man`echo $(MAN) | sed "s/.*\.//"`/$(MAN)
 
 co:
