--- man-1.5e/conf_script.orig	Mon Aug 24 17:19:07 1998
+++ man-1.5e/conf_script	Mon Aug 24 17:23:51 1998
@@ -25,9 +25,9 @@
 s,@LIBS@,,
 s,@LIBOBJS@,,
 s,@troff@,/usr/bin/groff -Tps -mandoc,
-s,@nroff@,/usr/bin/groff -Tlatin1 -mandoc,
+s,@nroff@,/usr/bin/groff -Tascii -mandoc,
 s,@eqn@,/usr/bin/geqn -Tps,
-s,@neqn@,/usr/bin/geqn -Tlatin1,
+s,@neqn@,/usr/bin/geqn -Tascii,
 s,@tbl@,/usr/bin/gtbl,
 s,@nocol@,# ,
 s,@pcol@,,
@@ -60,8 +60,8 @@
 s,@apropos@,/usr/bin/apropos,
 s,@whatis@,/usr/bin/whatis,
 s,@makewhatis@,/usr/sbin/makewhatis,
-s,@man_config_dir@,/usr/lib,
-s,@man_config_file@,/usr/lib/man.conf,
+s,@man_config_dir@,/etc,
+s,@man_config_file@,/etc/man.config,
 s,@manpathoption@,--path,
 s,@grepsilent@,q,
 s/@sections@/1:8:2:3:4:5:6:7:9:tcl:n:l:p:o/
