--- mgetty-1.1.21/Makefile	Sat Jun  3 23:09:35 2000
+++ mgetty-1.1.21/Makefile	Sat Jun  3 23:09:41 2000
@@ -587,15 +587,15 @@
 	test -d $(CONFDIR) || \
 		( ./mkidirs $(CONFDIR); chmod 755 $(CONFDIR))
 	test -f $(CONFDIR)/login.config || \
-		$(INSTALL) -o root -m 600 login.config $(CONFDIR)/
+		$(INSTALL) -m 600 login.config $(CONFDIR)/
 	test -f $(CONFDIR)/mgetty.config || \
-		$(INSTALL) -o root -m 600 mgetty.config $(CONFDIR)/
+		$(INSTALL) -m 600 mgetty.config $(CONFDIR)/
 	test -f $(CONFDIR)/sendfax.config || \
-		$(INSTALL) -o root -m 644 sendfax.config $(CONFDIR)/
+		$(INSTALL) -m 644 sendfax.config $(CONFDIR)/
 	test -f $(CONFDIR)/dialin.config || \
-		$(INSTALL) -o root -m 600 dialin.config $(CONFDIR)/
+		$(INSTALL) -m 600 dialin.config $(CONFDIR)/
 	test -f $(CONFDIR)/faxrunq.config || \
-		$(INSTALL) -o root -m 600 faxrunq.config $(CONFDIR)/
+		$(INSTALL) -m 600 faxrunq.config $(CONFDIR)/
 #
 # test for outdated stuff
 #
