--- mpage-2.5.1/Makefile.config	Mon Mar 16 17:52:40 1998
+++ mpage-2.5.1/Makefile	Sat Jun 10 15:29:41 2000
@@ -46,7 +46,7 @@
 #  See glob.c for all possibilities
 #
 # PAGESIZE=Letter
-PAGESIZE=A4
+PAGESIZE=Letter
 
 #
 # Define your spooler type
@@ -56,7 +56,7 @@
 SPOOL_TYPE=BSD_SPOOLER
 
 # PREFIX=e:/usr		# OS/2 type
-PREFIX=/usr/local
+PREFIX=/usr
 BINDIR=$(PREFIX)/bin
 LIBDIR=$(PREFIX)/lib
 MANDIR=$(PREFIX)/man/man1
@@ -71,7 +71,7 @@
 #
 # Set the default sheet margin in 1/72 inches. 
 #
-SHEETMARGIN=20 
+SHEETMARGIN=18
 
 
 DEFS = -DPAGE_DEF=\"${PAGESIZE}\" -DLIBDIR=\"${LIBDIR}/mpage\" -DDEFAULT_ENCODING=${ENCODING} -DDEFAULTSMARGIN=${SHEETMARGIN} -DSPOOLER=${SPOOL_TYPE}
