--- netkit-rwall-0.17/rpc.rwalld/Makefile.orig	Sun Dec 12 19:05:04 1999
+++ netkit-rwall-0.17/rpc.rwalld/Makefile	Wed Sep 26 17:45:00 2001
@@ -12,7 +12,7 @@
 daemon.o: daemon.h
 
 install: rwalld
-	install -s -m$(DAEMONMODE) rwalld $(INSTALLROOT)$(SBINDIR)/rpc.rwalld
+	install -m$(DAEMONMODE) rwalld $(INSTALLROOT)$(SBINDIR)/rpc.rwalld
 	install -m$(MANMODE) rpc.rwalld.8 $(INSTALLROOT)$(MANDIR)/man8
 	ln -sf rpc.rwalld.8 $(INSTALLROOT)$(MANDIR)/man8/rwalld.8
 
--- netkit-rwall-0.17/rwall/Makefile.orig	Sun Dec 12 19:05:04 1999
+++ netkit-rwall-0.17/rwall/Makefile	Wed Sep 26 17:45:12 2001
@@ -11,7 +11,7 @@
 rwall.o: rwall.h ../version.h
 
 install: rwall
-	install -s -m$(BINMODE) rwall $(INSTALLROOT)$(BINDIR)
+	install -m$(BINMODE) rwall $(INSTALLROOT)$(BINDIR)
 	install -m$(MANMODE) rwall.1 $(INSTALLROOT)$(MANDIR)/man1/
 
 clean:
