--- netkit-rusers-0.17/rpc.rusersd/Makefile.orig	Sun Dec 12 19:05:02 1999
+++ netkit-rusers-0.17/rpc.rusersd/Makefile	Wed Sep 26 17:33:50 2001
@@ -22,7 +22,7 @@
 	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
 
 install: rusersd
-	install -s -m$(DAEMONMODE) rusersd $(INSTALLROOT)$(SBINDIR)/rpc.rusersd
+	install -m$(DAEMONMODE) rusersd $(INSTALLROOT)$(SBINDIR)/rpc.rusersd
 	install -m$(MANMODE) rpc.rusersd.8 $(INSTALLROOT)$(MANDIR)/man8
 	ln -sf rpc.rusersd.8 $(INSTALLROOT)$(MANDIR)/man8/rusersd.8
 
--- netkit-rusers-0.17/rup/Makefile.orig	Wed Sep 26 17:18:40 2001
+++ netkit-rusers-0.17/rup/Makefile	Wed Sep 26 17:34:02 2001
@@ -22,7 +22,7 @@
 rup.o: ../version.h
 
 install: rup
-	install -s -m$(BINMODE) rup $(INSTALLROOT)$(BINDIR)
+	install -m$(BINMODE) rup $(INSTALLROOT)$(BINDIR)
 	install -m$(MANMODE) rup.1 $(INSTALLROOT)$(MANDIR)/man1
 
 clean:
--- netkit-rusers-0.17/rusers/Makefile.orig	Sun Dec 12 19:05:04 1999
+++ netkit-rusers-0.17/rusers/Makefile	Wed Sep 26 17:34:10 2001
@@ -20,7 +20,7 @@
 	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
 
 install: rusers
-	install -s -m$(BINMODE) rusers $(INSTALLROOT)$(BINDIR)
+	install -m$(BINMODE) rusers $(INSTALLROOT)$(BINDIR)
 	install -m$(MANMODE) rusers.1 $(INSTALLROOT)$(MANDIR)/man1
 
 clean:
