--- netkit-ftp-0.17-pre20000412/configure.mint	Sun Dec 12 19:35:56 1999
+++ netkit-ftp-0.17-pre20000412/configure	Tue Apr  3 13:18:16 2001
@@ -71,7 +71,7 @@
     echo -n 'Looking for a C compiler... '
     for TRY in egcs gcc g++ CC c++ cc; do
        (
-           $TRY __conftest.c -o __conftest || exit 1;
+           $TRY __conftest.c -o __conftest $LIBS || exit 1;
            ./__conftest || exit 1;
        ) >/dev/null 2>&1 || continue;
        CC=$TRY
@@ -87,7 +87,7 @@
 else
     echo -n 'Checking if C compiler works... '
     if (
-          $CC __conftest.c -o __conftest || exit 1
+          $CC __conftest.c -o __conftest $LIBS || exit 1
           ./__conftest || exit 1
        ) >/dev/null 2>&1; then
          echo 'yes'
@@ -101,7 +101,7 @@
 
 echo -n "Checking if $CC accepts gcc warnings... "
 if (
-    $CC $WARNINGS __conftest.c -o __conftest || exit 1
+    $CC $WARNINGS __conftest.c -o __conftest $LIBS || exit 1
    ) >/dev/null 2>&1; then
      echo 'yes'
      CC_WARNINGS=1
@@ -112,7 +112,7 @@
 if [ x$DEBUG = x ]; then
     echo -n "Checking if $CC accepts -O2... "
     if (
-         $CC -O2 __conftest.c -o __conftest
+         $CC -O2 __conftest.c -o __conftest $LIBS
        ) >/dev/null 2>&1; then
          echo 'yes'
          CFLAGS="$CFLAGS -O2"
@@ -120,7 +120,7 @@
          echo 'no'
          echo -n "Checking if $CC accepts -O... "
          if (
-              $CC -O __conftest.c -o __conftest
+              $CC -O __conftest.c -o __conftest $LIBS
             ) >/dev/null 2>&1; then
               echo 'yes'
               CFLAGS="$CFLAGS -O"
@@ -132,7 +132,7 @@
 else
     echo -n "Checking if $CC accepts -g... "
     if (
-         $CC -g __conftest.c -o __conftest
+         $CC -g __conftest.c -o __conftest $LIBS
        ) >/dev/null 2>&1; then
          echo 'yes'
          CFLAGS="$CFLAGS -g"
@@ -142,9 +142,6 @@
 
 fi
 
-LDFLAGS=
-LIBS=
-
 rm -f __conftest*
 
 ##################################################
@@ -167,13 +164,13 @@
 
 EOF
 if (
-      $CC $CFLAGS  __conftest.c  -o __conftest || exit 1
+      $CC $CFLAGS  __conftest.c  -o __conftest $LIBS || exit 1
       ./__conftest || exit 1
    ) >/dev/null 2>&1; then
     echo 'yes'
 else
     if (
-          $CC $CFLAGS -D__USE_BSD_SIGNAL __conftest.c  -o __conftest || exit 1
+          $CC $CFLAGS -D__USE_BSD_SIGNAL __conftest.c  -o __conftest $LIBS || exit 1
           ./__conftest || exit 1
        ) >/dev/null 2>&1; then
         echo '-D__USE_BSD_SIGNAL'
@@ -203,13 +200,13 @@
 
 EOF
 if (
-      $CC $CFLAGS  __conftest.c -lncurses -o __conftest || exit 1
+      $CC $CFLAGS  __conftest.c -lncurses -o __conftest $LIBS || exit 1
    ) >/dev/null 2>&1; then
     echo 'yes'
     NCURSES=1
 else
     if (
-          $CC $CFLAGS -I/usr/include/ncurses __conftest.c -lncurses -o __conftest || exit 1
+          $CC $CFLAGS -I/usr/include/ncurses __conftest.c -lncurses -o __conftest $LIBS || exit 1
        ) >/dev/null 2>&1; then
         echo '-I/usr/include/ncurses'
         CFLAGS="$CFLAGS -I/usr/include/ncurses"
@@ -232,7 +229,7 @@
 
 EOF
     if (
-          $CC $CFLAGS  __conftest.c -ltermcap -o __conftest || exit 1
+          $CC $CFLAGS  __conftest.c -ltermcap -o __conftest $LIBS || exit 1
        ) >/dev/null 2>&1; then
         echo '-ltermcap'
         LIBTERMCAP=-ltermcap
@@ -257,7 +254,7 @@
 
 EOF
 if (
-      $CC $CFLAGS  __conftest.c  -o __conftest || exit 1
+      $CC $CFLAGS  __conftest.c  -o __conftest $LIBS || exit 1
    ) >/dev/null 2>&1; then
     echo 'yes'
     USE_GLIBC=1
@@ -277,7 +274,7 @@
 
 EOF
 if (
-      $CC $CFLAGS  __conftest.c -lreadline $LIBTERMCAP -o __conftest || exit 1
+      $CC $CFLAGS  __conftest.c -lreadline $LIBTERMCAP -o __conftest $LIBS || exit 1
    ) >/dev/null 2>&1; then
     echo 'yes'
     USE_READLINE=1
@@ -302,18 +299,18 @@
 
 EOF
 if (
-      $CC $CFLAGS  __conftest.c  -o __conftest || exit 1
+      $CC $CFLAGS  __conftest.c  -o __conftest $LIBS || exit 1
    ) >/dev/null 2>&1; then
     echo 'yes'
 else
     if (
-          $CC $CFLAGS -Dsocklen_t=int __conftest.c  -o __conftest || exit 1
+          $CC $CFLAGS -Dsocklen_t=int __conftest.c  -o __conftest $LIBS || exit 1
        ) >/dev/null 2>&1; then
         echo 'int'
         CFLAGS="$CFLAGS -Dsocklen_t=int"
     else
         if (
-              $CC $CFLAGS -Dsocklen_t=size_t __conftest.c  -o __conftest || exit 1
+              $CC $CFLAGS -Dsocklen_t=size_t __conftest.c  -o __conftest $LIBS || exit 1
            ) >/dev/null 2>&1; then
             echo 'size_t'
             CFLAGS="$CFLAGS -Dsocklen_t=size_t"
@@ -340,12 +337,12 @@
 
 EOF
 if (
-      $CC $CFLAGS  __conftest.c  -o __conftest || exit 1
+      $CC $CFLAGS  __conftest.c  -o __conftest $LIBS || exit 1
    ) >/dev/null 2>&1; then
     echo 'ok'
 else
     if (
-          $CC $CFLAGS -D_GNU_SOURCE __conftest.c  -o __conftest || exit 1
+          $CC $CFLAGS -D_GNU_SOURCE __conftest.c  -o __conftest $LIBS || exit 1
           ./__conftest || exit 1
        ) >/dev/null 2>&1; then
         echo '-D_GNU_SOURCE'
@@ -376,20 +373,20 @@
 
 EOF
 if (
-      $CC $CFLAGS  __conftest.c $(LIBBSD) -o __conftest || exit 1
+      $CC $CFLAGS  __conftest.c $(LIBBSD) -o __conftest $LIBS || exit 1
       ./__conftest || exit 1
    ) >/dev/null 2>&1; then
     echo 'ok'
 else
     if (
-          $CC $CFLAGS  __conftest.c -lsnprintf $(LIBBSD) -o __conftest || exit 1
+          $CC $CFLAGS  __conftest.c -lsnprintf $(LIBBSD) -o __conftest $LIBS || exit 1
           ./__conftest || exit 1
        ) >/dev/null 2>&1; then
         echo '-lsnprintf'
         LIBS="$LIBS -lsnprintf"
     else
         if (
-              $CC $CFLAGS  __conftest.c -ldb $(LIBBSD) -o __conftest || exit 1
+              $CC $CFLAGS  __conftest.c -ldb $(LIBBSD) -o __conftest $LIBS || exit 1
               ./__conftest || exit 1
            ) >/dev/null 2>&1; then
             echo '-ldb'
--- netkit-ftp-0.17-pre20000412/ftp/glob.c.orig	Sat Oct  2 15:25:22 1999
+++ netkit-ftp-0.17-pre20000412/ftp/glob.c	Tue Apr  3 23:24:28 2001
@@ -282,12 +282,14 @@
 			return;
 		goto patherr2;
 	}
+#ifndef __MINT__
 	if (fstat(dirfd(dirp), &stb) < 0)
 		goto patherr1;
 	if (!isdir(stb)) {
 		errno = ENOTDIR;
 		goto patherr1;
 	}
+#endif
 	while ((dp = readdir(dirp)) != NULL) {
 		if (dp->d_ino == 0)
 			continue;
