--- netkit-rwho-0.17/ruptime/Makefile.orig	Sun Aug  1 08:00:12 1999
+++ netkit-rwho-0.17/ruptime/Makefile	Wed Sep 26 17:45:28 2001
@@ -9,7 +9,7 @@
 	$(CC) $(LDFLAGS) $^ $(LIBS) -o $@
 
 install: ruptime
-	install -s -m$(BINMODE) ruptime $(INSTALLROOT)$(BINDIR)
+	install -m$(BINMODE) ruptime $(INSTALLROOT)$(BINDIR)
 	install -m$(MANMODE) ruptime.1 $(INSTALLROOT)$(MANDIR)/man1/
 
 clean:
--- netkit-rwho-0.17/rwho/Makefile.orig	Wed Sep 26 17:22:32 2001
+++ netkit-rwho-0.17/rwho/Makefile	Wed Sep 26 17:45:40 2001
@@ -9,7 +9,7 @@
 rwho.o: ../version.h
 
 install: rwho
-	install -s -m$(BINMODE) rwho $(INSTALLROOT)$(BINDIR)
+	install -m$(BINMODE) rwho $(INSTALLROOT)$(BINDIR)
 	install -m$(MANMODE) rwho.1 $(INSTALLROOT)$(MANDIR)/man1/
 
 clean:
--- netkit-rwho-0.17/rwhod/Makefile.orig	Wed Sep 26 17:22:32 2001
+++ netkit-rwho-0.17/rwhod/Makefile	Wed Sep 26 17:45:54 2001
@@ -11,7 +11,7 @@
 $(OBJS): daemon.h
 
 install: rwhod
-	install -s -m$(DAEMONMODE) rwhod $(INSTALLROOT)$(SBINDIR)
+	install -m$(DAEMONMODE) rwhod $(INSTALLROOT)$(SBINDIR)
 	install -m$(MANMODE) rwhod.8 $(INSTALLROOT)$(MANDIR)/man8/
 
 clean:
