--- netpbm-9.5/ppm/xpmtoppm.c.orig	Fri Jun  9 09:33:32 2000
+++ netpbm-9.5/ppm/xpmtoppm.c	Sat Dec 23 14:16:26 2000
@@ -32,7 +32,7 @@
 #include "ppm.h"
 
 void ReadXPMFile();
-static void getline();
+static void my_getline();
 
 /* number of xpmColorKeys */
 #define NKEYS 5
@@ -142,22 +142,22 @@
     /* First try to read as an XPM version 3 file */
 
     /* Read the header line */
-    getline(line, sizeof(line), stream);
+    my_getline(line, sizeof(line), stream);
     if (sscanf(line, "/* %s */", str1) == 1
 	&& !strncmp(str1, "XPM", 3)) {
 
 	/* Read the assignment line */
-	getline(line, sizeof(line), stream);
+	my_getline(line, sizeof(line), stream);
 	if (strncmp(line, "static char", 11))
 	    pm_error("error scanning assignment line", 0, 0, 0, 0, 0);
 
 	/* Read the hints line */
-	getline(line, sizeof(line), stream);
+	my_getline(line, sizeof(line), stream);
 	/* skip the comment line if any */
 	if (!strncmp(line, "/*", 2)) {
 	    while (!strstr(line, "*/"))
-		getline(line, sizeof(line), stream);
-	    getline(line, sizeof(line), stream);
+		my_getline(line, sizeof(line), stream);
+	    my_getline(line, sizeof(line), stream);
 	}
 	if (sscanf(line, "\"%d %d %d %d\",", widthP, heightP,
 		   ncolorsP, chars_per_pixelP) != 4)
@@ -178,10 +178,10 @@
 
 	/* Read the color table */
 	for (i = 0; i < *ncolorsP; i++) {
-	    getline(line, sizeof(line), stream);
+	    my_getline(line, sizeof(line), stream);
 	    /* skip the comment line if any */
 	    if (!strncmp(line, "/*", 2))
-		getline(line, sizeof(line), stream);
+		my_getline(line, sizeof(line), stream);
 
 	    /* read the chars */
 	    if ((t1 = strchr(line, '"')) == NULL)
@@ -264,17 +264,17 @@
 		pm_error("error scanning color table", 0, 0, 0, 0, 0);
 	}
 	/* Read pixels. */
-	getline(line, sizeof(line), stream);
+	my_getline(line, sizeof(line), stream);
 	/* skip the comment line if any */
 	if (!strncmp(line, "/*", 2))
-	    getline(line, sizeof(line), stream);
+	    my_getline(line, sizeof(line), stream);
 
     } else {				/* try as an XPM version 1 file */
 
 	/* Read the initial defines. */
 	for (;;) {
 	    if (flag)
-		getline(line, sizeof(line), stream);
+		my_getline(line, sizeof(line), stream);
 	    else
 		flag++;
 
@@ -320,7 +320,7 @@
 	/* If there's a monochrome color table, skip it. */
 	if (!strncmp(t1, "mono", 4)) {
 	    for (;;) {
-		getline(line, sizeof(line), stream);
+		my_getline(line, sizeof(line), stream);
 		if (!strncmp(line, "static char", 11))
 		    break;
 	    }
@@ -340,7 +340,7 @@
 
 	/* Read color table. */
 	for (i = 0; i < *ncolorsP; ++i) {
-	    getline(line, sizeof(line), stream);
+	    my_getline(line, sizeof(line), stream);
 
 	    if ((t1 = strchr(line, '"')) == NULL)
 		pm_error("error scanning color table", 0, 0, 0, 0, 0);
@@ -376,7 +376,7 @@
 
 	/* Read pixels. */
 	for (;;) {
-	    getline(line, sizeof(line), stream);
+	    my_getline(line, sizeof(line), stream);
 	    if (!strncmp(line, "static char", 11))
 		break;
 	}
@@ -389,7 +389,7 @@
     ptr = *dataP;
     for (;;) {
 	if (flag)
-	    getline(line, sizeof(line), stream);
+	    my_getline(line, sizeof(line), stream);
 	else
 	    flag++;
 
@@ -425,7 +425,7 @@
 
 
 static void
-getline(line, size, stream)
+my_getline(line, size, stream)
     char *line;
     int size;
     FILE *stream;
--- netpbm-9.5/ppm/xvminitoppm.c.orig	Sun Mar 19 07:18:30 2000
+++ netpbm-9.5/ppm/xvminitoppm.c	Sat Dec 23 14:17:52 2000
@@ -14,7 +14,7 @@
 #include "ppm.h"
 #define BUFSIZE 256
 
-static void getline ARGS((FILE *fp, char *buf));
+static void my_getline ARGS((FILE *fp, char *buf));
 
 int 
 main(argc, argv)
@@ -48,18 +48,18 @@
                 i++;
             }
     
-    getline(ifp, buf);
+    my_getline(ifp, buf);
     if( strncmp(buf, "P7 332", 6) != 0 )
         pm_error("bad magic number - not a XV thumbnail picture");
 
     while(1) {
-        getline(ifp, buf);
+        my_getline(ifp, buf);
         if( strncmp(buf, "#END_OF_COMMENTS", 16)==0 )
             break;
         if( strncmp(buf, "#BUILTIN", 8)==0 )
             pm_error("cannot convert builtin XV thumbnail pictures");
     }
-    getline(ifp, buf);
+    my_getline(ifp, buf);
     if( sscanf(buf, "%d %d %d", &cols, &rows, &maxval) != 3 ) 
         pm_error("error parsing dimension info");
     if( maxval != 255 )
@@ -85,7 +85,7 @@
 
 
 static void
-getline(fp, buf)
+my_getline(fp, buf)
     FILE *fp;
     char *buf;
 {
