--- netpbm-9.5/Makefile.common.orig	Fri Jun 30 21:59:36 2000
+++ netpbm-9.5/Makefile.common	Sat Dec 23 15:19:58 2000
@@ -213,19 +213,19 @@
 # because STATICLIB just tells how the Netpbm programs are to be
 # built.  If a previous make STATICLIB=N didn't build the static
 # library, the dependency here will do that.
-.PHONY: install.staticlib
-install.staticlib: lib$(LIBROOT).a
+.PHONY: install.lib.common
+install.lib.common: lib$(LIBROOT).a
 	$(INSTALL) -c -m $(INSTALL_PERM_LIBS) $< $(INSTALLLIBS)/$<
 
-.PHONY: install.lib.common
-ifeq ($(NETPBMLIBSUFFIX),so)
+#.PHONY: install.lib.common
+#ifeq ($(NETPBMLIBSUFFIX),so)
 # install a shared library
-install.lib.common: lib$(LIBROOT).so.$(MAJ).$(MIN) 
-	cd $(INSTALLLIBS) ; rm -f lib$(LIBROOT).so.$(MAJ).*
-	$(INSTALL) -c -m $(INSTALL_PERM_LIBD) $< $(INSTALLLIBS)
-	cd $(INSTALLLIBS) ; rm -f lib$(LIBROOT).so; ln -s $< lib$(LIBROOT).so
-	cd $(INSTALLLIBS) ; rm -f $(SONAME); ln -s $< $(SONAME)
-endif
+#install.lib.common: lib$(LIBROOT).so.$(MAJ).$(MIN) 
+#	cd $(INSTALLLIBS) ; rm -f lib$(LIBROOT).so.$(MAJ).*
+#	$(INSTALL) -c -m $(INSTALL_PERM_LIBD) $< $(INSTALLLIBS)
+#	cd $(INSTALLLIBS) ; rm -f lib$(LIBROOT).so; ln -s $< lib$(LIBROOT).so
+#	cd $(INSTALLLIBS) ; rm -f $(SONAME); ln -s $< $(SONAME)
+#endif
 
 .PHONY: clean.common
 clean.common:
