--- netpbm-9.5/ppm/ppmtompeg/Makefile.orig	Tue Jun 13 19:07:48 2000
+++ netpbm-9.5/ppm/ppmtompeg/Makefile	Sat Dec 23 14:35:16 2000
@@ -67,12 +67,12 @@
 
 ppmtompeg: $(MP_ALL_OBJS) $(NETPBMLIBS) $(LIBOPT)
 	$(LD) -o $@ $(LDFLAGS) $(INCLUDE) \
-	  $(MP_ALL_OBJS) `$(LIBOPT) $(NETPBMLIBS)` $(JPEGLD) $(NETWORKLD) -lm \
+	  $(MP_ALL_OBJS) `$(LIBOPT) $(NETPBMLIBS)` $(JPEGLD) $(NETWORKLD) -lm -lsocket \
 	  $(CDEBUG)
 
 profile: $(MP_ALL_OBJS) $(NETPBMLIBS) $(LIBOPT)
 	$(LD) -o $@ -Bstatic -pg $(LDFLAGS) $(INCLUDE) \
-	  $(MP_ALL_OBJS) `$(LIBOPT) $(NETPBMLIBS)` $(JPEGLD) $(NETWORKLD) -lm \
+	  $(MP_ALL_OBJS) `$(LIBOPT) $(NETPBMLIBS)` $(JPEGLD) $(NETWORKLD) -lm -lsocket \
 	  $(CDEBUG)
 
 
