--- nfs-server-2.2beta47/config.h.in.orig	Fri Jun 11 12:01:22 1999
+++ nfs-server-2.2beta47/config.h.in	Fri Sep 28 12:03:44 2001
@@ -101,27 +101,6 @@
 /* Define if your rpc/xdr.h declares the xdrproc_t type. */
 #undef HAVE_XDRPROC_T
 
-/* Define if your system uses the OSF/1 method of getting the mount list.  */
-#undef MOUNTED_GETFSSTAT
-
-/* Define if your system uses the SysVr4 method of getting the mount list.  */
-#undef MOUNTED_GETMNTENT2
-
-/* Define if your system uses the AIX method of getting the mount list.  */
-#undef MOUNTED_VMOUNT
-
-/* Define if your system uses the SysVr3 method of getting the mount list.  */
-#undef MOUNTED_FREAD_FSTYP
-
-/* Define if your system uses the BSD4.3 method of getting the mount list.  */
-#undef MOUNTED_GETMNTENT1
-
-/* Define if your system uses the BSD4.4 method of getting the mount list.  */
-#undef MOUNTED_GETMNTINFO
-
-/* Define if your system uses the Ultrix method of getting the mount list.  */
-#undef MOUNTED_GETMNT
-
 /* Define if your system uses the OSF/1 method of getting fs usage.  */
 #undef STATFS_OSF1
 
