diff -rup openssl-1.0.0/Configure openssl-1.0.0-mint/Configure
--- openssl-1.0.0/Configure	Tue Jan 19 16:40:54 2010
+++ openssl-1.0.0-mint/Configure	Mon May 17 16:54:56 2010
@@ -563,6 +563,11 @@ my %table=(
 ##### GNU Hurd
 "hurd-x86",  "gcc:-DL_ENDIAN -DTERMIOS -O3 -fomit-frame-pointer -march=i486 -Wall::-D_REENTRANT::-ldl:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:linux-shared:-fPIC",
 
+##### Atari FreeMiNT
+"m68k-mint", "gcc:-DB_ENDIAN -DTERMIOS -O3 ::(unknown):::BN_LLONG:::",
+"m680x0-mint", "gcc:-DB_ENDIAN -DTERMIOS -m68020-60 -O3 ::(unknown):::BN_LLONG:::",
+"cf-mint", "gcc:-DB_ENDIAN -DTERMIOS -mcpu=5475 -O3 ::(unknown):::BN_LLONG:::",
+
 ##### OS/2 EMX
 "OS2-EMX", "gcc::::::::",
 
diff -rup openssl-1.0.0/crypto/bn/bn_div.c openssl-1.0.0-mint/crypto/bn/bn_div.c
--- openssl-1.0.0/crypto/bn/bn_div.c	Tue Feb 23 05:36:30 2010
+++ openssl-1.0.0-mint/crypto/bn/bn_div.c	Fri May 21 18:13:56 2010
@@ -164,6 +164,21 @@ int BN_div(BIGNUM *dv, BIGNUM *rem, cons
 	    q;					\
 	})
 #  define REMAINDER_IS_ALREADY_CALCULATED
+#  elif defined(__mc68020__)
+   /* Same rationale as for i386.  On the 68020-60/CPU32 the divu
+    * instruction will take a 64 bit dividend in a register pair
+    * and overwrite that pair with the remainder:quotient.
+    *                                  hyc@highlandsun.com
+    */
+#  define bn_div_words(n0,n1,d0)                \
+        ({  asm volatile (                      \
+                "divu.l %4,%2:%3"               \
+                : "=d"(rem), "=d"(q)            \
+                : "0"(n0), "1"(n1), "d"(d0)     \
+                : "cc");                        \
+            q;                                  \
+        })
+#  define REMAINDER_IS_ALREADY_CALCULATED
 #  endif /* __<cpu> */
 # endif /* __GNUC__ */
 #endif /* OPENSSL_NO_ASM */
diff -rup openssl-1.0.0/e_os.h openssl-1.0.0-mint/e_os.h
--- openssl-1.0.0/e_os.h	Wed Aug 26 10:13:42 2009
+++ openssl-1.0.0-mint/e_os.h	Mon May 17 19:36:56 2010
@@ -629,7 +629,7 @@ static unsigned int _strlen31(const char
  * versions.
  */
 #  if !defined(OPENSSL_USE_IPV6)
-#    if defined(AF_INET6) && !defined(OPENSSL_SYS_BEOS_BONE) && !defined(NETWARE_CLIB)
+#    if defined(AF_INET6) && !defined(OPENSSL_SYS_BEOS_BONE) && !defined(NETWARE_CLIB) && !defined(__MINT__)
 #      define OPENSSL_USE_IPV6 1
 #    else
 #      define OPENSSL_USE_IPV6 0
