--- pine4.30/imap/src/osdep/unix/env_unix.c.boguswarning	Tue Oct 31 14:52:40 2000
+++ pine4.30/imap/src/osdep/unix/env_unix.c	Tue Oct 31 14:53:01 2000
@@ -65,7 +65,7 @@
 static long locktimeout = 5;	/* default lock timeout */
 				/* flock() emulator is a no-op */
 static long disableFcntlLock = NIL;
-static long lockEaccesError = T;/* warning on EACCES errors on .lock files */
+static long lockEaccesError = NIL;/* warning on EACCES errors on .lock files */
 static long hideDotFiles = NIL;	/* hide files whose names start with . */
 				/* default prototypes */
 static MAILSTREAM *createProto = NIL;
