--- pine4.31/imap/src/osdep/unix/Makefile.ssl.orig	Sat Oct 28 21:16:48 2000
+++ pine4.31/imap/src/osdep/unix/Makefile.ssl	Tue Dec 12 02:14:36 2000
@@ -20,10 +20,10 @@
 
 # Extended flags needed for additional authenticators.  You may need to modify.
 
-SSLDIR=/usr/local/ssl
+SSLDIR=/usr/share/ssl
 SSLCERTS=$(SSLDIR)/certs
-SSLINCLUDE=$(SSLDIR)/include
-SSLLIB=$(SSLDIR)/lib
+SSLINCLUDE=/usr/include
+SSLLIB=/usr/lib
 
 SSLCRYPTO=-lcrypto
 
