diff -urN rdate-1.0/Makefile rdate-1.0.new/Makefile
--- rdate-1.0/Makefile	Fri Feb  4 20:07:08 2000
+++ rdate-1.0.new/Makefile	Wed Aug  9 16:35:26 2000
@@ -15,6 +15,8 @@
 	install -s -m 555 rdate $(prefix)/bin
 	install -m 444 rdate.1 $(prefix)/man/man1
 
+clean:
+	rm -f rdate
 dist:
 	rm -rf $(PACKAGE)-$(VERSION)
 	mkdir $(PACKAGE)-$(VERSION)
diff -urN rdate-1.0/rdate.c rdate-1.0.new/rdate.c
--- rdate-1.0/rdate.c	Wed Aug  9 15:14:23 2000
+++ rdate-1.0.new/rdate.c	Wed Aug  9 15:30:23 2000
@@ -34,6 +34,7 @@
 
 /* difference between Unix time and net time */
 #define BASE1970	2208988800L
+#define DEFAULT_PORT    37
 
 static int
 rdate(const char *hostname, time_t *retval)
@@ -64,12 +65,10 @@
       memcpy(&saddr.sin_addr, hent->h_addr_list[0], hent->h_length);
     }
 
-  if(!(sent = getservbyname("time", "tcp")))
-    {
-      fprintf(stderr, "%s: time/tcp: service not found\n", program_invocation_short_name);
-      return -1;
-    }
-  saddr.sin_port = sent->s_port;
+  if((sent = getservbyname("time", "tcp")))
+    saddr.sin_port = sent->s_port;      
+  else
+    saddr.sin_port = htons(DEFAULT_PORT);
 
   fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
   if(fd < 0)
