--- shadow-20000902/configure.in.orig	Thu Sep  6 14:23:40 2001
+++ shadow-20000902/configure.in	Thu Sep  6 14:24:44 2001
@@ -113,6 +113,7 @@
 AC_CHECK_FUNCS(a64l fchmod fchown fsync getgroups gethostname getspnam)
 AC_CHECK_FUNCS(gettimeofday getusershell getutent initgroups lchown lckpwdf)
 AC_CHECK_FUNCS(lstat memcpy memset setgroups sigaction strchr updwtmp updwtmpx)
+AC_CHECK_FUNCS(innetgr)
 
 AC_REPLACE_FUNCS(mkdir putgrent putpwent putspent rename rmdir)
 AC_REPLACE_FUNCS(sgetgrent sgetpwent sgetspent)
