--- shadow-19990827/src/useradd.c.group	Wed Jan 19 17:40:48 2000
+++ shadow-19990827/src/useradd.c	Wed Jan 19 17:42:04 2000
@@ -1929,6 +1929,19 @@
 	}
 
 	/*
+	 * Don't blindly overwrite a group when a user is added...
+	 * If you already have a group username, and want to add the user
+	 * to that group, use useradd -g username username.
+	 * --bero
+	 */
+	if (! (nflg || gflg)) {
+	    if (getgrnam(user_name)) {
+                fprintf(stderr, _("%s: group %s exists - if you want to add this user to that group, use -g.\n"), Prog, user_name);
+		exit(E_NAME_IN_USE);
+	    }
+	}
+
+	/*
 	 * Do the hard stuff - open the files, create the user entries,
 	 * create the home directory, then close and update the files.
 	 */
