--- sudo.v1.5.9p1/configure.fixvi	Mon Apr 26 14:06:34 1999
+++ sudo.v1.5.9p1/configure	Mon Apr 26 14:07:53 1999
@@ -4321,28 +4321,22 @@
 
 echo $ac_n "checking for vi""... $ac_c" 1>&6
 echo "configure:4324: checking for vi" >&5
-if test -f "/usr/bin/vi"; then
-    echo "$ac_t""/usr/bin/vi" 1>&6
+if test -f "/bin/vi"; then
+    echo "$ac_t""/bin/vi" 1>&6
     cat >> confdefs.h <<\EOF
-#define _PATH_VI "/usr/bin/vi"
+#define _PATH_VI "/bin/vi"
 EOF
 
-elif test -f "/usr/ucb/vi"; then
-    echo "$ac_t""/usr/ucb/vi" 1>&6
+elif test -f "/usr/bin/vi"; then
+    echo "$ac_t""/usr/bin/vi" 1>&6
     cat >> confdefs.h <<\EOF
-#define _PATH_VI "/usr/ucb/vi"
+#define _PATH_VI "/usr/bin/vi"
 EOF
 
 elif test -f "/usr/bsd/vi"; then
     echo "$ac_t""/usr/bsd/vi" 1>&6
     cat >> confdefs.h <<\EOF
 #define _PATH_VI "/usr/bsd/vi"
-EOF
-
-elif test -f "/bin/vi"; then
-    echo "$ac_t""/bin/vi" 1>&6
-    cat >> confdefs.h <<\EOF
-#define _PATH_VI "/bin/vi"
 EOF
 
 elif test -f "/usr/local/bin/vi"; then
