--- sysklogd-1.3/ksym_mod.c.broken	Thu Oct 23 18:12:16 1997
+++ sysklogd-1.3/ksym_mod.c	Thu Oct 23 18:12:33 1997
@@ -502,6 +502,7 @@
 	if ( num_modules == 0 )
 		return((char *) 0);
 	
+#if !defined(__sparc__)
 	for(nmod= 0; nmod < num_modules; ++nmod)
 	{
 		mp = &sym_array_modules[nmod];
@@ -571,6 +572,7 @@
 			return(mp->name);
 		}
 	}
+#endif	/* !defined(sparc) */
 
 	/* It has been a hopeless exercise. */
 	return((char *) 0);
