--- tcltk-8.0.5/tclX8.0.4/unix/Config.mk.jbj	Mon May  3 16:42:11 1999
+++ tcltk-8.0.5/tclX8.0.4/unix/Config.mk	Mon May  3 16:42:33 1999
@@ -56,8 +56,8 @@
 # to ${TCL_LIB_SPEC}.
 # 
 
-TCL_SRC=${srcbasedir}/../tcl8.0.4
-TCL_BUILD=${bldbasedir}/../tcl8.0.4
+TCL_SRC=${srcbasedir}/../tcl8.0.5
+TCL_BUILD=${bldbasedir}/../tcl8.0.5
 TCL_CONFIG=${TCL_BUILD}/unix/tclConfig.sh
 TCL_LIB=${TCL_BUILD_LIB_SPEC}
 
@@ -68,8 +68,8 @@
 # neccessary to link with the Tk shared library (libtk8.0.so).  If you want to
 # use the installed Tk library, set TK_LIB to ${TK_LIB_SPEC}.
 
-TK_SRC=${srcbasedir}/../tk8.0.4
-TK_BUILD=${bldbasedir}/../tk8.0.4
+TK_SRC=${srcbasedir}/../tk8.0.5
+TK_BUILD=${bldbasedir}/../tk8.0.5
 TK_CONFIG=${TK_BUILD}/unix/tkConfig.sh
 TK_LIB=${TK_BUILD_LIB_SPEC}
 
