--- termcap.mint	Fri Jul 30 18:41:19 1999
+++ termcap	Fri Jul 30 18:47:33 1999
@@ -254,6 +254,98 @@
 	:co#132:li#66:\
 	:bl=^G:cr=^M:do=^J:ff=^L:le=^H:sf=^J:
 
+#############################################################################
+#
+# Atari ST terminals.
+# From Guido Flohr <gufl0000@stud.uni-sb.de>.
+#
+tw52|tw52-color|Toswin window manager with color:\
+	:ut:\
+	:Co#16:pa#256:\
+	:AB=\Ec%?%p1%{0}%=%t?%e%p1%{7}%=%t0%e%p1%{15}%=%t7%e%p1%{48}%+%c:\
+	:AF=\Eb%?%p1%{0}%=%t?%e%p1%{7}%=%t0%e%p1%{15}%=%t7%e%p1%{48}%+%c:\
+	:..Sb=\Ec%?%p1%{0}%=%t?%e%p1%{7}%=%t0%e%p1%{15}%=%t7%e%p1%{48}%+%c:\
+	:..Sf=\Eb%?%p1%{0}%=%t?%e%p1%{7}%=%t0%e%p1%{15}%=%t7%e%p1%{48}%+%c:\
+	:oc=\Eb?\Ec0:op=\Eb?\Ec0:\
+	:tc=tw52-m:
+tw52-m|Toswin window manager monochrome:\
+	:ul:\
+	:ma#999:\
+	:dc=\Ea:is=\Ev\Eq\Ez_\Ee\Ei\Eb?\Ec0:md=\Eya:me=\Ez_:\
+	:mh=\EyB:mr=\EyP:r2=\Ev\Eq\Ez_\Ee\Ei\Eb?\Ec0:se=\EzQ:\
+	:so=\EyQ:ue=\EzH:us=\EyH:\
+	:tc=at-m:
+tt52|Atari TT medium and high resolution:\
+	:li#30:tc=at-color:
+# (untranslatable capabilities removed to fit entry within 1023 bytes)
+st52-color|at-color|atari-color|atari_st-color|Atari ST with color:\
+	:ut:\
+	:Co#16:pa#256:\
+	:AB=\Ec%?%p1%{0}%=%t1%e%p1%{1}%=%t2%e%p1%{2}%=%t3%e%p1%{3}%=%t>%e%p1%{4}%=%t4%e%p1%{5}%=%t7%e%p1%{6}%=%t5%e%p1%{7}%=%t0%e%p1%{8}%=%t8%e%p1%{9}%=%t9%e%p1%{10}%=%t\072%e%p1%{11}%=%t;%e%p1%{12}%=%t<%e%p1%{13}%=%t=%e%p1%{14}%=%t6%e?:\
+	:AF=\Eb%?%p1%{0}%=%t1%e%p1%{1}%=%t2%e%p1%{2}%=%t3%e%p1%{3}%=%t>%e%p1%{4}%=%t4%e%p1%{5}%=%t7%e%p1%{6}%=%t5%e%p1%{7}%=%t0%e%p1%{8}%=%t8%e%p1%{9}%=%t9%e%p1%{10}%=%t\072%e%p1%{11}%=%t;%e%p1%{12}%=%t<%e%p1%{13}%=%t=%e%p1%{14}%=%t6%e?:\
+	:is=\Ev\Eq\Ee\Eb1\Ec0:r2=\Ev\Eq\Ee\Eb1\Ec0:\
+	:tc=st52:
+st52|st52-m|at|at-m|atari|atari-m|atari_st|atarist-m|Atari ST:\
+	:NP:am:eo:mi:\
+	:co#80:it#8:li#24:\
+	:#4=\Ed:%1=\EH:%i=\Ec:&8=\EK:F1=\Ep:F2=\Eq:F3=\Er:F4=\Es:\
+	:F5=\Et:F6=\Eu:F7=\Ev:F8=\Ew:F9=\Ex:FA=\Ey:al=\EL:bl=^G:\
+	:cb=\Eo:cd=\EJ:ce=\EK:cl=\EE:cm=\EY%+ %+ :cr=^M:dl=\EM:\
+	:do=\EB:ho=\EH:is=\Ev\Eq\Ee:k1=\EP:k2=\EQ:k3=\ER:k4=\ES:\
+	:k5=\ET:k6=\EU:k7=\EV:k8=\EW:k9=\EX:k;=\EY:kD=\177:kI=\EI:\
+	:kN=\Eb:kP=\Ea:kb=^H:kd=\EB:kh=\EE:kl=\ED:kr=\EC:ku=\EA:\
+	:le=\ED:me=\Eq:mr=\Ep:nd=\EC:nw=^M^J:r2=\Ev\Eq\Ee:rc=\Ek:\
+	:sc=\Ej:se=\Eq:sf=^J:so=\Ep:sr=\EI:ta=^I:up=\EA:ve=\Ee:\
+	:vi=\Ef:
+# (untranslatable capabilities removed to fit entry within 1023 bytes)
+tw100|toswin vt100 window mgr:\
+	:eo:mi:ms:xo:\
+	:Co#8:co#80:it#8:li#24:pa#64:vt#3:\
+	:%1=\EH:&8=\EK:DL=\E[%dM:DO=\E[%dB:F1=\Ep:F2=\Eq:F3=\Er:\
+	:F4=\Es:F5=\Et:F6=\Eu:F7=\Ev:F8=\Ew:F9=\Ex:FA=\Ey:IC=\E[%d@:\
+	:LE=\E[%dD:RI=\E[%dC:UP=\E[%dA:\
+	:ac=++\054\054--..00II``aaffgghhjjkkllmmnnooppqqrrssttuuvvwwxxyyzz{{||}}~~:\
+	:ae=^O:al=\EL:as=^N:bl=^G:cb=\E[1K:cd=\E[J:ce=\E[K:\
+	:cl=\E[2J\E[H:cm=\E[%i%d;%dH:cr=^M:cs=\E[%i%d;%dr:\
+	:ct=\E[3g:dc=\Ea:dl=\E[M:do=\EB:ei=\Ei:ho=\E[H:im=\Eh:\
+	:is=\E<\E)0:k1=\EOP:k2=\EOQ:k3=\EOR:k4=\EOS:k5=\EOT:\
+	:k6=\EOU:k7=\EOV:k8=\EOW:k9=\EOX:k;=\EOY:kD=\177:kI=\EI:\
+	:kN=\Eb:kP=\E\Ea:kb=^H:kd=\EOB:ke=\E[?1l\E>:kh=\E\EE:\
+	:kl=\EOD:kr=\EOC:ks=\E[?1h\E=:ku=\EOA:le=^H:ll=\E[24H:\
+	:mb=\E[5m:md=\E[1m:me=\E[m:mh=\E[2m:mr=\E[7m:nd=\EC:nw=\EE:\
+	:oc=\E[30;47m:op=\E[30;47m:\
+	:r1=\E<\E[20l\E[?3;6;9l\E[r\Eq\E(B\017\E)0\E>:rc=\E8:\
+	:sc=\E7:se=\E[m:sf=^J:so=\E[7m:sr=\EM:st=\EH:ta=^I:\
+	:te=\E[?7h:ti=\E[?7l:ue=\E[m:up=\EA:us=\E[4m:ve=\Ee:vi=\Ef:
+# The entries for stv52 and stv52pc probably need a revision.
+stv52|MiNT virtual console:\
+	:am:ms:\
+	:co#80:it#8:li#30:\
+	:%1=\EH:&8=\EK:F1=\Ep:F2=\Eq:F3=\Er:F4=\Es:F5=\Et:F6=\Eu:\
+	:F7=\Ev:F8=\Ew:F9=\Ex:FA=\Ey:al=\EL:bl=^G:cd=\EJ:ce=\EK:\
+	:cl=\EE:cm=\EY%+ %+ :cr=^M:dl=\EM:do=\EB:ho=\EH:k1=\EP:\
+	:k2=\EQ:k3=\ER:k4=\ES:k5=\ET:k6=\EU:k7=\EV:k8=\EW:k9=\EX:\
+	:k;=\EY:kD=\177:kI=\EI:kN=\Eb:kP=\Ea:kb=^H:kd=\EB:kh=\EE:\
+	:kl=\ED:kr=\EC:ku=\EA:le=^H:mb=\Er:md=\EyA:me=\Ez_:mh=\Em:\
+	:mr=\Ep:nd=\EC:nw=2*\r\n:op=\Eb@\EcO:r1=\Ez_\Eb@\EcA:\
+	:se=\Eq:sf=2*\n:so=\Ep:sr=2*\EI:ta=^I:te=\Ev\E. \Ee\Ez_:\
+	:ti=\Ev\Ee\Ez_:ue=\EzH:up=\EA:us=\EyH:ve=\E. \Ee:vi=\Ef:\
+	:vs=\E.":
+stv52pc|MiNT virtual console with PC charset:\
+	:am:ms:\
+	:co#80:it#8:li#30:\
+	:%1=\EH:&8=\EK:F1=\Ep:F2=\Eq:F3=\Er:F4=\Es:F5=\Et:F6=\Eu:\
+	:F7=\Ev:F8=\Ew:F9=\Ex:FA=\Ey:\
+	:ac=+\257\054\256-\136.v0\333I\374`\177a\260f\370g\361h\261j\331k\277l\332m\300n\305o\377p-q\304r-s_t+u+v+w+x\263y\363z\362{\343|\366}\234~\371:\
+	:al=\EL:bl=^G:cd=\EJ:ce=\EK:cl=\EE:cm=\EY%+ %+ :cr=^M:\
+	:dl=\EM:do=\EB:ho=\EH:k1=\EP:k2=\EQ:k3=\ER:k4=\ES:k5=\ET:\
+	:k6=\EU:k7=\EV:k8=\EW:k9=\EX:k;=\EY:kD=\177:kI=\EI:kN=\Eb:\
+	:kP=\Ea:kb=^H:kd=\EB:kh=\EE:kl=\ED:kr=\EC:ku=\EA:le=^H:\
+	:mb=\Er:md=\EyA:me=\Ez_:mh=\Em:mr=\Ep:nd=\EC:nw=2*\r\n:\
+	:r1=\Ez_\Eb@\EcA:se=\Eq:sf=2*\n:so=\Ep:sr=2*\EI:ta=^I:\
+	:te=\Ev\E. \Ee\Ez_:ti=\Ev\Ee\Ez_:ue=\EzH:up=\EA:us=\EyH:\
+	:ve=\E. \Ee:vi=\Ef:vs=\E.":
+
 #### ANSI.SYS/ISO 6429 Capabilities
 #
 # See the end-of-file comment for more on these.
