--- teTeX-1.0/texk/web2c/fmtutil.in.orig	Thu May 27 10:48:59 1999
+++ teTeX-1.0/texk/web2c/fmtutil.in	Fri Jun 25 22:40:20 1999
@@ -40,11 +40,11 @@
 @OMEGA@lambda		omega		language.dat	lambda.ini
 
 # The amstex format:
-# amstex	@TEXBIN@		-		amstex.ini
+amstex		@TEXBIN@		-		amstex.ini
 
 # Variants of plain and amstex which allow to customize hyphenation:
-# bplain	@TEXBIN@		language.dat	bplain.ini
-# bamstex	@TEXBIN@		language.dat	bamstex.ini
+bplain		@TEXBIN@		language.dat	bplain.ini
+bamstex		@TEXBIN@		language.dat	bamstex.ini
 
 # Context formats:
 # cont-de	pdfetex		cont-usr.tex	*cont-de.ini
