--- teTeX-1.0/texmf/dvips/config/config.generic Thu Jan  1 01:00:00 1970
+++ teTeX-1.0/texmf/dvips/config/config.generic Wed Aug 25 18:07:23 1999
@@ -0,0 +1,34 @@
+% This is a config file for dvips 5.85(k) and teTeX. Created on Aug 25 1999
+% by Giuseppe Ghibò <ghibo@linux-mandrake.com>. It is intended to create
+% PostScript output for a 'generic' PostScript printer (resolution
+% indipendent if possible). Type 1 fonts are used if found. It could be used
+% together with -D option in dvips command line without having to create 
+% PK fonts for proper resolution. Of course if a font is not found in 
+% Type 1 format, it is tried in PK, using 'linotzzh' as METAFONT mode.
+
+% Default Memory
+m 4000000
+
+% Default is to save output as file.
+o
+
+% Default resolution.
+D 2540
+
+% Default METAFONT mode
+M linotzzh
+
+% This is for using Type 1 fonts where possibile.
+p +bsr.map
+p +bsr-interpolated.map
+p +hoekwater.map
+p +xypic.map
+p +marvosym.map
+p +omega.map
+p +utopia.map
+p +charter.map
+p +cmcyr.map
+p +pl.map
+
+% Last resort sizes.
+R 600
