--- vim60/src/ui.c.orig	Tue Sep 25 19:25:36 2001
+++ vim60/src/ui.c	Thu Oct  4 11:04:02 2001
@@ -1445,6 +1445,8 @@
  */
 #ifdef VMS
 # define INBUFLEN 10000 /* for proper cut/paste between X windows in ch. mode */
+#elif defined(__MINT__)
+# define INBUFLEN 5  /* MiNT doesn't seem to like a large typeahead */
 #else
 # if defined(FEAT_SUN_WORKSHOP) || defined(FEAT_CLIENTSERVER)
    /*
--- vim60/src/edit.c.orig	Sun Sep 23 21:34:28 2001
+++ vim60/src/edit.c	Thu Oct  4 11:34:14 2001
@@ -4057,6 +4057,7 @@
     dont_scroll = FALSE;		/* allow scrolling here */
 #endif
 
+#ifndef __MINT__    /* don't do this for MiNT, because you otherwise don't see what you type ... */
     if (       !ISSPECIAL(c)
 #ifdef FEAT_MBYTE
 	    && (!has_mbyte || (*mb_char2len)(c) == 1)
@@ -4129,6 +4130,7 @@
 	    AppendToRedobuffLit(buf + i);
     }
     else
+#endif /* MiNT */
     {
 #ifdef FEAT_MBYTE
 	if (has_mbyte && (cc = (*mb_char2len)(c)) > 1)
