--- xchat-1.4.2/src/common/ctcp.c.orig	Thu Feb  3 15:11:24 2000
+++ xchat-1.4.2/src/common/ctcp.c	Sat Dec 23 21:59:34 2000
@@ -71,7 +71,7 @@
          case 't':
             {
                char timestr[64];
-               long tt = time (0);
+               time_t tt = time (0);
                char *p, *t = ctime (&tt);
                strcpy (timestr, t);
                p = strchr (timestr, '\n');
--- xchat-1.4.2/src/common/inbound.c.orig	Thu Mar  9 04:44:16 2000
+++ xchat-1.4.2/src/common/inbound.c	Sat Dec 23 21:59:08 2000
@@ -521,7 +521,7 @@
 static void
 channel_date (struct session *sess, char *tbuf, char *chan, char *timestr)
 {
-   long n = atol (timestr);
+   time_t n = atol (timestr);
    char *p, *tim = ctime (&n);
    p = strchr (tim, '\n');
    if (p)
@@ -532,7 +532,7 @@
 static void
 topic_nametime (struct server *serv, char *tbuf, char *chan, char *nick, char *date)
 {
-   long n = atol (date);
+   time_t n = atol (date);
    char *tim = ctime (&n);
    char *p;
    struct session *sess = find_session_from_channel (chan, serv);
@@ -1135,7 +1135,7 @@
                break;
             case 317:
                {
-                  long n = atol (find_word (pdibuf, 6));
+                  time_t n = atol (find_word (pdibuf, 6));
                   long idle = atol (find_word (pdibuf, 5));
                   char *po, *tim = ctime (&n);
                   sprintf (outbuf, "%02ld:%02ld:%02ld", idle / 3600, (idle / 60) % 60, idle % 60);
--- xchat-1.4.2/src/common/xchat.h.orig	Thu Mar 23 08:03:00 2000
+++ xchat-1.4.2/src/common/xchat.h	Sat Dec 23 22:02:20 2000
@@ -282,9 +282,9 @@
    unsigned long oldack;
    unsigned long pos;
    unsigned long oldpos;
-   unsigned long starttime;
-   unsigned long offertime;
-   unsigned long lasttime;
+   time_t starttime;
+   time_t offertime;
+   time_t lasttime;
    char *file;
    char *destfile;
    char *nick;
