--- xpuzzles-5.5.2/xtriangles/Triangles.c.orig	Mon Aug  2 18:42:02 1999
+++ xpuzzles-5.5.2/xtriangles/Triangles.c	Fri Dec 29 15:34:22 2000
@@ -251,7 +251,7 @@
 #endif /* defined(SYSV) || defined(SVR4) */
 
 static int
-usleep(unsigned int usec)
+my_usleep(unsigned int usec)
 {
 #if (defined (SYSV) || defined(SVR4)) && !defined(__hpux)
 #if defined(HAVE_NANOSLEEP)
@@ -297,7 +297,7 @@
 static void
 Sleep(unsigned int cMilliseconds)
 {
-	(void) usleep(cMilliseconds * 1000);
+	(void) my_usleep(cMilliseconds * 1000);
 }
 
 static void
--- xpuzzles-5.5.2/xskewb/Skewb.c.orig	Mon Aug  2 18:41:44 1999
+++ xpuzzles-5.5.2/xskewb/Skewb.c	Fri Dec 29 15:32:58 2000
@@ -441,7 +441,7 @@
 #endif /* defined(SYSV) || defined(SVR4) */
 
 static int
-usleep(unsigned int usec)
+my_usleep(unsigned int usec)
 {
 #if (defined (SYSV) || defined(SVR4)) && !defined(__hpux)
 #if defined(HAVE_NANOSLEEP)
@@ -487,7 +487,7 @@
 static void
 Sleep(unsigned int cMilliseconds)
 {
-	(void) usleep(cMilliseconds * 1000);
+	(void) my_usleep(cMilliseconds * 1000);
 }
 
 static void
--- xpuzzles-5.5.2/xcubes/Cubes.c.orig	Mon Aug  2 18:42:10 1999
+++ xpuzzles-5.5.2/xcubes/Cubes.c	Fri Dec 29 15:33:56 2000
@@ -234,7 +234,7 @@
 #endif /* defined(SYSV) || defined(SVR4) */
 
 static int
-usleep(unsigned int usec)
+my_usleep(unsigned int usec)
 {
 #if (defined (SYSV) || defined(SVR4)) && !defined(__hpux)
 #if defined(HAVE_NANOSLEEP)
@@ -280,7 +280,7 @@
 static void
 Sleep(unsigned int cMilliseconds)
 {
-	(void) usleep(cMilliseconds * 1000);
+	(void) my_usleep(cMilliseconds * 1000);
 }
 
 static void
--- xpuzzles-5.5.2/xhexagons/Hexagons.c.orig	Mon Aug  2 18:41:54 1999
+++ xpuzzles-5.5.2/xhexagons/Hexagons.c	Fri Dec 29 15:35:00 2000
@@ -282,7 +282,7 @@
 #endif /* defined(SYSV) || defined(SVR4) */
 
 static int
-usleep(unsigned int usec)
+my_usleep(unsigned int usec)
 {
 #if (defined (SYSV) || defined(SVR4)) && !defined(__hpux)
 #if defined(HAVE_NANOSLEEP)
@@ -328,7 +328,7 @@
 static void
 Sleep(unsigned int cMilliseconds)
 {
-	(void) usleep(cMilliseconds * 1000);
+	(void) my_usleep(cMilliseconds * 1000);
 }
 
 static void
--- xpuzzles-5.5.2/xpanex/Panex.c.orig	Mon Aug  2 23:08:40 1999
+++ xpuzzles-5.5.2/xpanex/Panex.c	Fri Dec 29 15:35:24 2000
@@ -228,7 +228,7 @@
 #endif /* defined(SYSV) || defined(SVR4) */
 
 static int
-usleep(unsigned int usec)
+my_usleep(unsigned int usec)
 {
 #if (defined (SYSV) || defined(SVR4)) && !defined(__hpux)
 #if defined(HAVE_NANOSLEEP)
@@ -274,7 +274,7 @@
 static void
 Sleep(unsigned int cMilliseconds)
 {
-	(void) usleep(cMilliseconds * 1000);
+	(void) my_usleep(cMilliseconds * 1000);
 }
 
 static void
