--- XServer-0.12/src/Makefile.orig	Thu Jul 12 15:58:56 2001
+++ XServer-0.12/src/Makefile	Wed Aug 15 12:16:12 2001
@@ -1,6 +1,6 @@
 WARN    = -Wall -Wpointer-arith -Wwrite-strings -Wmissing-prototypes -Wshadow \
           -Werror #-Wcast-qual
-CCOPTS  = -D_TRACE -g -O2 -fstrength-reduce #-DNODEBUG
+CCOPTS  = -D_TRACE -O2 #-DNODEBUG
 CCFLAGS = -funsigned-char -fomit-frame-pointer -m68030 \
           -I. -I../include -I/usr/GEM/include \
           $(CCOPTS) $(WARN)
