--- SDL_ttf-2.0.11/Makefile.am.orig	2019-03-21 07:21:22.640711095 +0100
+++ SDL_ttf-2.0.11/Makefile.am	2019-03-21 07:21:54.660710398 +0100
@@ -1,4 +1,6 @@
 # Makefile.am for the SDL truetype font loading library and viewer
+ACLOCAL_AMFLAGS = -I acinclude
+AUTOMAKE_OPTIONS = foreign
 
 lib_LTLIBRARIES = libSDL_ttf.la
 
--- a/SDL_ttf.pc.in	2023-04-08 14:15:48.119095650 +0200
+++ b/SDL_ttf.pc.in	2023-04-08 14:16:47.318998029 +0200
@@ -7,6 +7,6 @@
 Description: ttf library for Simple DirectMedia Layer with FreeType 2 support
 Version: @VERSION@
 Requires: sdl >= @SDL_VERSION@
-Libs: -L${libdir} -lSDL_ttf
+Libs: -L${libdir} -lSDL_ttf -lfreetype -lpng -lz -lbz2 -lgem -lm
 Cflags: -I${includedir}/SDL
 
