From 4dbd1069a6d8d00d06af4dd9a5e061ef427a1495 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Fri, 9 Mar 2018 03:24:29 +0100
Subject: [PATCH 6/8] Drop the "last-updated" footer in documentation

---
 Documentation/asciidoc.conf | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Documentation/asciidoc.conf b/Documentation/asciidoc.conf
index 2c16c536b..c752ef7ff 100644
--- a/Documentation/asciidoc.conf
+++ b/Documentation/asciidoc.conf
@@ -21,6 +21,8 @@ tilde=&#126;
 apostrophe=&#39;
 backtick=&#96;
 litdd=&#45;&#45;
+# drops the "last-updated" footer, with asciidoc-8.6.9+
+footer-style=none
 
 ifdef::backend-docbook[]
 [linkgit-inlinemacro]
-- 
2.16.2

