--- a/src/psl.c.orig	2022-12-26 13:39:11.000000000 +0100
+++ b/src/psl.c	2023-03-07 07:27:34.072755166 +0100
@@ -1601,10 +1601,19 @@ static int isip(const char *hostname)
 	return (WSAStringToAddressW(wName, AF_INET,  NULL, (struct sockaddr *)&addr,  &size) != SOCKET_ERROR) |
 	       (WSAStringToAddressW(wName, AF_INET6, NULL, (struct sockaddr *)&addr6, &size6) != SOCKET_ERROR);
 #else
+	int ret;
 	struct in_addr addr;
-	struct in6_addr addr6;
 
-	return inet_pton(AF_INET, hostname, &addr) || inet_pton(AF_INET6, hostname, &addr6);
+	if ((ret = inet_pton(AF_INET, hostname, &addr)) != 0)
+		return ret;
+#ifndef __MINT__
+	{
+		struct in6_addr addr6;
+		if ((ret = inet_pton(AF_INET6, hostname, &addr6)) != 0)
+			return ret;
+	}
+#endif
+	return ret;
 #endif
 }
 
