Index: libedit-20210910-3.1/src/sys.h
===================================================================
--- libedit-20210910-3.1.orig/src/sys.h
+++ libedit-20210910-3.1/src/sys.h
@@ -40,6 +40,10 @@
 #ifndef _h_sys
 #define	_h_sys
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #if defined(HAVE_SYS_TYPES_H) && defined(__sun)
 #include <sys/types.h>
 #endif
@@ -83,12 +87,12 @@ typedef void	*ioctl_t;
 
 #ifndef HAVE_STRLCAT
 //#define	strlcat libedit_strlcat
-size_t	strlcat(char *dst, const char *src, size_t size);
+size_t	strlcat(char *dst, const char *src, size_t size) libedit_private;
 #endif
 
 #ifndef HAVE_STRLCPY
 //#define	strlcpy libedit_strlcpy
-size_t	strlcpy(char *dst, const char *src, size_t size);
+size_t	strlcpy(char *dst, const char *src, size_t size) libedit_private;
 #endif
 
 #ifndef HAVE_GETLINE
