diff --git a/gcc/symtab.c b/gcc/symtab.c
index 342cc36f3af..785311fdb6d 100644
--- a/gcc/symtab.c
+++ b/gcc/symtab.c
@@ -2178,7 +2178,7 @@ increase_alignment_1 (symtab_node *n, void *v)
 void
 symtab_node::increase_alignment (unsigned int align)
 {
-  gcc_assert (can_increase_alignment_p () && align < MAX_OFILE_ALIGNMENT);
+  gcc_assert (can_increase_alignment_p () && align <= MAX_OFILE_ALIGNMENT);
   ultimate_alias_target()->call_for_symbol_and_aliases (increase_alignment_1,
 						        (void *)(size_t) align,
 						        true);
