diff -rupN binutils-2.29.1.orig/bfd/aoutx.h binutils-2.29.1/bfd/aoutx.h
--- binutils-2.29.1.orig/bfd/aoutx.h	2017-10-06 17:49:09.916544835 +0200
+++ binutils-2.29.1/bfd/aoutx.h	2017-10-06 17:50:05.104543634 +0200
@@ -1303,6 +1303,8 @@ NAME (aout, set_section_contents) (bfd *
 static bfd_boolean
 aout_get_external_symbols (bfd *abfd)
 {
+  if (bfd_get_flavour(abfd) != bfd_target_aout_flavour)
+    return FALSE;
   if (obj_aout_external_syms (abfd) == NULL)
     {
       bfd_size_type count;
diff -rupN binutils-2.29.1.orig/bfd/archive.c binutils-2.29.1/bfd/archive.c
--- binutils-2.29.1.orig/bfd/archive.c	2017-09-05 12:12:10.000000000 +0200
+++ binutils-2.29.1/bfd/archive.c	2017-10-06 17:50:05.108543634 +0200
@@ -870,7 +870,7 @@ bfd_generic_archive_p (bfd *abfd)
       return NULL;
     }
 
-  if (abfd->target_defaulted && bfd_has_map (abfd))
+  if (bfd_has_map (abfd))
     {
       bfd *first;
 
@@ -891,7 +891,14 @@ bfd_generic_archive_p (bfd *abfd)
 	  first->target_defaulted = FALSE;
 	  if (bfd_check_format (first, bfd_object)
 	      && first->xvec != abfd->xvec)
-	    bfd_set_error (bfd_error_wrong_object_format);
+	    {
+	      bfd_set_error (bfd_error_wrong_object_format);
+	      if (abfd->is_linker_input)
+	        {
+	          bfd_ardata (abfd) = tdata_hold;
+	          return NULL;
+	        }
+	    }
 	  /* And we ought to close `first' here too.  */
 	}
     }
diff -rupN binutils-2.29.1.orig/bfd/compress.c binutils-2.29.1/bfd/compress.c
--- binutils-2.29.1.orig/bfd/compress.c	2017-07-10 11:54:41.000000000 +0200
+++ binutils-2.29.1/bfd/compress.c	2017-10-06 17:50:05.108543634 +0200
@@ -319,7 +319,7 @@ bfd_get_full_section_contents (bfd *abfd
 	return FALSE;
       if (p == NULL)
 	{
-	  p = (bfd_byte *) bfd_malloc (sz);
+	  p = (bfd_byte *) bfd_zmalloc (sz);
 	  if (p == NULL)
 	    return FALSE;
 	  *ptr = p;
diff -rupN binutils-2.29.1.orig/bfd/config.bfd binutils-2.29.1/bfd/config.bfd
--- binutils-2.29.1.orig/bfd/config.bfd	2017-10-06 17:49:09.916544835 +0200
+++ binutils-2.29.1/bfd/config.bfd	2017-10-06 17:50:05.108543634 +0200
@@ -1005,6 +1005,10 @@ case "${targ}" in
     # targ_selvecs=m68kmach3_vec
     # targ_cflags=-DSTAT_FOR_EXEC
     ;;
+  m68*-*-mintelf*)
+    targ_defvec=m68k_elf32_vec
+    targ_selvecs="aout0_be_vec m68k_aout_mintprg_vec"
+    ;;
   m68*-*-mint*)
     targ_defvec=aout0_be_vec
     targ_selvecs=m68k_aout_mintprg_vec
diff -rupN binutils-2.29.1.orig/bfd/elf.c binutils-2.29.1/bfd/elf.c
--- binutils-2.29.1.orig/bfd/elf.c	2017-09-05 12:12:10.000000000 +0200
+++ binutils-2.29.1/bfd/elf.c	2017-10-06 17:50:05.108543634 +0200
@@ -8087,7 +8087,11 @@ _bfd_elf_canonicalize_symtab (bfd *abfd,
   long symcount = bed->s->slurp_symbol_table (abfd, allocation, FALSE);
 
   if (symcount >= 0)
-    bfd_get_symcount (abfd) = symcount;
+    {
+      bfd_get_symcount (abfd) = symcount;
+      /* Cache symbols for the generic linker.  */
+      bfd_get_outsymbols (abfd) = allocation;
+    }
   return symcount;
 }
 
diff -rupN binutils-2.29.1.orig/bfd/reloc.c binutils-2.29.1/bfd/reloc.c
--- binutils-2.29.1.orig/bfd/reloc.c	2017-07-10 11:54:41.000000000 +0200
+++ binutils-2.29.1/bfd/reloc.c	2017-10-06 17:50:05.108543634 +0200
@@ -952,6 +952,17 @@ space consuming.  For each target:
       return bfd_reloc_other;
     }
 
+  extern const bfd_target m68k_aout_mintprg_vec;
+  if (flag == bfd_reloc_ok
+      && input_section->output_section->owner->xvec == &m68k_aout_mintprg_vec
+      && !howto->pc_relative && (howto->size == 2 || howto->size == -2))
+    {
+      bfd_vma tpa_address = input_section->output_section->vma
+	+ input_section->output_offset + reloc_entry->address;
+	
+      bfd_m68kmint_add_tpa_relocation_entry (input_section->output_section->owner, tpa_address);
+    }
+
   return flag;
 }
 
diff -rupN binutils-2.29.1.orig/config.sub binutils-2.29.1/config.sub
--- binutils-2.29.1.orig/config.sub	2017-07-10 11:54:41.000000000 +0200
+++ binutils-2.29.1/config.sub	2017-10-06 17:50:05.108543634 +0200
@@ -1535,6 +1535,9 @@ case $os in
 	-xenix)
 		os=-xenix
 		;;
+	-mintelf*)
+		os=-mintelf
+		;;
 	-*mint | -mint[0-9]* | -*MiNT | -MiNT[0-9]*)
 		os=-mint
 		;;
diff -rupN binutils-2.29.1.orig/gas/config/tc-m68k.h binutils-2.29.1/gas/config/tc-m68k.h
--- binutils-2.29.1.orig/gas/config/tc-m68k.h	2017-07-10 11:54:41.000000000 +0200
+++ binutils-2.29.1/gas/config/tc-m68k.h	2017-10-06 17:50:05.108543634 +0200
@@ -169,7 +169,11 @@ extern struct relax_type md_relax_table[
 #define TARGET_USE_CFIPOP 1
 
 #define DWARF2_DEFAULT_RETURN_COLUMN 24
+#ifdef TE_MINT
+#define DWARF2_CIE_DATA_ALIGNMENT (-2)
+#else
 #define DWARF2_CIE_DATA_ALIGNMENT (-4)
+#endif
 
 #define tc_regname_to_dw2regnum tc_m68k_regname_to_dw2regnum
 extern int tc_m68k_regname_to_dw2regnum (const char *regname);
diff -rupN binutils-2.29.1.orig/gas/configure.tgt binutils-2.29.1/gas/configure.tgt
--- binutils-2.29.1.orig/gas/configure.tgt	2017-10-06 17:49:09.924544835 +0200
+++ binutils-2.29.1/gas/configure.tgt	2017-10-06 17:50:05.108543634 +0200
@@ -325,6 +325,7 @@ case ${generic_target} in
   m68k-*-linux-*)			fmt=elf em=linux ;;
   m68k-*-uclinux*)			fmt=elf em=uclinux ;;
   m68k-*-gnu*)				fmt=elf ;;
+  m68k-*-mintelf*)			fmt=elf em=mint bfd_gas=yes ;;
   m68k-*-mint*)				fmt=aout em=mint bfd_gas=yes ;;
   m68k-*-netbsdelf*)			fmt=elf em=nbsd ;;
   m68k-*-netbsd*)			fmt=aout em=nbsd bfd_gas=yes ;;
diff -rupN binutils-2.29.1.orig/ld/scripttempl/m68kmint.sc binutils-2.29.1/ld/scripttempl/m68kmint.sc
--- binutils-2.29.1.orig/ld/scripttempl/m68kmint.sc	2017-10-06 17:49:09.928544835 +0200
+++ binutils-2.29.1/ld/scripttempl/m68kmint.sc	2017-10-06 17:50:05.108543634 +0200
@@ -19,29 +19,38 @@ SECTIONS
   ${RELOCATING+/* The VMA of the .text section is ${TEXT_START_ADDR} instead of 0
      because the extended MiNT header is just before,
      at the beginning of the TEXT segment.  */}
-  .text ${RELOCATING+${TEXT_START_ADDR}}:
+  .text ${RELOCATING+${TEXT_START_ADDR}}: SUBALIGN(2)
   {
     CREATE_OBJECT_SYMBOLS
-    *(.text)
+    *(.text .text.*)
+    *(.rodata .rodata.*) /* Only present in ELF objects */
+    ${RELOCATING+ *(.ctors)}
+    ${RELOCATING+ *(.dtors)}
     ${CONSTRUCTING+CONSTRUCTORS}
+    ${RELOCATING+etext = .;}
     ${RELOCATING+_etext = .;}
-    ${RELOCATING+__etext = .;}
   }
 
-  .data :
+  .data : SUBALIGN(2)
   {
-    *(.data)
+    *(.data .data.*)
+    ${RELOCATING+edata = .;}
     ${RELOCATING+_edata = .;}
-    ${RELOCATING+__edata = .;}
   }
 
   .bss :
   {
-    ${RELOCATING+__bss_start = .;}
-    *(.bss)
+    ${RELOCATING+_bss_start = .;}
+    *(.bss .bss.*)
     *(COMMON)
+    ${RELOCATING+end = .;}
     ${RELOCATING+_end = .;}
-    ${RELOCATING+__end = .;}
   }
+
+  /* Unfortunately, stabs are not mappable from ELF to a.out.
+     It can probably be fixed with some amount of work.  */
+  /DISCARD/ :
+  { *(.stab) *(.stab*) *(.debug) *(.debug*) *(.comment) *(.gnu.warning.*) }
+
 }
 EOF
