Index: src/version.c
===================================================================
--- a/src/version.c.orig
+++ b/src/version.c
@@ -25,9 +25,6 @@
    making the distdir. */
 const char * gdbm_version = "GDBM version " PACKAGE_VERSION ". "
 "16/05/2016"
-#if defined(__STDC__) && defined(__DATE__) && defined(__TIME__)
-		" (built " __DATE__ " " __TIME__ ")"
-#endif
 ;
 int const gdbm_version_number[3] = {
   GDBM_VERSION_MAJOR,
