diff -rupN binutils-2.29.1.orig/ld/ldfile.c binutils-2.29.1/ld/ldfile.c
--- binutils-2.29.1.orig/ld/ldfile.c	2017-07-10 11:54:42.000000000 +0200
+++ binutils-2.29.1/ld/ldfile.c	2017-10-06 18:18:10.840737296 +0200
@@ -340,7 +340,10 @@ ldfile_open_file_search (const char *arc
     {
       if (entry->flags.sysrooted && IS_ABSOLUTE_PATH (entry->filename))
 	{
-	  char *name = concat (ld_sysroot, entry->filename,
+	  char *name;
+	  if (strncmp(ld_sysroot, entry->filename, strlen(ld_sysroot)) == 0 && ldfile_try_open_bfd (entry->filename, entry))
+	return TRUE;
+	  name = concat (ld_sysroot, entry->filename,
 			       (const char *) NULL);
 	  if (ldfile_try_open_bfd (name, entry))
 	    {
