--- SDL_mixer-1.2.12/configure.in.orig	2019-03-19 18:10:39.659158397 +0100
+++ SDL_mixer-1.2.12/configure.in	2019-03-19 18:11:27.255157361 +0100
@@ -251,32 +251,12 @@
     libmikmod_min=1
     libmikmod_rev=10
     libmikmod_ver="$libmikmod_maj.$libmikmod_min.$libmikmod_rev"
-    AC_PATH_PROG(LIBMIKMOD_CONFIG, libmikmod-config, no, [$PATH])
-    if test "$LIBMIKMOD_CONFIG" != "no" ; then
-        
-        CFLAGS_SAVED="$CFLAGS"
-        LIBS_SAVED="$LIBS"
-        CFLAGS="$CFLAGS -DLIBMIKMOD_MUSIC `$LIBMIKMOD_CONFIG --cflags`"
-        LIBS="$LIBS `$LIBMIKMOD_CONFIG --libs`"
+    AC_MSG_CHECKING([for libmikmod - version >= $libmikmod_ver])
+    PKG_CHECK_MODULES(LIBMIKMOD, [libmikmod >= 3.1.10], have_libmikmod=yes)
+    if test "$have_libmikmod" != "no" ; then
+        CFLAGS="$CFLAGS -DLIBMIKMOD_MUSIC $LIBMIKMOD_CFLAGS"
+        LIBS="$LIBS $LIBMIKMOD_LIBS"
         have_libmikmod=yes
-        AC_MSG_CHECKING([for libmikmod - version >= $libmikmod_ver])
-        AC_TRY_RUN([
-#include "mikmod.h"
-#include "stdio.h"
-
-int main(int argc, char **argv)
-{
-long maj=$libmikmod_maj,min=$libmikmod_min,rev=$libmikmod_rev,ver=MikMod_GetVersion();
-/*printf("(%d.%d.%d) ",ver>>16,(ver>>8)&0xff,ver&0xff);*/
-if(ver>=((maj<<16)|(min<<8)|(rev))) {
-    printf("yes\n");
-    return 0;
-}
-printf("no\n*** libmikmod is older than %d.%d.%d, not using.\n",maj,min,rev);
-return 1;
-}
-], [], have_libmikmod=no; CFLAGS="$CFLAGS_SAVED"; LIBS="$LIBS_SAVED",
-        [echo $ac_n "cross compiling; assumed OK... $ac_c"])
     fi
 
     if test x$have_libmikmod = xyes; then
@@ -295,12 +275,12 @@
                 ;;
         esac
         SOURCES="$SOURCES $srcdir/*_mod.c"
-        EXTRA_CFLAGS="$EXTRA_CFLAGS -DMOD_MUSIC `$LIBMIKMOD_CONFIG --cflags`"
+        EXTRA_CFLAGS="$EXTRA_CFLAGS -DMOD_MUSIC $LIBMIKMOD_CFLAGS"
         if test x$enable_music_mod_shared = xyes && test x$mikmod_lib != x; then
             echo "-- dynamic libmikmod -> $mikmod_lib"
             EXTRA_CFLAGS="$EXTRA_CFLAGS -DMOD_DYNAMIC=\\\"$mikmod_lib\\\""
         else
-            EXTRA_LDFLAGS="$EXTRA_LDFLAGS `$LIBMIKMOD_CONFIG --libs`"
+            EXTRA_LDFLAGS="$EXTRA_LDFLAGS $LIBMIKMOD_LIBS"
         fi
     else
         AC_MSG_WARN([*** Unable to find MikMod library (http://mikmod.raphnet.net/)])
@@ -465,7 +465,7 @@
         fi
     else
         AC_CHECK_HEADER([vorbis/vorbisfile.h], [have_ogg_hdr=yes])
-        AC_CHECK_LIB([vorbisfile], [ov_open_callbacks], [have_ogg_lib=yes])
+        AC_CHECK_LIB([vorbisfile], [ov_open_callbacks], [have_ogg_lib=yes],, [-lvorbis -logg -lm])
         if test x$have_ogg_hdr = xyes -a x$have_ogg_lib = xyes; then
             case "$host" in
                 *-*-darwin*)
@@ -525,7 +525,7 @@
 
     if test x$have_flac_ver = xyes; then
         AC_CHECK_HEADER([FLAC/stream_decoder.h], [have_flac_hdr=yes])
-        AC_CHECK_LIB([FLAC], [FLAC__stream_decoder_new], [have_flac_lib=yes])
+        AC_CHECK_LIB([FLAC], [FLAC__stream_decoder_new], [have_flac_lib=yes],, [-logg -lm])
         if test x$have_flac_hdr = xyes -a x$have_flac_lib = xyes; then    
             case "$host" in
                 *-*-darwin*)
@@ -547,7 +547,7 @@
                 echo "-- dynamic libFLAC -> $flac_lib"
                 EXTRA_CFLAGS="$EXTRA_CFLAGS -DFLAC_DYNAMIC=\\\"$flac_lib\\\""
             else
-                EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lFLAC"
+                EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lFLAC -logg -lm"
             fi
         fi
     else
--- SDL_mixer-1.2.12/Makefile.in.orig	2019-03-20 11:20:30.039016501 +0100
+++ SDL_mixer-1.2.12/Makefile.in	2019-03-20 11:20:59.631015857 +0100
@@ -66,10 +66,10 @@
 	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJECTS) $(VERSION_OBJECTS) $(LDFLAGS) $(EXTRA_LDFLAGS) $(LT_LDFLAGS)
 
 $(objects)/playwave$(EXE): $(objects)/playwave.lo $(objects)/$(TARGET)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playwave.lo $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET)
+	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playwave.lo $(LDFLAGS) $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET)
 
 $(objects)/playmus$(EXE): $(objects)/playmus.lo $(objects)/$(TARGET)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playmus.lo $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET)
+	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playmus.lo $(LDFLAGS) $(SDL_CFLAGS) $(SDL_LIBS) $(objects)/$(TARGET)
 
 install: all install-hdrs install-lib #install-bin
 install-hdrs:
