--- SDL_mixer-1.2.13/music_cmd.c.orig	2019-04-01 15:41:13.903395969 +0200
+++ SDL_mixer-1.2.13/music_cmd.c	2019-04-01 16:10:04.953486210 +0200
@@ -162,7 +162,9 @@
 		    {
 			sigset_t mask;
 			sigemptyset(&mask);
+#if !defined(__AMIGA__) || defined(__ixemul__) /* HAVE_SIGPROCMASK */
 			sigprocmask(SIG_SETMASK, &mask, NULL);
+#endif
 		    }
 
 		    /* Execute the command */
--- SDL_mixer-1.2.13/music.c.orig	2019-04-01 16:11:18.941484600 +0200
+++ SDL_mixer-1.2.13/music.c	2019-04-01 16:12:23.065483204 +0200
@@ -1574,7 +1574,7 @@
 		return 0;
 	}
 
-#if defined(__MINGW32__) || defined(__MINGW64__) || defined(__WATCOMC__)
+#if defined(__MINGW32__) || defined(__MINGW64__) || defined(__WATCOMC__) || defined(__AMIGA__)
 	for (path = strtok(paths, ";"); path; path = strtok(NULL, ";")) {
 #elif defined(_WIN32)
 	for (path = strtok_s(paths, ";", &context); path; path = strtok_s(NULL, ";", &context)) {
