diff -ur libwebp-main/src/dec/tree_dec.c libwebp-Mint/src/dec/tree_dec.c
--- libwebp-main/src/dec/tree_dec.c	2022-07-29 07:10:22.000000000 +0200
+++ libwebp-Mint/src/dec/tree_dec.c	2022-08-01 22:04:41.000000000 +0200
@@ -15,7 +15,7 @@
 #include "src/utils/bit_reader_inl_utils.h"
 
 #if !defined(USE_GENERIC_TREE)
-#if !defined(__arm__) && !defined(_M_ARM) && !defined(__aarch64__)
+#if !defined(__arm__) && !defined(_M_ARM) && !defined(__aarch64__) && !defined(__MINT__)
 // using a table is ~1-2% slower on ARM. Prefer the coded-tree approach then.
 #define USE_GENERIC_TREE 1   // ALTERNATE_CODE
 #else
diff -ur libwebp-main/src/dec/vp8l_dec.c libwebp-Mint/src/dec/vp8l_dec.c
--- libwebp-main/src/dec/vp8l_dec.c	2022-07-29 07:10:22.000000000 +0200
+++ libwebp-Mint/src/dec/vp8l_dec.c	2022-08-01 22:05:43.000000000 +0200
@@ -924,7 +924,7 @@
     switch (dist) {
       case 1:
         pattern = src[0];
-#if defined(__arm__) || defined(_M_ARM)   // arm doesn't like multiply that much
+#if defined(__arm__) || defined(_M_ARM) || defined(__MINT__)  // arm doesn't like multiply that much
         pattern |= pattern << 8;
         pattern |= pattern << 16;
 #elif defined(WEBP_USE_MIPS_DSP_R2)
@@ -939,7 +939,7 @@
 #else
         pattern = ((uint32_t)src[0] << 8) | src[1];
 #endif
-#if defined(__arm__) || defined(_M_ARM)
+#if defined(__arm__) || defined(_M_ARM) || defined(__MINT__)
         pattern |= pattern << 16;
 #elif defined(WEBP_USE_MIPS_DSP_R2)
         __asm__ volatile ("replv.ph %0, %0" : "+r"(pattern));
diff -ur libwebp-main/src/utils/bit_reader_utils.c libwebp-Mint/src/utils/bit_reader_utils.c
--- libwebp-main/src/utils/bit_reader_utils.c	2022-07-29 07:10:22.000000000 +0200
+++ libwebp-Mint/src/utils/bit_reader_utils.c	2022-08-01 22:08:31.000000000 +0200
@@ -123,7 +123,8 @@
 
 #if defined(__arm__) || defined(_M_ARM) || defined(__aarch64__) || \
     defined(__i386__) || defined(_M_IX86) || \
-    defined(__x86_64__) || defined(_M_X64)
+    defined(__x86_64__) || defined(_M_X64) || \
+    defined(__MINT__)
 #define VP8L_USE_FAST_LOAD
 #endif
 
--- a/configure.ac.orig	2023-03-31 17:11:59.912775297 +0200
+++ b/configure.ac	2023-03-31 17:13:04.336703946 +0200
@@ -591,7 +591,7 @@
   LIBCHECK_PROLOGUE([TIFF])
   AC_CHECK_HEADER(tiffio.h,
     AC_CHECK_LIB(tiff, TIFFGetVersion,
-                 [TIFF_LIBS="$TIFF_LIBS -ltiff -llzma -ljpeg -lm"
+                 [TIFF_LIBS="$TIFF_LIBS -ltiff -lzstd -llzma -ljpeg -lm"
                   TIFF_INCLUDES="$TIFF_INCLUDES -DWEBP_HAVE_TIFF"
                   AC_DEFINE(WEBP_HAVE_TIFF, [1],
                             [Set to 1 if TIFF library is installed])
