--- libwebp-1.2.3/configure.ac.orig	2022-08-02 18:53:08.688616683 +0200
+++ libwebp-1.2.3/configure.ac	2022-08-02 18:53:48.452615329 +0200
@@ -524,13 +524,8 @@
                                      @<:@default=auto@:>@]))
 AS_IF([test "x$enable_png" != "xno"], [
   CLEAR_LIBVARS([PNG])
-  AC_PATH_PROGS([LIBPNG_CONFIG],
-                [libpng-config libpng16-config libpng15-config libpng14-config \
-                 libpng12-config])
-  if test -n "$LIBPNG_CONFIG"; then
-    PNG_INCLUDES=`$LIBPNG_CONFIG --cflags`
-    PNG_LIBS="`$LIBPNG_CONFIG --ldflags`"
-  fi
+  PNG_INCLUDES=""
+  PNG_LIBS="-lpng -lz -lm"
 
   WITHLIB_OPTION([png], [PNG])
 
--- libwebp-1.2.3/configure.ac.orig	2022-08-02 18:56:55.240608967 +0200
+++ libwebp-1.2.3/configure.ac	2022-08-02 18:57:52.540607015 +0200
@@ -591,7 +591,7 @@
   LIBCHECK_PROLOGUE([TIFF])
   AC_CHECK_HEADER(tiffio.h,
     AC_CHECK_LIB(tiff, TIFFGetVersion,
-                 [TIFF_LIBS="$TIFF_LIBS -ltiff"
+                 [TIFF_LIBS="$TIFF_LIBS -ltiff -llzma -ljpeg -lm"
                   TIFF_INCLUDES="$TIFF_INCLUDES -DWEBP_HAVE_TIFF"
                   AC_DEFINE(WEBP_HAVE_TIFF, [1],
                             [Set to 1 if TIFF library is installed])
