diff -rup x3270.orig/Common/codepage.c x3270/Common/codepage.c
--- x3270.orig/Common/codepage.c	2023-03-06 00:35:02.000000000 +0100
+++ x3270/Common/codepage.c	2023-05-10 16:55:20.517546644 +0200
@@ -52,7 +52,7 @@
 #include "utils.h"
 
 #include <locale.h>
-#if !defined(_WIN32) /*[*/
+#if !defined(_WIN32) && !defined(__MINT__) /*[*/
 # include <langinfo.h>
 #endif /*]*/
 
@@ -100,7 +100,10 @@ codepage_init(const char *cpname)
     const char *realname;
     bool is_dbcs;
 
-#if !defined(_WIN32) /*[*/
+#if defined(__MINT__)
+    codeset_name = "ATARI";
+#else  
+#if !defined(_WIN32) && !defined(__MINT__) /*[*/
     /* Get all of the locale stuff right. */
     setlocale(LC_ALL, "");
 
@@ -123,6 +126,7 @@ codepage_init(const char *cpname)
 #else /*][*/
     codeset_name = lazyaf("CP%d", appres.local_cp);
 #endif /*]*/
+#endif
     set_codeset(codeset_name, appres.utf8);
 
     if (cpname == NULL) {
diff -rup x3270.orig/c3270/Makefile.obj.in x3270/c3270/Makefile.obj.in
--- x3270.orig/c3270/Makefile.obj.in	2023-03-06 00:35:02.000000000 +0100
+++ x3270/c3270/Makefile.obj.in	2023-05-08 17:11:35.275514636 +0200
@@ -69,13 +69,16 @@ fallbacks.c: mkfb $(FALLBACKS)
 	./mkfb -c -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^)
+	gcc -o $@ $(XCPPFLAGS) $(filter %.c,$^)
 
 keypad.o: compiled_keypad.h
 
 compiled_keypad.h: keypad.full keypad.labels keypad.map keypad.outline keypad.callbacks mkkeypad
 	./mkkeypad -I$(TOP)/Common/c3270 >$@
 
+mkkeypad: mkkeypad.c
+	gcc -o $@ $<
+
 c3270: $(OBJS1) $(DEP3270) $(DEP3270I) $(DEP32XX) $(DEP3270STUBS) $(DEP32XX)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LD3270) $(LD3270I) $(LD32XX) $(LD3270STUBS) $(LIBS)
 
diff -rup x3270.orig/lib/3270/Makefile.obj.in x3270/lib/3270/Makefile.obj.in
--- x3270.orig/lib/3270/Makefile.obj.in	2023-03-06 00:35:02.000000000 +0100
+++ x3270/lib/3270/Makefile.obj.in	2023-05-08 17:07:18.851736875 +0200
@@ -59,6 +59,9 @@ favicon.o: favicon.c
 favicon.c: favicon.ico mkicon
 	./mkicon $< >$@
 
+mkicon: mkicon.c
+	gcc -o $@ $<
+
 clean:
 	$(RM) *.o favicon.c mkicon $(LIB3270)
 
diff -rup x3270.orig/x3270/Makefile.obj.in x3270/x3270/Makefile.obj.in
--- x3270.orig/x3270/Makefile.obj.in	2023-03-06 00:35:02.000000000 +0100
+++ x3270/x3270/Makefile.obj.in	2023-05-08 17:42:03.501953814 +0200
@@ -82,7 +82,7 @@ fallbacks.c: mkfb $(FALLBACKS)
 	./mkfb -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^)
+	gcc -o $@ $(XCPPFLAGS) $(filter %.c,$^)
 
 x3270: $(OBJS1) $(DEP3270) $(DEP3270I) $(DEP32XX) $(DEP3270STUBS)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LD3270) $(LD3270I) $(LD32XX) $(LD3270STUBS) $(LIBS)
--- a/x3270/s3270/Makefile.obj.in	2023-03-06 00:35:02.000000000 +0100
+++ b/s3270/Makefile.obj.in	2023-05-10 18:59:49.343103364 +0200
@@ -69,7 +69,7 @@
 	./mkfb -c -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^)
+	gcc -o $@ $(XCPPFLAGS) $(filter %.c,$^)
 
 s3270: $(OBJS1) $(DEP3270) $(DEP32XX) $(DEP3270STUBS)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LD3270) $(LD32XX) $(LD3270STUBS) $(LIBS)
--- a/x3270.orig/b3270/Makefile.obj.in	2023-03-06 00:35:02.000000000 +0100
+++ b/b3270/Makefile.obj.in	2023-05-10 19:37:03.757699737 +0200
@@ -69,7 +69,7 @@
 	./mkfb -c -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^)
+	gcc -o $@ $(XCPPFLAGS) $(filter %.c,$^)
 
 b3270: $(OBJS1) $(DEP3270) $(DEP3270I) $(DEP32XX) $(DEP3270STUBS)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LD3270) $(LD3270I) $(LD32XX) $(LD3270STUBS) $(LIBS)
