diff --git a/Common/codepage.c b/Common/codepage.c
index e3923ce7..3812ac8a 100644
--- a/Common/codepage.c
+++ b/Common/codepage.c
@@ -52,7 +52,7 @@
 #include "utils.h"
 
 #include <locale.h>
-#if !defined(_WIN32) /*[*/
+#if !defined(_WIN32) && !defined(__MINT__) /*[*/
 # include <langinfo.h>
 #endif /*]*/
 
@@ -100,7 +100,10 @@ codepage_init(const char *cpname)
     const char *realname;
     bool is_dbcs;
 
-#if !defined(_WIN32) /*[*/
+#if defined(__MINT__)
+    codeset_name = "ATARI";
+#else
+#if !defined(_WIN32) && !defined(__MINT__) /*[*/
     /* Get all of the locale stuff right. */
     setlocale(LC_ALL, "");
 
@@ -123,6 +126,7 @@ codepage_init(const char *cpname)
 #else /*][*/
     codeset_name = lazyaf("CP%d", appres.local_cp);
 #endif /*]*/
+#endif
     set_codeset(codeset_name, appres.utf8);
 
     if (cpname == NULL) {
diff --git a/Common/pr3287/codepage.c b/Common/pr3287/codepage.c
index b91eb96f..ca67db25 100644
--- a/Common/pr3287/codepage.c
+++ b/Common/pr3287/codepage.c
@@ -37,7 +37,9 @@
 #include <stdlib.h>
 #if !defined(_WIN32) /*[*/
 # include <locale.h>
+#if !defined(__MINT__)
 # include <langinfo.h>
+#endif
 #endif /*]*/
 
 #if defined(__CYGWIN__) /*[*/
@@ -76,6 +78,10 @@ codepage_init(const char *cpname)
     const char *host_codepage;
     const char *cgcsgid_str;
 
+#if defined(__MINT__)
+	codeset_name = "ATARI";
+	set_codeset(codeset_name, false);
+#else
 #if !defined(_WIN32) /*[*/
     setlocale(LC_ALL, "");
     codeset_name = nl_langinfo(CODESET);
@@ -100,6 +106,7 @@ codepage_init(const char *cpname)
 # endif /*]*/
     set_codeset(codeset_name, false);
 #endif /*]*/
+#endif
 
     if (!set_uni(cpname, LOCAL_CODEPAGE, &host_codepage, &cgcsgid_str, NULL,
 		NULL)) {
diff --git a/b3270/Makefile.obj.in b/b3270/Makefile.obj.in
index fa4d7e79..d8d9efa6 100644
--- a/b3270/Makefile.obj.in
+++ b/b3270/Makefile.obj.in
@@ -69,7 +69,7 @@ fallbacks.c: mkfb $(FALLBACKS)
 	./mkfb -c -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^)
+	gcc -o $@ $(XCPPFLAGS) $(filter %.c,$^)
 
 b3270: $(OBJS1) $(DEP3270) $(DEP3270I) $(DEP32XX) $(DEP3270STUBS)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LD3270) $(LD3270I) $(LD32XX) $(LD3270STUBS) $(LIBS)
diff --git a/c3270/Makefile.obj.in b/c3270/Makefile.obj.in
index f52e0e90..819ad9e7 100644
--- a/c3270/Makefile.obj.in
+++ b/c3270/Makefile.obj.in
@@ -69,13 +69,16 @@ fallbacks.c: mkfb $(FALLBACKS)
 	./mkfb -c -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^)
+	gcc -o $@ $(XCPPFLAGS) $(filter %.c,$^)
 
 keypad.o: compiled_keypad.h
 
 compiled_keypad.h: keypad.full keypad.labels keypad.map keypad.outline keypad.callbacks mkkeypad
 	./mkkeypad -I$(TOP)/Common/c3270 >$@
 
+mkkeypad: mkkeypad.c
+	gcc -o $@ $<
+
 c3270: $(OBJS1) $(DEP3270) $(DEP3270I) $(DEP32XX) $(DEP3270STUBS) $(DEP32XX)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LD3270) $(LD3270I) $(LD32XX) $(LD3270STUBS) $(LIBS)
 
diff --git a/lib/3270/Makefile.obj.in b/lib/3270/Makefile.obj.in
index 65ace7a5..ef2c0ad1 100644
--- a/lib/3270/Makefile.obj.in
+++ b/lib/3270/Makefile.obj.in
@@ -59,6 +59,9 @@ favicon.o: favicon.c
 favicon.c: favicon.ico mkicon
 	./mkicon $< >$@
 
+mkicon: mkicon.c
+	gcc -o $@ $<
+
 clean:
 	$(RM) *.o favicon.c mkicon $(LIB3270)
 
diff --git a/s3270/Makefile.obj.in b/s3270/Makefile.obj.in
index 08d17fdb..2b6b2234 100644
--- a/s3270/Makefile.obj.in
+++ b/s3270/Makefile.obj.in
@@ -69,7 +69,7 @@ fallbacks.c: mkfb $(FALLBACKS)
 	./mkfb -c -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^)
+	gcc -o $@ $(XCPPFLAGS) $(filter %.c,$^)
 
 s3270: $(OBJS1) $(DEP3270) $(DEP32XX) $(DEP3270STUBS)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LD3270) $(LD32XX) $(LD3270STUBS) $(LIBS)
diff --git a/tcl3270/Makefile.obj.in b/tcl3270/Makefile.obj.in
index 52469961..237ad3d0 100644
--- a/tcl3270/Makefile.obj.in
+++ b/tcl3270/Makefile.obj.in
@@ -70,7 +70,7 @@ fallbacks.c: mkfb $(FALLBACKS)
 	./mkfb -c -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^)
+	gcc -o $@ $(XCPPFLAGS) $(filter %.c,$^)
 
 tcl3270: $(OBJS1)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LIBS)
diff --git a/x3270/Makefile.obj.in b/x3270/Makefile.obj.in
index e8d924e8..10ca298b 100644
--- a/x3270/Makefile.obj.in
+++ b/x3270/Makefile.obj.in
@@ -82,7 +82,7 @@ fallbacks.c: mkfb $(FALLBACKS)
 	./mkfb -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(filter %.c,$^)
+	gcc -o $@ $(XCPPFLAGS) $(filter %.c,$^)
 
 x3270: $(OBJS1) $(DEP3270) $(DEP3270I) $(DEP32XX) $(DEP3270STUBS)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LD3270) $(LD3270I) $(LD32XX) $(LD3270STUBS) $(LIBS)
