diff -rup wolfssl-5.7.0.orig/examples/echoserver/echoserver.c wolfssl-5.7.0/examples/echoserver/echoserver.c
--- wolfssl-5.7.0.orig/examples/echoserver/echoserver.c	2024-03-20 22:39:57.000000000 +0100
+++ wolfssl-5.7.0/examples/echoserver/echoserver.c	2024-06-09 15:24:36.910274170 +0200
@@ -67,7 +67,7 @@
 
 static void SignalReady(void* args, word16 port)
 {
-#if defined(NO_MAIN_DRIVER) && defined(WOLFSSL_COND)
+#if defined(NO_MAIN_DRIVER) && defined(WOLFSSL_COND) && !defined(SINGLE_THREADED)
     /* signal ready to tcp_accept */
     func_args* server_args = (func_args*)args;
     tcp_ready* ready = server_args->signal;
diff -rup wolfssl-5.7.0.orig/wolfssl/test.h wolfssl-5.7.0/wolfssl/test.h
--- wolfssl-5.7.0.orig/wolfssl/test.h	2024-03-20 22:39:57.000000000 +0100
+++ wolfssl-5.7.0/wolfssl/test.h	2024-06-09 15:29:05.092398079 +0200
@@ -190,8 +190,10 @@
     #include <sys/time.h>
     #include <sys/socket.h>
     #ifdef HAVE_PTHREAD
+    #ifndef SINGLE_THREADED
         #include <pthread.h>
     #endif
+    #endif
     #include <fcntl.h>
     #ifdef TEST_IPV6
         #include <netdb.h>
