--- a/bzip2.c	2019-04-01 11:07:21.667254478 +0200
+++ b/bzip2.c	2019-04-01 11:08:00.719253629 +0200
@@ -62,6 +62,11 @@
 #define ERROR_IF_NOT_ZERO(i)  { if ((i) != 0)    ioError(); }
 #define ERROR_IF_MINUS_ONE(i) { if ((i) == (-1)) ioError(); }
 
+#ifdef __AMIGA__
+# define ftello ftell
+# define fseeko fseek
+#endif
+
 
 /*---------------------------------------------*/
 /*--
@@ -1132,7 +1132,7 @@
 static 
 void applySavedFileAttrToOutputFile ( IntNative fd )
 {
-#  if BZ_UNIX
+#  if BZ_UNIX && !defined(__AMIGA__)
    IntNative retVal;
 
    retVal = fchmod ( fd, fileMetaInfo.st_mode );
