diff -aurN bzip2-1.0.6/bzip2.c bzip2-1.0.6-mint-20140922/bzip2.c
--- bzip2-1.0.6/bzip2.c	2010-09-11 01:04:53.000000000 +0200
+++ bzip2-1.0.6-mint-20140922/bzip2.c	2014-09-22 22:11:59.920644600 +0200
@@ -124,6 +124,12 @@
            ERROR_IF_MINUS_ONE ( retVal );               \
         } while ( 0 )
 #   endif
+
+#   ifdef __MINT__
+#     undef SET_BINARY_MODE
+#     define SET_BINARY_MODE(fd)                        \
+        __set_binmode(fd, 1)
+#   endif
 #endif /* BZ_UNIX */
 
 
--- bzip2-1.0.6/bzlib.h.orig	2019-04-02 18:00:16.762853570 +0200
+++ bzip2-1.0.6/bzlib.h	2019-04-02 18:01:06.710852483 +0200
@@ -193,7 +193,7 @@
 BZ_EXTERN void BZ_API(BZ2_bzWrite) ( 
       int*    bzerror, 
       BZFILE* b, 
-      void*   buf, 
+      const void*   buf, 
       int     len 
    );
 
@@ -271,7 +271,7 @@
 
 BZ_EXTERN int BZ_API(BZ2_bzwrite) (
       BZFILE* b, 
-      void*   buf, 
+      const void*   buf, 
       int     len 
    );
 
--- bzip2-1.0.6/bzlib.c.orig	2019-04-02 18:03:30.930849345 +0200
+++ bzip2-1.0.6/bzlib.c	2019-04-02 18:03:51.718848892 +0200
@@ -964,7 +964,7 @@
 void BZ_API(BZ2_bzWrite)
              ( int*    bzerror, 
                BZFILE* b, 
-               void*   buf, 
+               const void*   buf, 
                int     len )
 {
    Int32 n, n2, ret;
@@ -1489,7 +1489,7 @@
 
 
 /*---------------------------------------------------*/
-int BZ_API(BZ2_bzwrite) (BZFILE* b, void* buf, int len )
+int BZ_API(BZ2_bzwrite) (BZFILE* b, const void* buf, int len )
 {
    int bzerr;
 
