diff -aurN bzip2-1.0.6/bzip2.c bzip2-1.0.6-mint-20140922/bzip2.c
--- bzip2-1.0.6/bzip2.c	2010-09-11 01:04:53.000000000 +0200
+++ bzip2-1.0.6-mint-20140922/bzip2.c	2014-09-22 22:11:59.920644600 +0200
@@ -121,6 +121,12 @@
            ERROR_IF_MINUS_ONE ( retVal );               \
         } while ( 0 )
 #   endif
+
+#   ifdef __MINT__
+#     undef SET_BINARY_MODE
+#     define SET_BINARY_MODE(fd)                        \
+        __set_binmode(fd, 1)
+#   endif
 #endif /* BZ_UNIX */
 
 
