--- a/xpuzzles.Makefile	1999-08-11 16:00:29.000000000 +0200
+++ b/xpuzzles.Makefile	2023-03-20 18:06:18.629983117 +0100
@@ -55,6 +55,7 @@
 #VER=
 VER=-5.5.2
 PROGRAMS=${PUZZLES}
+XMKMF = xmkmf
 
 # Used by me to zip and write and read from my floppy drive
 UNIXDIR=./net
@@ -110,7 +111,7 @@
 		if [ -d ${PRE}$${i} ]; then\
 			cd ${PRE}$${i};\
 			make -f Makefile.in distclean;\
-			xmkmf;\
+			$(XMKMF);\
 			cd ..;\
 		fi;\
 	done
@@ -335,10 +336,14 @@
 	fi
 
 read :
+install:
 	for i in ${PROGRAMS}; do\
 		if [ -d ${PRE}$${i} ]; then\
 			cd ${PRE}$${i};\
-			more README;\
+			mkdir -p $(DESTDIR)/usr/X11R6/bin;\
+			install $(INSTPGMFLAGS) ${PRE}$${i} $(DESTDIR)/usr/X11R6/bin;\
+			mkdir -p $(DESTDIR)/usr/X11R6/man/man1;\
+			install $(INSTMANFLAGS) ${PRE}$${i}.man $(DESTDIR)/usr/X11R6/man/man1/${PRE}$${i}.1;\
 			cd ..;\
 		fi;\
 	done
