--- a/src/apprentice.c.orig	2023-08-27 19:04:37.541025349 +0200
+++ b/src/apprentice.c	2023-08-27 19:05:06.401002779 +0200
@@ -56,6 +56,7 @@ FILE_RCSID("@(#)$File: apprentice.c,v 1.
 #ifdef HAVE_SYS_BSWAP_H
 #include <sys/bswap.h>
 #endif
+#include <byteswap.h>
 
 
 #define	EATAB {while (isascii(CAST(unsigned char, *l)) && \
--- a/src/cdf.c.orig	2022-09-24 22:56:49.000000000 +0200
+++ b/src/cdf.c	2023-08-27 19:05:06.401002779 +0200
@@ -54,6 +54,7 @@ FILE_RCSID("@(#)$File: cdf.c,v 1.123 202
 #ifdef HAVE_SYS_BSWAP_H
 #include <sys/bswap.h>
 #endif
+#include <byteswap.h>
 
 #ifndef EFTYPE
 #define EFTYPE EINVAL
