diff -rupN automake-1.16.orig/m4/make.m4 automake-1.16/m4/make.m4
--- automake-1.16.orig/m4/make.m4	2018-02-25 15:13:58.000000000 +0100
+++ automake-1.16/m4/make.m4	2018-03-14 00:59:39.555179151 +0100
@@ -30,7 +30,7 @@ for s in GNU BSD; do
       ['0:this is the am__doit target'],
       [AS_CASE([$s],
           [BSD], [am__include='.include' am__quote='"'],
-          [am__include='include' am__quote=''])])
+          [am__include='-include' am__quote=''])])
   if test "$am__include" != "#"; then
     _am_result="yes ($s style)"
     break
