--- a/config.sub	2018-03-18 12:09:14.882284284 +0100
+++ b/config.sub	2023-03-06 09:13:32.723763261 +0100
@@ -232,6 +232,10 @@
 	-psos*)
 		os=-psos
 		;;
+	-mintelf | -mintelf[0-9]*)
+		basic_machine=m68k-atari
+		os=-mintelf
+		;;
 	-mint | -mint[0-9]*)
 		basic_machine=m68k-atari
 		os=-mint
@@ -826,6 +830,10 @@
 	miniframe)
 		basic_machine=m68000-convergent
 		;;
+	*mintelf | -mintelf[0-9]*)
+		basic_machine=m68k-atari
+		os=-mintelf
+		;;
 	*mint | -mint[0-9]* | *MiNT | *MiNT[0-9]*)
 		basic_machine=m68k-atari
 		os=-mint
@@ -1496,6 +1504,9 @@
 	-ose*)
 		os=-ose
 		;;
+	-*mintelf | -mintelf[0-9]*)
+		os=-mintelf
+		;;
 	-*mint | -mint[0-9]* | -*MiNT | -MiNT[0-9]*)
 		os=-mint
 		;;
@@ -1779,6 +1790,9 @@
 			-mpw* | -macos*)
 				vendor=apple
 				;;
+			-*mintelf | -mintelf[0-9]*)
+				vendor=atari
+				;;
 			-*mint | -mint[0-9]* | -*MiNT | -MiNT[0-9]*)
 				vendor=atari
 				;;
