--- a/configure.ac	2019-03-20 09:17:27.649570200 +0100
+++ b/configure.ac	2019-03-20 09:18:18.729569088 +0100
@@ -20,55 +20,6 @@
 AM_PROG_LIBTOOL
 AM_PROG_CC_C_O
 
-dnl Set some options based on environment
-
-cflags_save="$CFLAGS"
-if test -z "$GCC"; then
-        case $host in 
-        *-*-irix*)
-                DEBUG="-g -signed" 
-                CFLAGS="-O2 -w -signed"
-                PROFILE="-p -g3 -O2 -signed"
-		;;
-        sparc-sun-solaris*)
-                DEBUG="-v -g"
-                CFLAGS="-xO4 -fast -w -fsimple -native -xcg92"
-                PROFILE="-v -xpg -g -xO4 -fast -native -fsimple -xcg92 -Dsuncc"
-		;;
-        *)
-                DEBUG="-g"
-                CFLAGS="-O"
-                PROFILE="-g -p" 
-		;;
-        esac
-else
-        case $host in 
-        *-*-linux*)
-                DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -Wall -ffast-math -fsigned-char"
-                PROFILE="-Wall -W -pg -g -O20 -ffast-math -fsigned-char"
-		;;
-        sparc-sun-*)
-                DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -ffast-math -fsigned-char"
-                PROFILE="-pg -g -O20 -fsigned-char" 
-		;;
-	*-*-darwin*)
-		DEBUG="-fno-common -g -Wall -fsigned-char"
-		CFLAGS="-fno-common -O4 -Wall -fsigned-char -ffast-math"
-		PROFILE="-fno-common -O4 -Wall -pg -g -fsigned-char -ffast-math"
-		;;
-        *)
-                DEBUG="-g -Wall -fsigned-char"
-                CFLAGS="-O20 -fsigned-char"
-                PROFILE="-O20 -g -pg -fsigned-char" 
-		;;
-        esac
-fi
-CFLAGS="$CFLAGS $cflags_save"
-DEBUG="$DEBUG $cflags_save"
-PROFILE="$PROFILE $cflags_save"
-
 dnl Checks for programs.
 
 dnl Checks for libraries.
