From beb0f6707101a36110fb1e49f7d4a842bb42520a Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Mon, 15 Apr 2024 09:36:13 +0200
Subject: [PATCH 05/22] python3-3.3.0b1-test-posix_fadvise.patch

---
 Lib/test/test_posix.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/test/test_posix.py b/Lib/test/test_posix.py
index c210331..386ce35 100644
--- a/Lib/test/test_posix.py
+++ b/Lib/test/test_posix.py
@@ -430,7 +430,7 @@ class PosixTester(unittest.TestCase):
     def test_posix_fadvise(self):
         fd = os.open(os_helper.TESTFN, os.O_RDONLY)
         try:
-            posix.posix_fadvise(fd, 0, 0, posix.POSIX_FADV_WILLNEED)
+            posix.posix_fadvise(fd, 0, 0, posix.POSIX_FADV_RANDOM)
         finally:
             os.close(fd)
 
-- 
2.41.0

