From 4500d9b921067207dfc46653c946239e3810db6e Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Mon, 15 Apr 2024 09:22:17 +0200
Subject: [PATCH 02/22] distutils-reproducible-compile

---
 Lib/distutils/util.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/distutils/util.py b/Lib/distutils/util.py
index 2ce5c5b..241ccc8 100644
--- a/Lib/distutils/util.py
+++ b/Lib/distutils/util.py
@@ -436,7 +436,7 @@ byte_compile(files, optimize=%r, force=%r,
     else:
         from py_compile import compile
 
-        for file in py_files:
+        for file in sorted(py_files):
             if file[-3:] != ".py":
                 # This lets us be lazy and not filter filenames in
                 # the "install_lib" command.
-- 
2.41.0

