From 19c6e034ac2806cd6df2e17b2e3af459d4b1945f Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Mon, 15 Apr 2024 10:34:50 +0200
Subject: [PATCH 20/22] mintnosharedmod

---
 Makefile.pre.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 73d2eff..91bae74 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -585,7 +585,7 @@ LIBEXPAT_HEADERS= \
 
 # Default target
 all:		@DEF_MAKE_ALL_RULE@
-build_all:	check-clean-src $(BUILDPYTHON) platform oldsharedmods sharedmods \
+build_all:	check-clean-src $(BUILDPYTHON) platform \
 		gdbhooks Programs/_testembed python-config
 build_wasm: check-clean-src $(BUILDPYTHON) platform oldsharedmods python-config
 
@@ -1776,7 +1776,7 @@ altinstall: commoninstall
 
 commoninstall:  check-clean-src @FRAMEWORKALTINSTALLFIRST@ \
 		altbininstall libinstall inclinstall libainstall \
-		sharedinstall oldsharedinstall altmaninstall @FRAMEWORKALTINSTALLLAST@
+		altmaninstall @FRAMEWORKALTINSTALLLAST@
 
 # Install shared libraries enabled by Setup
 DESTDIRS=	$(exec_prefix) $(LIBDIR) $(BINLIBDEST) $(DESTSHARED)
@@ -2074,7 +2074,7 @@ TESTSUBDIRS=	ctypes/test \
 		unittest/test/testmock
 
 TEST_MODULES=@TEST_MODULES@
-libinstall:	all $(srcdir)/Modules/xxmodule.c
+libinstall:	all $(srcdir)/Modules/xxmodule.c pybuilddir.txt
 	@for i in $(SCRIPTDIR) $(LIBDEST); \
 	do \
 		if test ! -d $(DESTDIR)$$i; then \
-- 
2.41.0

