From 2b0b28b6515ec39a62b7024732159cb21ff5e799 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 04:42:50 +0100
Subject: [PATCH 01/63] Fix bashisms

---
 scripts/check-rpaths | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/check-rpaths b/scripts/check-rpaths
index b94630bb1..0dd75a5b8 100755
--- a/scripts/check-rpaths
+++ b/scripts/check-rpaths
@@ -17,7 +17,7 @@
 
 
 test -z "$QA_SKIP_RPATHS" || {
-    echo $"WARNING: '\$QA_SKIP_RPATHS' is obsoleted by 'QA_RPATHS=[0-7]'" >&2
+    echo "WARNING: '\$QA_SKIP_RPATHS' is obsoleted by 'QA_RPATHS=[0-7]'" >&2
     exit 0
 }
 
-- 
2.16.2

