From b1ecc3d74f337aca396816cc5d1fa0675bf4463e Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:41:56 +0100
Subject: [PATCH 30/63] specfilemacro

---
 build/parseSpec.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/parseSpec.c b/build/parseSpec.c
index d0c42a43e..e0abc727a 100644
--- a/build/parseSpec.c
+++ b/build/parseSpec.c
@@ -762,6 +762,7 @@ static rpmSpec parseSpec(const char *specFile, rpmSpecFlags flags,
     spec = newSpec();
 
     spec->specFile = rpmGetPath(specFile, NULL);
+    addMacro(spec->macros, "_specfile", NULL, spec->specFile, RMIL_SPEC);
     pushOFI(spec, spec->specFile);
     /* If buildRoot not specified, use default %{buildroot} */
     if (buildRoot) {
-- 
2.16.2

