From a3f84977de56e967aa3dc043da079e6c5f280c15 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:43:50 +0100
Subject: [PATCH 33/63] debugsubpkg

---
 scripts/find-debuginfo.sh | 32 +++++++++++++++++++-------------
 1 file changed, 19 insertions(+), 13 deletions(-)

diff --git a/scripts/find-debuginfo.sh b/scripts/find-debuginfo.sh
index 36ee9d7f5..dc4e91026 100755
--- a/scripts/find-debuginfo.sh
+++ b/scripts/find-debuginfo.sh
@@ -552,19 +552,25 @@ if $run_dwz \
   fi
 fi
 
-# For each symlink whose target has a .debug file,
-# make a .debug symlink to that file.
-find "$RPM_BUILD_ROOT" ! -path "${debugdir}/*" -type l -print |
-while read f
-do
-  t=$(readlink -m "$f").debug
-  f=${f#$RPM_BUILD_ROOT}
-  t=${t#$RPM_BUILD_ROOT}
-  if [ -f "$debugdir$t" ]; then
-    echo "symlinked /usr/lib/debug$t to /usr/lib/debug${f}.debug"
-    debug_link "/usr/lib/debug$t" "${f}.debug"
-  fi
-done
+# We used to make a .debug symlink for each symlink whose target
+# has a .debug file to that file.  This is not necessary because
+# the debuglink section contains only the destination of those links.
+# Creating those links anyway results in debuginfo packages for
+# devel packages just because of the .so symlinks in them.
+
+## For each symlink whose target has a .debug file,
+## make a .debug symlink to that file.
+#find "$RPM_BUILD_ROOT" ! -path "${debugdir}/*" -type l -print |
+#while read f
+#do
+#  t=$(readlink -m "$f").debug
+#  f=${f#$RPM_BUILD_ROOT}
+#  t=${t#$RPM_BUILD_ROOT}
+#  if [ -f "$debugdir$t" ]; then
+#    echo "symlinked /usr/lib/debug$t to /usr/lib/debug${f}.debug"
+#    debug_link "/usr/lib/debug$t" "${f}.debug"
+#  fi
+#done
 
 if [ -s "$SOURCEFILE" ]; then
   # See also debugedit invocation. Directories must match up.
-- 
2.16.2

