From bad515c1a2e4c2a6ec382c44cd307e0f102fe2ce Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:58:00 +0100
Subject: [PATCH 50/63] headerchk2

---
 lib/header.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/header.c b/lib/header.c
index fdfe48cde..97d309c29 100644
--- a/lib/header.c
+++ b/lib/header.c
@@ -929,7 +929,8 @@ rpmRC hdrblobImport(hdrblob blob, int fast, Header *hdrp, char **emsg)
 
 	rdlen += REGION_TAG_COUNT;
 
-	if (rdlen != blob->dl)
+	/* should be equality test, but can be off if entries are not perfectly aligned */
+	if (rdlen > blob->dl)
 	    goto errxit;
     }
 
-- 
2.16.2

