From c0c7071cd00e9314c7dc44a12b7b46aeb03a4e3a Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:59:22 +0100
Subject: [PATCH 52/63] weak deps compat

---
 build/parsePreamble.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/parsePreamble.c b/build/parsePreamble.c
index c645d4060..41c29471f 100644
--- a/build/parsePreamble.c
+++ b/build/parsePreamble.c
@@ -343,6 +343,7 @@ static struct tokenBits_s const installScriptBits[] = {
     { "verify",		RPMSENSE_SCRIPT_VERIFY },
     { "pretrans",	RPMSENSE_PRETRANS },
     { "posttrans",	RPMSENSE_POSTTRANS },
+    { "hint",		RPMSENSE_MISSINGOK },
     { NULL, 0 }
 };
 
-- 
2.16.2

