From 03c81945e0f91039c943bc18c882f46c1fce3dd1 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Fri, 16 Mar 2018 01:47:32 +0100
Subject: [PATCH] add coldfire patch

---
 rpmrc.in | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/rpmrc.in b/rpmrc.in
index bc03cee81..775eb3cf0 100644
--- a/rpmrc.in
+++ b/rpmrc.in
@@ -220,6 +220,7 @@ arch_canon:     armv7hl: armv7hl 	12
 arch_canon:     armv7hnl: armv7hnl 	12
 
 arch_canon:	m68kmint: m68kmint	13
+arch_canon:	m5475mint: m5475mint	13
 arch_canon:	atarist: m68kmint	13
 arch_canon:	atariste: m68kmint	13
 arch_canon:	ataritt: m68kmint	13
@@ -483,8 +484,8 @@ arch_compat: m68k: noarch
 arch_compat: atarist: m68kmint noarch
 arch_compat: atariste: m68kmint noarch
 arch_compat: ataritt: m68kmint noarch
-arch_compat: falcon: m68kmint noarch
-arch_compat: atariclone: m68kmint noarch
+arch_compat: falcon: m68kmint noarch m5475mint
+arch_compat: atariclone: m68kmint noarch m5475mint
 arch_compat: milan: m68kmint noarch
 arch_compat: hades: m68kmint noarch
 
@@ -620,8 +621,8 @@ buildarch_compat: parisc: noarch
 buildarch_compat: atarist: m68kmint noarch
 buildarch_compat: atariste: m68kmint noarch
 buildarch_compat: ataritt: m68kmint noarch
-buildarch_compat: falcon: m68kmint noarch
-buildarch_compat: atariclone: m68kmint noarch
+buildarch_compat: falcon: m68kmint noarch m5475mint
+buildarch_compat: atariclone: m68kmint noarch m5475mint
 buildarch_compat: milan: m68kmint noarch
 buildarch_compat: hades: m68kmint noarch
 
-- 
2.16.2

