--- a/scripts/Makefile.am.orig	2017-10-05 10:04:57.571602038 +0000
+++ b/scripts/Makefile.am	2017-12-01 14:29:56.760975726 +0000
@@ -28,6 +28,7 @@ EXTRA_DIST = \
 rpmconfig_SCRIPTS = \
 	brp-compress brp-python-bytecompile brp-java-gcjcompile \
 	brp-strip brp-strip-comment-note brp-python-hardlink \
+	brp-suse \
 	brp-strip-shared brp-strip-static-archive \
 	check-files check-prereqs \
 	check-buildroot check-rpaths check-rpaths-worker \
--- a/scripts/brp-strip-comment-note.orig	2017-08-10 08:08:07.150108692 +0000
+++ b/scripts/brp-strip-comment-note	2017-12-01 14:29:56.761975721 +0000
@@ -16,6 +16,8 @@ esac
 # for already stripped elf files in the build root
 for f in `find "$RPM_BUILD_ROOT" -type f \( -perm -0100 -or -perm -0010 -or -perm -0001 \) -exec file {} \; | \
         grep -v "^${RPM_BUILD_ROOT}/\?usr/lib/debug"  | \
+	grep -v ' shared object,' | \
+	grep -v '/lib/modules/' | \
 	sed -n -e 's/^\(.*\):[ 	]*ELF.*, stripped.*/\1/p'`; do
 	note="-R .note"
 	if $OBJDUMP -h $f | grep '^[ 	]*[0-9]*[ 	]*.note[ 	]' -A 1 | \
--- a/scripts/brp-strip.orig	2017-08-10 08:08:07.150108692 +0000
+++ b/scripts/brp-strip	2017-12-01 14:29:56.761975721 +0000
@@ -15,6 +15,7 @@ esac
 for f in `find "$RPM_BUILD_ROOT" -type f \( -perm -0100 -or -perm -0010 -or -perm -0001 \) -exec file {} \; | \
         grep -v "^${RPM_BUILD_ROOT}/\?usr/lib/debug"  | \
 	grep -v ' shared object,' | \
+	grep -v '/lib/modules/' | \
 	sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p'`; do
 	$STRIP -g "$f" || :
 done
--- a/scripts/brp-suse.orig	2017-12-01 14:29:56.762975717 +0000
+++ b/scripts/brp-suse	2017-12-01 14:29:56.761975721 +0000
@@ -0,0 +1,13 @@
+#! /bin/sh
+
+# If using normal root, avoid changing anything:
+if [ -z "$RPM_BUILD_ROOT" -o "$RPM_BUILD_ROOT" = "/" ]; then
+       exit 0
+fi
+
+for script in /usr/lib/rpm/brp-suse.d/brp*; do
+  if test -x "$script"; then
+    echo "calling $script"
+    $script || exit 1
+  fi
+done
