--- a/tools/debugedit.c.orig	2018-01-22 12:09:07.477955907 +0100
+++ b/tools/debugedit.c	2018-01-22 12:09:22.210197759 +0100
@@ -2192,9 +2192,9 @@
 		      /* Offset (pointing into the line program) moves
 			 from old to new index including the header
 			 size diff. */
-		      r_offset += ((dso->lines.table[lndx].new_idx
-				    - dso->lines.table[lndx].old_idx)
-				   + dso->lines.table[lndx].size_diff);
+		      r_offset += (ssize_t)((dso->lines.table[lndx].new_idx
+					     - dso->lines.table[lndx].old_idx)
+					    + dso->lines.table[lndx].size_diff);
 
 		      if (rtype == SHT_RELA)
 			{
