diff -rupN -x Makefile.in -x aclocal.m4 -x compile -x config.guess -x config.sub -x config.log -x configure -x depcomp -x install-sh -x '*.m4' -x missing -x mkinstalldirs -x Makevars.template -x suse_macros rpm-4.14.1.orig/configure.ac rpm-4.14.1/configure.ac
--- rpm-4.14.1.orig/configure.ac	2018-03-14 23:07:20.170947446 +0100
+++ rpm-4.14.1/configure.ac	2018-03-14 23:33:17.010913568 +0100
@@ -434,7 +434,7 @@ AC_CHECK_HEADER([magic.h], [
       WITH_MAGIC_LIB="-lmagic"
     ],[
       AC_MSG_ERROR([missing required library 'libmagic']) 
-    ])
+    ], [$WITH_ZLIB_LIB])
 ],[
       AC_MSG_ERROR([missing required header magic.h]) 
 ])
