--- a/scripts/python-macro-helper.orig	2018-02-05 10:06:51.837690381 +0000
+++ b/scripts/python-macro-helper	2018-02-05 10:07:22.500594058 +0000
@@ -1,4 +1,3 @@
-#!/usr/bin/python -Es
 
 from distutils.sysconfig import get_python_lib
 from platform import python_version_tuple
--- a/scripts/pythondistdeps.py.orig	2017-10-05 10:04:57.587602034 +0000
+++ b/scripts/pythondistdeps.py	2018-02-05 10:06:37.013736947 +0000
@@ -1,4 +1,3 @@
-#!/usr/bin/python
 # -*- coding: utf-8 -*-
 #
 # Copyright 2010 Per Øyvind Karlsen <proyvind@moondrake.org>
