From bf0b16f6c281db1d261238f2b9c7da6cba05e38e Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 16:17:39 +0100
Subject: [PATCH 68/87] Remove dependance on __errno_location()

---
 misc/fts.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/misc/fts.c b/misc/fts.c
index 5c6f53d40..67ad21dc2 100644
--- a/misc/fts.c
+++ b/misc/fts.c
@@ -32,7 +32,7 @@ static char sccsid[] = "@(#)fts.c	8.6 (Berkeley) 8/14/94";
 #endif /* LIBC_SCCS and not lint */
 
 /* Conditional to set up proper fstat64 implementation */
-#if defined(hpux) || defined(sun)
+#if defined(hpux) || defined(sun) || defined(__MINT__)
 #   define FTS_FSTAT64(_fd, _sbp)   fstat((_fd), (_sbp))
 #else
 #   define FTS_FSTAT64(_fd, _sbp)   fstat64((_fd), (_sbp))
@@ -53,16 +53,14 @@ static char sccsid[] = "@(#)fts.c	8.6 (Berkeley) 8/14/94";
 /* Conditionals for working around non-GNU environments */
 #if defined(hpux)
 #   define        _INCLUDE_POSIX_SOURCE
-#   define __errno_location() 	(&errno)
 #   define dirfd(dirp)		-1
 #   define stat64		stat
 #endif
 #if defined(sun)
-#   define __errno_location()	(&errno)
 #   define dirfd(dirp)		-1
 #endif
-#if defined(__APPLE__)
-#   define __errno_location()	(__error())
+#if defined(__MINT__)
+#   define stat64		stat
 #endif
 
 #include "system.h"
@@ -70,11 +68,12 @@ static char sccsid[] = "@(#)fts.c	8.6 (Berkeley) 8/14/94";
 #include <string.h>
 #include <errno.h>
 #include "misc/rpmfts.h"
-#   define __set_errno(val) (*__errno_location ()) = (val)
 #   define __open	open
 #   define __close	close
 #   define __fchdir	fchdir
 #endif
+#undef __set_errno
+#define __set_errno(val) ((errno) = (val))
 
 
 /* Largest alignment size needed, minus one.
-- 
2.24.0

