From 5c0fe5447e0bedbeb2af264fcfa3c79d218cac66 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:55:58 +0100
Subject: [PATCH 48/63] Disable dependency tracking for build

---
 macros.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/macros.in b/macros.in
index 24afbe446..3d3208e88 100644
--- a/macros.in
+++ b/macros.in
@@ -1039,7 +1039,8 @@ package or when debugging this package.\
 	--localstatedir=%{_localstatedir} \\\
 	--sharedstatedir=%{_sharedstatedir} \\\
 	--mandir=%{_mandir} \\\
-	--infodir=%{_infodir}
+	--infodir=%{_infodir} \\\
+	--disable-dependency-tracking
 
 #------------------------------------------------------------------------------
 # Tested features of make
-- 
2.16.2

