From 99bf2af579844b47392ae741a2dce36ddce31671 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 17:02:01 +0100
Subject: [PATCH 74/81] Add missing include of rpmug.h

---
 lib/rpmchroot.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/rpmchroot.c b/lib/rpmchroot.c
index 2513e4f69..9b9ed057b 100644
--- a/lib/rpmchroot.c
+++ b/lib/rpmchroot.c
@@ -4,6 +4,7 @@
 #include <rpm/rpmlog.h>
 #include "lib/rpmchroot.h"
 #include "debug.h"
+#include "rpmug.h"
 
 struct rootState_s {
     char *rootDir;
-- 
2.16.2

