From 05994495160eac65ff9369b94e32e3200b874922 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:58:00 +0100
Subject: [PATCH 48/78] headerchk2

---
 lib/header.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/header.c b/lib/header.c
index 9ec7ed0e4..24e2af451 100644
--- a/lib/header.c
+++ b/lib/header.c
@@ -952,7 +952,8 @@ rpmRC hdrblobImport(hdrblob blob, int fast, Header *hdrp, char **emsg)
 
 	rdlen += REGION_TAG_COUNT;
 
-	if (rdlen != blob->dl)
+	/* should be equality test, but can be off if entries are not perfectly aligned */
+	if (rdlen > blob->dl)
 	    goto errxit;
     }
 
-- 
2.24.0

