From 1dbd9bb273030a327fb1af291276e2aa34e5e1f7 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:48:10 +0100
Subject: [PATCH 37/78] no prereq deprec

---
 build/parsePreamble.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/parsePreamble.c b/build/parsePreamble.c
index c4799e9e5..526222e14 100644
--- a/build/parsePreamble.c
+++ b/build/parsePreamble.c
@@ -1033,7 +1033,7 @@ static struct PreambleRec_s const preambleList[] = {
     {RPMTAG_SUGGESTNAME,	0, 0, 0, LEN_AND_STR("suggests")},
     {RPMTAG_SUPPLEMENTNAME,	0, 0, 0, LEN_AND_STR("supplements")},
     {RPMTAG_ENHANCENAME,	0, 0, 0, LEN_AND_STR("enhances")},
-    {RPMTAG_PREREQ,		2, 1, 0, LEN_AND_STR("prereq")},
+    {RPMTAG_PREREQ,		2, 0, 0, LEN_AND_STR("prereq")},
     {RPMTAG_CONFLICTNAME,	0, 0, 0, LEN_AND_STR("conflicts")},
     {RPMTAG_OBSOLETENAME,	0, 0, 0, LEN_AND_STR("obsoletes")},
     {RPMTAG_PREFIXES,		0, 0, 1, LEN_AND_STR("prefixes")},
@@ -1042,7 +1042,7 @@ static struct PreambleRec_s const preambleList[] = {
     {RPMTAG_BUILDARCHS,		0, 0, 0, LEN_AND_STR("buildarchitectures")},
     {RPMTAG_BUILDARCHS,		0, 0, 0, LEN_AND_STR("buildarch")},
     {RPMTAG_BUILDCONFLICTS,	0, 0, 0, LEN_AND_STR("buildconflicts")},
-    {RPMTAG_BUILDPREREQ,	0, 1, 0, LEN_AND_STR("buildprereq")},
+    {RPMTAG_BUILDPREREQ,	0, 0, 0, LEN_AND_STR("buildprereq")},
     {RPMTAG_BUILDREQUIRES,	0, 0, 0, LEN_AND_STR("buildrequires")},
     {RPMTAG_AUTOREQPROV,	0, 0, 0, LEN_AND_STR("autoreqprov")},
     {RPMTAG_AUTOREQ,		0, 0, 0, LEN_AND_STR("autoreq")},
-- 
2.24.0

