From 87ceec8996afa5ebc6693f4cb09f4cd121ad6a2f Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 17:02:01 +0100
Subject: [PATCH 66/78] Add missing include of rpmug.h

---
 lib/rpmchroot.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/rpmchroot.c b/lib/rpmchroot.c
index 264b35ea0..59387e625 100644
--- a/lib/rpmchroot.c
+++ b/lib/rpmchroot.c
@@ -5,6 +5,7 @@
 #include <rpm/rpmlog.h>
 #include "lib/rpmchroot.h"
 #include "debug.h"
+#include "rpmug.h"
 
 int _rpm_nouserns = 0;
 
-- 
2.24.0

