From a6737fb7dc55e7f589d808af1eacf54ae71cb521 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:37:03 +0100
Subject: [PATCH 25/78] rpm short changelog

---
 build/pack.c           | 66 ++++++++++++++++++++++++++++++++++++++++++
 build/parseChangelog.c |  5 ++++
 2 files changed, 71 insertions(+)

diff --git a/build/pack.c b/build/pack.c
index e6cec1816..5d49c524d 100644
--- a/build/pack.c
+++ b/build/pack.c
@@ -743,11 +743,77 @@ static rpmRC packageBinary(rpmSpec spec, Package pkg, const char *cookie, int ch
     return rc;
 }
 
+static void trimChangelog(Header h)
+{
+    static int oneshot;
+    static int cuttime, minnum, maxnum;
+    rpm_count_t i, keep;
+    struct rpmtd_s timestd, namestd, textstd;
+    rpm_count_t count;
+
+    if (!oneshot) {
+	char *binarychangelogtrim = rpmExpand("%{?_binarychangelogtrim}", NULL);
+	oneshot = 1;
+	if (binarychangelogtrim && *binarychangelogtrim) {
+	    maxnum = atoi(binarychangelogtrim);
+	    binarychangelogtrim = strchr(binarychangelogtrim, ',');
+	    if (binarychangelogtrim)
+	      binarychangelogtrim++;
+	}
+	if (binarychangelogtrim && *binarychangelogtrim) {
+	    cuttime = atoi(binarychangelogtrim);
+	    binarychangelogtrim = strchr(binarychangelogtrim, ',');
+	    if (binarychangelogtrim)
+	      binarychangelogtrim++;
+	}
+	if (binarychangelogtrim && *binarychangelogtrim) {
+	    minnum = atoi(binarychangelogtrim);
+	    binarychangelogtrim = strchr(binarychangelogtrim, ',');
+	}
+    }
+    if (!cuttime && !minnum && !maxnum) {
+	return;
+    }
+
+    if (!headerGet(h, RPMTAG_CHANGELOGTIME, &timestd, HEADERGET_MINMEM))
+	return;
+    count = rpmtdCount(&timestd);
+    if ((!cuttime || count <= minnum) && (!maxnum || count <= maxnum)) {
+	rpmtdFreeData(&timestd);
+	return;
+    }
+    keep = count;
+    if (maxnum && keep > maxnum)
+	keep = maxnum;
+    if (cuttime) {
+	for (i = 0; i < keep; i++) {
+	    uint32_t *tp = rpmtdNextUint32(&timestd);
+	    if (i >= minnum && tp && *tp < cuttime)
+		break;
+	}
+	keep = i;
+    }
+    if (keep >= count) {
+	rpmtdFreeData(&timestd);
+	return;
+    }
+    headerGet(h, RPMTAG_CHANGELOGNAME, &namestd, HEADERGET_MINMEM);
+    headerGet(h, RPMTAG_CHANGELOGTEXT, &textstd, HEADERGET_MINMEM);
+    timestd.count = namestd.count = textstd.count = keep;
+    headerMod(h, &timestd);
+    headerMod(h, &namestd);
+    headerMod(h, &textstd);
+    rpmtdFreeData(&textstd);
+    rpmtdFreeData(&namestd);
+    rpmtdFreeData(&timestd);
+}
+
 rpmRC packageBinaries(rpmSpec spec, const char *cookie, int cheating)
 {
     rpmRC rc = RPMRC_OK;
     Package pkg;
 
+    trimChangelog(spec->packages->header);
     /* Run binary creation in parallel */
     #pragma omp parallel
     #pragma omp single
diff --git a/build/parseChangelog.c b/build/parseChangelog.c
index ad6d834a1..121ff991b 100644
--- a/build/parseChangelog.c
+++ b/build/parseChangelog.c
@@ -253,6 +253,11 @@ static rpmRC addChangelog(Header h, ARGV_const_t sb)
 	    goto exit;
 	}
 
+        /* workaround old suse oddity */
+        if (*s == '-' && s[1] == ' ') {
+            s += 2;
+        }
+
 	/* name */
 	name = s;
 	while (*s != '\0') s++;
-- 
2.24.0

