From a142c593793c28f30b4171d2217ebf673e97d271 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:59:22 +0100
Subject: [PATCH 50/78] weak deps compat

---
 build/parsePreamble.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/parsePreamble.c b/build/parsePreamble.c
index 526222e14..722c38edd 100644
--- a/build/parsePreamble.c
+++ b/build/parsePreamble.c
@@ -371,6 +371,7 @@ static struct tokenBits_s const installScriptBits[] = {
     { "verify",		RPMSENSE_SCRIPT_VERIFY },
     { "pretrans",	RPMSENSE_PRETRANS },
     { "posttrans",	RPMSENSE_POSTTRANS },
+    { "hint",		RPMSENSE_MISSINGOK },
     { NULL, 0 }
 };
 
-- 
2.24.0

