From 6d0484cee38ba1ae1804d8797d9d9b9b01757d81 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 17:03:53 +0100
Subject: [PATCH 68/78] Add missing include of signal.h

---
 lib/rpmscript.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/rpmscript.c b/lib/rpmscript.c
index 8d5b531f6..db400eab6 100644
--- a/lib/rpmscript.c
+++ b/lib/rpmscript.c
@@ -3,6 +3,8 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <errno.h>
+#include <unistd.h>
+#include <signal.h>
 
 #include <rpm/rpmfileutil.h>
 #include <rpm/rpmmacro.h>
-- 
2.24.0

