From 3d108c742c0e7043afc67df1a606489f087c7838 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 15:32:50 +0100
Subject: [PATCH 66/87] Check wether mkdtemp() is available

---
 configure.ac     | 1 +
 sign/rpmgensig.c | 8 +++++++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 542d93ab5..c802acec2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -771,6 +771,7 @@ AC_CHECK_FUNCS(mergesort)
 AC_CHECK_FUNCS(getauxval)
 AC_CHECK_FUNCS(setprogname, [], [], [#include <stdlib.h>])
 AC_CHECK_FUNCS(nanosleep)
+AC_CHECK_FUNCS(mkdtemp)
 
 AC_MSG_CHECKING([whether __progname is defined])
 AC_LINK_IFELSE([AC_LANG_PROGRAM([extern const char *__progname;],
diff --git a/sign/rpmgensig.c b/sign/rpmgensig.c
index d29c17837..fff298762 100644
--- a/sign/rpmgensig.c
+++ b/sign/rpmgensig.c
@@ -48,7 +48,13 @@ static char *mkTempFifo(void)
 
     tmpdir = rpmGetPath(tmppath, "/rpm-tmp.XXXXXX", NULL);
     mode = umask(0077);
+#ifdef HAVE_MKDTEMP
     tmpdir = mkdtemp(tmpdir);
+#else
+    tmpdir = mktemp(tmpdir);
+    if (tmpdir && mkdir(tmpdir, 0700) != 0)
+       tmpdir = _free(tmpdir);
+#endif
     umask(mode);
     if (tmpdir == NULL) {
 	rpmlog(RPMLOG_ERR, _("error creating temp directory %s: %m\n"),
@@ -65,7 +71,7 @@ static char *mkTempFifo(void)
 
 exit:
     if (fifofn == NULL && tmpdir != NULL)
-	unlink(tmpdir);
+	rmdir(tmpdir);
 
     free(tmppath);
     free(tmpdir);
-- 
2.24.0

