From 2104e30fdb06d98bfc803b0af5a8270250810cd2 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 17:03:14 +0100
Subject: [PATCH 67/78] Add detection of default machine m68k

---
 lib/rpmrc.c | 25 ++++++++++++++++++++++++-
 1 file changed, 24 insertions(+), 1 deletion(-)

diff --git a/lib/rpmrc.c b/lib/rpmrc.c
index 52c161588..a1055e640 100644
--- a/lib/rpmrc.c
+++ b/lib/rpmrc.c
@@ -1050,8 +1050,10 @@ static void defaultMachine(rpmrcCtx ctx, const char ** arch, const char ** os)
 
 #if !defined(__linux__)
 	if (rstreq(un.sysname, "AIX")) {
+		char buf[sizeof(un.version) + sizeof(un.release) + 5];
 	    strcpy(un.machine, __power_pc() ? "ppc" : "rs6000");
-	    sprintf(un.sysname,"aix%s.%s", un.version, un.release);
+	    sprintf(buf, "aix%s.%s", un.version, un.release);
+	    strncpy(un.sysname, buf, sizeof(un.sysname));
 	}
 	else if (rstreq(un.sysname, "Darwin")) { 
 #if defined(__ppc__)
@@ -1060,10 +1062,31 @@ static void defaultMachine(rpmrcCtx ctx, const char ** arch, const char ** os)
 	    strcpy(un.machine, "i386");
 #elif defined(__x86_64__)
 	    strcpy(un.machine, "x86_64");
+#elif defined(__m68k__)
+	    strcpy(un.machine, "m68k");
 #else
 	    #warning "No architecture defined! Automatic detection may not work!"
 #endif 
 	}
+#ifdef __MINT__
+	else if (rstreq(un.sysname, "FreeMiNT") || rstreq(un.sysname, "MiNT")) {
+#if defined(__mcoldfire__)
+	    strcpy(un.machine, "coldfire");
+#elif defined(__m68020__)
+	    strcpy(un.machine, "m68020");
+#elif defined(__m68k__)
+	    strcpy(un.machine, "m68k");
+#else
+	    #warning "No architecture defined! Automatic detection may not work!"
+#endif
+		/*
+		 * seems to be broken currently in the interface between
+		 * aranym, mintlib, and mint kernel
+		 */
+		if (strcmp(un.machine, "aranym") == 0)
+			strcpy(un.machine, "m68020");
+	}
+#endif
 	else if (rstreq(un.sysname, "SunOS")) {
 	    /* Solaris 2.x: n.x.x becomes n-3.x.x */
 	    sprintf(un.sysname, "solaris%1d%s", atoi(un.release)-3,
-- 
2.24.0

