From d19b72b0a9c184c02a9c69896fe4b6d4d6081c7b Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 06:05:30 +0100
Subject: [PATCH 62/63] auto config update aarch64 ppc64le

---
 build/parseBuildInstallClean.c | 56 +++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 55 insertions(+), 1 deletion(-)

diff --git a/build/parseBuildInstallClean.c b/build/parseBuildInstallClean.c
index 6386c2365..7a8d8b141 100644
--- a/build/parseBuildInstallClean.c
+++ b/build/parseBuildInstallClean.c
@@ -8,6 +8,39 @@
 #include "build/rpmbuild_internal.h"
 #include "debug.h"
 
+#define SKIPSPACE(s) { while (*(s) && risspace(*(s))) (s)++; }
+#define SKIPNONSPACE(s) { while (*(s) && !risspace(*(s))) (s)++; }
+
+/**
+ */
+static int matchTok(const char *token, const char *line)
+{
+    const char *b, *be = line;
+    size_t toklen = strlen(token);
+    int rc = 0;
+
+    while ( *(b = be) != '\0' ) {
+	SKIPSPACE(b);
+	be = b;
+	SKIPNONSPACE(be);
+	if (be == b)
+	    break;
+	if (toklen != (be-b) || rstrncasecmp(token, b, (be-b)))
+	    continue;
+	rc = 1;
+	break;
+    }
+
+    return rc;
+}
+
+#define ARGMATCH(s,token,match) \
+do { \
+    char *exp = rpmExpand(token, NULL); \
+    match = matchTok(exp, s); \
+    free(exp); \
+} while (0)
+
 
 int parseBuildInstallClean(rpmSpec spec, int parsePart)
 {
@@ -46,7 +79,28 @@ int parseBuildInstallClean(rpmSpec spec, int parsePart)
     } else if (rc < 0) {
 	goto exit;
     }
-    
+
+    if (parsePart == PART_BUILD) {
+        int match_aarch64, match_ppc64le, match_riscv64;
+        ARGMATCH("aarch64", "%{_target_cpu}", match_aarch64);
+        ARGMATCH("ppc64le", "%{_target_cpu}", match_ppc64le);
+        ARGMATCH("riscv64", "%{_target_cpu}", match_riscv64);
+        if (match_aarch64 || match_ppc64le || match_riscv64) {
+        static char const buf[] =
+            "ref=/usr/lib/rpm\n"
+            "for s in guess sub; do\n"
+            "    for c in $(find -maxdepth 8 -name \"config.$s\"); do\n"
+            "         grep -q config-patches@ $c || continue\n"
+            "         grep -q aarch64 $c || install -m 755 $ref/config.$s $c\n"
+            "         grep -q ppc64le $c || install -m 755 $ref/config.$s $c\n"
+            "         grep -q 'riscv64[-:]' $c || install -m 755 $ref/config.$s $c\n"
+            "     done\n"
+            "done\n"
+        ;
+        appendLineStringBuf(*sbp, buf);
+        }
+    }
+
     while (! (nextPart = isPart(spec->line))) {
 	appendStringBuf(*sbp, spec->line);
 	if ((rc = readLine(spec, STRIP_NOTHING)) > 0) {
-- 
2.16.2

