From 37fb766b65d0da6faf8850ff15f482f2fb3ee85d Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:43:04 +0100
Subject: [PATCH 31/78] Hmm, SUSE doesn't use it, so what's the purpose of this
 patch?

---
 macros.in | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/macros.in b/macros.in
index 20ea15931..30fde507c 100644
--- a/macros.in
+++ b/macros.in
@@ -321,6 +321,16 @@ package or when debugging this package.\
 #
 #%bugurl
 
+#	Configurable distribution tag, same as Disttag: tag in a specfile.
+#	The tag will be used to supply reliable information to tools like
+#	rpmfind.
+#
+# Note: You should not configure with disturl (or build packages with
+# the Disttag: tag) unless you are willing to supply content in a
+# yet-to-be-determined format at the tag specified.
+#
+#%disttag
+
 #	Boolean (i.e. 1 == "yes", 0 == "no") that controls whether files
 #	marked as %doc should be installed.
 #%_excludedocs
-- 
2.24.0

