From 8362d51e019a276b8f9a36c27bbc5e60819dfa18 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:39:26 +0100
Subject: [PATCH 29/87] Add firmware files in /lib/firmware into RPM provides

---
 firmware.prov | 8 ++++++++
 1 file changed, 8 insertions(+)
 create mode 100644 firmware.prov

diff --git a/firmware.prov b/firmware.prov
new file mode 100644
index 000000000..12f8336eb
--- /dev/null
+++ b/firmware.prov
@@ -0,0 +1,8 @@
+#!/bin/sh
+# Add firmware files in /lib/firmware into RPM provides
+
+while read instfile ; do
+  case $instfile in
+    */lib/firmware/*) test -f "$instfile" && echo "firmware(${instfile##*/lib/firmware/})" ;;
+  esac
+done
-- 
2.24.0

