From 84874ba20aa3042200ebbfcacf9e8514dd2fe542 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:33:37 +0100
Subject: [PATCH 19/87] Adjust default values in platform.in

---
 platform.in | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/platform.in b/platform.in
index c12dfc759..66b07fbac 100644
--- a/platform.in
+++ b/platform.in
@@ -41,7 +41,7 @@
 %_oldincludedir		@oldincludedir@
 %_infodir		@infodir@
 %_mandir		@mandir@
-%_initddir		%{_sysconfdir}/rc.d/init.d
+%_initddir		%{_sysconfdir}/init.d
 # Deprecated misspelling, present for backwards compatibility.
 %_initrddir		%{_initddir}
 %_rundir		@RUNDIR@
@@ -83,3 +83,21 @@
 @apple@#
 @apple@%_use_internal_dependency_generator     0
 @apple@%__so	dylib
+@suse@#---------------------------------------------------------------------
+@suse@#        Expanded at end of %prep
+@suse@#
+@suse@%__id_u          %{__id} -u
+@suse@%__chown_Rhf     %{__chown} -Rhf
+@suse@%__chgrp_Rhf     %{__chgrp} -Rhf
+@suse@%_fixowner       [ `%{__id_u}` = '0' ] && %{__chown_Rhf} root
+@suse@%_fixgroup       [ `%{__id_u}` = '0' ] && %{__chgrp_Rhf} root
+@suse@%_fixperms       %{__chmod} -Rf @FIXPERMS@
+@suse@
+@suse@#---------------------------------------------------------------------
+@suse@#        Expanded at start of %build
+@suse@#
+@suse@%__spec_build_pre %{___build_pre}\
+@suse@%{?buildroot:  %__rm -rf "$RPM_BUILD_ROOT"\
+@suse@  %__mkdir_p `dirname "$RPM_BUILD_ROOT"`\
+@suse@  %__mkdir "$RPM_BUILD_ROOT"\
+@suse@}
-- 
2.24.0

