From c15f8e6b617a79686aa296f50a09740c9c045df9 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Mon, 31 Aug 2020 14:41:58 +0200
Subject: [PATCH] Omit some tag alias names from table

---
 lib/gentagtbl.sh |  2 +-
 lib/rpmtag.h     | 14 +++++++-------
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/lib/gentagtbl.sh b/lib/gentagtbl.sh
index 0558d09f3..0e8a3db24 100755
--- a/lib/gentagtbl.sh
+++ b/lib/gentagtbl.sh
@@ -4,7 +4,7 @@ cat << EOF
 static const struct headerTagTableEntry_s rpmTagTable[] = {
 EOF
 
-${AWK} '/[\t ](RPMTAG_[A-Z0-9]*)[ \t]+([0-9]*)/ && !/internal/ && !/unimplemented/ {
+${AWK:-awk} '/[\t ](RPMTAG_[A-Z0-9]*)[ \t]+([0-9]*)/ && !/internal/ && !/unimplemented/ && !/alias/ {
 	tt = "NULL"
 	ta = "ANY"
 	ext = "0"
diff --git a/lib/rpmtag.h b/lib/rpmtag.h
index b9623ef24..3ac8dfc0b 100644
--- a/lib/rpmtag.h
+++ b/lib/rpmtag.h
@@ -69,13 +69,13 @@ typedef enum rpmTag_e {
     /* RPMTAG_SIG_BASE+19 reserved for RPMSIGTAG_FILESIGNATURELENGTH */
 
     RPMTAG_NAME  		= 1000,	/* s */
-#define	RPMTAG_N	RPMTAG_NAME	/* s */
+#define	RPMTAG_N	RPMTAG_NAME	/* s alias */
     RPMTAG_VERSION		= 1001,	/* s */
-#define	RPMTAG_V	RPMTAG_VERSION	/* s */
+#define	RPMTAG_V	RPMTAG_VERSION	/* s alias */
     RPMTAG_RELEASE		= 1002,	/* s */
-#define	RPMTAG_R	RPMTAG_RELEASE	/* s */
+#define	RPMTAG_R	RPMTAG_RELEASE	/* s alias */
     RPMTAG_EPOCH   		= 1003,	/* i */
-#define	RPMTAG_E	RPMTAG_EPOCH	/* i */
+#define	RPMTAG_E	RPMTAG_EPOCH	/* i alias */
     RPMTAG_SUMMARY		= 1004,	/* s{} */
     RPMTAG_DESCRIPTION		= 1005,	/* s{} */
     RPMTAG_BUILDTIME		= 1006,	/* i */
@@ -122,7 +122,7 @@ typedef enum rpmTag_e {
     RPMTAG_ARCHIVESIZE		= 1046,	/* i */
     RPMTAG_PROVIDENAME		= 1047,	/* s[] */
 #define	RPMTAG_PROVIDES RPMTAG_PROVIDENAME	/* s[] */
-#define	RPMTAG_P	RPMTAG_PROVIDENAME	/* s[] */
+#define	RPMTAG_P	RPMTAG_PROVIDENAME	/* s[] alias */
     RPMTAG_REQUIREFLAGS		= 1048,	/* i[] */
     RPMTAG_REQUIRENAME		= 1049,	/* s[] */
 #define	RPMTAG_REQUIRES RPMTAG_REQUIRENAME	/* s[] */
@@ -132,7 +132,7 @@ typedef enum rpmTag_e {
     RPMTAG_CONFLICTFLAGS	= 1053, /* i[] */
     RPMTAG_CONFLICTNAME		= 1054,	/* s[] */
 #define	RPMTAG_CONFLICTS RPMTAG_CONFLICTNAME	/* s[] */
-#define	RPMTAG_C	RPMTAG_CONFLICTNAME	/* s[] */
+#define	RPMTAG_C	RPMTAG_CONFLICTNAME	/* s[] alias */
     RPMTAG_CONFLICTVERSION	= 1055,	/* s[] */
     RPMTAG_DEFAULTPREFIX	= 1056, /* s internal - deprecated */
     RPMTAG_BUILDROOT		= 1057, /* s internal - obsolete */
@@ -161,7 +161,7 @@ typedef enum rpmTag_e {
     RPMTAG_BUILDARCHS		= 1089, /* s[] */
     RPMTAG_OBSOLETENAME		= 1090,	/* s[] */
 #define	RPMTAG_OBSOLETES RPMTAG_OBSOLETENAME	/* s[] */
-#define	RPMTAG_O	RPMTAG_OBSOLETENAME	/* s[] */
+#define	RPMTAG_O	RPMTAG_OBSOLETENAME	/* s[] alias */
     RPMTAG_VERIFYSCRIPTPROG	= 1091,	/* s[] */
     RPMTAG_TRIGGERSCRIPTPROG	= 1092,	/* s[] */
     RPMTAG_DOCDIR		= 1093, /* internal */
-- 
2.24.0

