From dec8d6492d5ef8fee9bebd47f5f38d217bd6692b Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:55:58 +0100
Subject: [PATCH 46/78] Disable dependency tracking for build

---
 macros.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/macros.in b/macros.in
index 30fde507c..6679ca0d3 100644
--- a/macros.in
+++ b/macros.in
@@ -1096,7 +1096,8 @@ package or when debugging this package.\
 	--localstatedir=%{_localstatedir} \\\
 	--sharedstatedir=%{_sharedstatedir} \\\
 	--mandir=%{_mandir} \\\
-	--infodir=%{_infodir}
+	--infodir=%{_infodir} \\\
+	--disable-dependency-tracking
 
 #------------------------------------------------------------------------------
 # Tested features of make
-- 
2.24.0

