From 19929931caf9c380c698ffbb468086744dd78ad7 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 17:46:51 +0100
Subject: [PATCH 72/78] Add new entries to .gitignore

---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index 14c260ee7..50fffd2d1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -30,6 +30,7 @@ Makefile.in.in
 /db.h
 /debugedit
 /depcomp
+/compile
 /doc/librpm.doxy
 /doc/librpm/html
 /elfdeps
@@ -77,6 +78,7 @@ Makefile.in.in
 /rpmsign
 /rpmsort
 /rpmspec
+/rpmqpack
 /scripts/macros.perl
 /scripts/macros.php
 /scripts/macros.python
-- 
2.24.0

