From ad7c7f6882e6bcfd464244e33a302209c4ea2a18 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 06:02:07 +0100
Subject: [PATCH 56/63] empty manifest

---
 macros.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/macros.in b/macros.in
index 7ff208f4e..54040030b 100644
--- a/macros.in
+++ b/macros.in
@@ -446,7 +446,7 @@ package or when debugging this package.\
 # Should empty %files manifest file terminate a build?
 #
 # Note: The default value should be 0 for legacy compatibility.
-%_empty_manifest_terminate_build	1
+%_empty_manifest_terminate_build	0
 
 #
 # Should binaries in noarch packages terminate a build?
-- 
2.16.2

