From 2aac37153cec6ae458b061432f6607541fd22269 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Sun, 30 Aug 2020 13:35:32 +0200
Subject: [PATCH 73/78] Fix type of callback functions

---
 .gitignore    | 1 +
 lib/tagexts.c | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/tagexts.c b/lib/tagexts.c
index b7305d025..d0f332410 100644
--- a/lib/tagexts.c
+++ b/lib/tagexts.c
@@ -2,8 +2,8 @@
  * \file lib/formats.c
  */
 
-#include <wchar.h>
 #include "system.h"
+#include <wchar.h>
 
 #include <rpm/rpmtypes.h>
 #include <rpm/rpmlib.h>
@@ -648,10 +648,10 @@ static int filerequireTag(Header h, rpmtd td, headerGetFlags hgflags)
 
 #if defined(ENABLE_NLS)
 extern int _nl_msg_cat_cntr;	/* XXX GNU gettext voodoo */
-#endif
 static const char * const language = "LANGUAGE";
 
 static const char * const _macro_i18ndomains = "%{?_i18ndomains}";
+#endif
 
 /**
  * Retrieve i18n text.
-- 
2.24.0

