From 0b8594621d9c2bba2adea3d1185c420eab6044f0 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 16:42:05 +0100
Subject: [PATCH 64/78] Fix test for -lmagic, which might need zlib

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 7c46a4c73..bb3ee8fb1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -452,7 +452,7 @@ AC_CHECK_HEADER([magic.h], [
       WITH_MAGIC_LIB="-lmagic"
     ],[
       AC_MSG_ERROR([missing required library 'libmagic']) 
-    ])
+    ], [$WITH_ZLIB_LIB])
 ],[
       AC_MSG_ERROR([missing required header magic.h]) 
 ])
-- 
2.24.0

