From 33279ae05d4d02654e18ab5f85e60318902ecffb Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 05:52:09 +0100
Subject: [PATCH 42/78] Disable file coloring for SUSE systems

---
 build/rpmfc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/build/rpmfc.c b/build/rpmfc.c
index 53c6ecce0..b8da3731b 100644
--- a/build/rpmfc.c
+++ b/build/rpmfc.c
@@ -1490,7 +1490,8 @@ rpmRC rpmfcGenerateDepends(const rpmSpec spec, Package pkg)
     /* XXX Make sure only primary (i.e. Elf32/Elf64) colors are added. */
     for (int i = 0; i < fc->nfiles; i++)
 	fc->fcolor[i] &= 0x0f;
-    headerPutUint32(pkg->header, RPMTAG_FILECOLORS, fc->fcolor, fc->nfiles);
+    if (rpmExpandNumeric("%{?_transaction_color}") != 0)
+	headerPutUint32(pkg->header, RPMTAG_FILECOLORS, fc->fcolor, fc->nfiles);
     
     /* Add classes(#classes) */
     for (rpmsid id = 1; id <= rpmstrPoolNumStr(fc->cdict); id++) {
-- 
2.24.0

