From 90f296477594265a5f315a17aa3e0909d9229635 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 04:42:50 +0100
Subject: [PATCH 01/78] Fix bashisms

---
 scripts/check-rpaths | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/check-rpaths b/scripts/check-rpaths
index c3d860050..089e43c00 100755
--- a/scripts/check-rpaths
+++ b/scripts/check-rpaths
@@ -17,7 +17,7 @@
 
 
 test -z "$QA_SKIP_RPATHS" || {
-    echo $"WARNING: '\$QA_SKIP_RPATHS' is obsoleted by 'QA_RPATHS=[0-7]'" >&2
+    echo "WARNING: '\$QA_SKIP_RPATHS' is obsoleted by 'QA_RPATHS=[0-7]'" >&2
     exit 0
 }
 
-- 
2.24.0

