From a9a0197eb80b2b0d04113e4cf140b61e2310b94a Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 15 Mar 2018 16:36:10 +0100
Subject: [PATCH 71/87] Fix tests wether lzma supports multi-threading

---
 configure.ac  |  1 +
 rpmio/rpmio.c | 10 ++++++----
 2 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5337cf0bf..30f2a17e9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -200,6 +200,7 @@ AC_SUBST(WITH_BZ2_LIB)
 
 AC_CHECK_HEADERS([lzma.h],[
   AC_CHECK_LIB(lzma, lzma_easy_encoder, [WITH_LZMA_LIB=-llzma])
+  AC_CHECK_LIB(lzma, lzma_stream_encoder_mt, [AC_DEFINE([HAVE_LZMA_MT], 1, [Define as 1 if your lzma lib has multi-threaded encoder()])])
 ])
 AC_SUBST(WITH_LZMA_LIB)
 
diff --git a/rpmio/rpmio.c b/rpmio/rpmio.c
index c7cbc32aa..038e28a38 100644
--- a/rpmio/rpmio.c
+++ b/rpmio/rpmio.c
@@ -723,10 +723,8 @@ static const FDIO_t bzdio = &bzdio_s ;
 #include <sys/types.h>
 #include <inttypes.h>
 #include <lzma.h>
-/* Multithreading support in stable API since xz 5.2.0 */
-#if LZMA_VERSION >= 50020002
-#define HAVE_LZMA_MT
-#endif
+/* Multithreading support in stable API since xz 5.2.0; detected by autoconf */
+/* #define HAVE_LZMA_MT */
 
 #define kBufferSize (1 << 15)
 
@@ -795,7 +793,11 @@ static LZFILE *lzopen_internal(const char *mode, int fd, int xz)
 #ifdef HAVE_LZMA_MT
 	    } else {
 		if (threads == -1)
+#ifdef _SC_NPROCESSORS_ONLN
 		    threads = sysconf(_SC_NPROCESSORS_ONLN);
+#else
+			threads = 1;
+#endif
 		lzma_mt mt_options = {
 		    .flags = 0,
 		    .threads = threads,
-- 
2.24.0

