diff -Nur sh-utils-2.0.11.orig/src/getgid.c sh-utils-2.0.11/src/getgid.c
--- sh-utils-2.0.11.orig/src/getgid.c	Thu Jan  1 01:00:00 1970
+++ sh-utils-2.0.11/src/getgid.c	Thu Jun  7 23:59:52 2001
@@ -0,0 +1,41 @@
+/* getgid - print GID of given group
+   Copyright (C) 1999 Artur Frysiak <wiget@pld.org.pl>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+
+				    
+#include <grp.h>
+#include <sys/types.h>
+
+int main(int argc, char **argv)
+{
+	struct group 	*gr;
+	int		retval;
+
+	if (argc > 1) {
+		gr = getgrnam(argv[1]);
+		if (gr) {
+			printf("%d\n",gr->gr_gid);
+			retval = 0;
+		} else {
+			retval = 1;
+		}
+	} else {
+		printf("Usage:\n\t%s groupname\n",argv[0]);
+		retval = 1;
+	}
+	return retval;
+}
+
