--- sh-utils-2.0/src/uptime.c.orig	Mon May 10 17:10:54 1999
+++ sh-utils-2.0/src/uptime.c	Fri Sep  7 01:59:52 2001
@@ -18,6 +18,9 @@
 /* Created by hacking who.c by Kaveh Ghazi ghazi@caip.rutgers.edu.  */
 
 #include <config.h>
+#ifdef __MINT__
+#define HAVE_PROC_UPTIME 1
+#endif
 #include <getopt.h>
 #include <stdio.h>
 
@@ -58,7 +61,11 @@
   FILE *fp;
   double upsecs;
 
+#ifdef __MINT__
+  fp = fopen ("/kern/uptime", "r");
+#else
   fp = fopen ("/proc/uptime", "r");
+#endif
   if (fp != NULL)
     {
       char buf[BUFSIZ];
