diff -rup coreutils-8.32.orig/src/uptime.c coreutils-8.32/src/uptime.c
--- coreutils-8.32.orig/src/uptime.c	2020-02-04 20:27:08.000000000 +0100
+++ coreutils-8.32/src/uptime.c	2020-09-05 14:50:36.618758792 +0200
@@ -17,6 +17,9 @@
 /* Created by hacking who.c by Kaveh Ghazi ghazi@caip.rutgers.edu.  */
 
 #include <config.h>
+#ifdef __MINT__
+#define HAVE_PROC_UPTIME 1
+#endif
 #include <stdio.h>
 
 #include <sys/types.h>
@@ -62,7 +65,11 @@ print_uptime (size_t n, const STRUCT_UTM
 #ifdef HAVE_PROC_UPTIME
   FILE *fp;
 
+#ifdef __MINT__
+  fp = fopen ("/kern/uptime", "r");
+#else
   fp = fopen ("/proc/uptime", "r");
+#endif
   if (fp != NULL)
     {
       char buf[BUFSIZ];
