diff -rupN arj-3.10.22.orig/garble.c arj-3.10.22/garble.c
--- arj-3.10.22.orig/garble.c	2004-02-21 23:53:34.000000000 +0100
+++ arj-3.10.22/garble.c	2017-10-15 07:14:26.150006187 +0200
@@ -13,7 +13,7 @@
 #ifdef TILED
  #include <dos.h>
 #endif
-#if TARGET==UNIX
+#if TARGET==UNIX && !defined(__MINT__)
  #include <dlfcn.h>
 #endif
 
@@ -175,6 +175,8 @@ static int arjcrypt_init(char *name)
   atexit(remove_arjcrypt);
   if((arjcrypt_entry=(VOID *)GetProcAddress(arjcrypt_hmod, (LPCSTR)1L))==NULL)
    error(M_NO_ARJCRYPT_ENTRY);
+ #elif defined(__MINT__)
+   error(M_ARJCRYPT_ERROR);
  #elif TARGET==UNIX
   if((arjcrypt_hmod=dlopen(name, RTLD_NOW))==NULL)
   {
diff -rupN arj-3.10.22.orig/gnu/makefile.in arj-3.10.22/gnu/makefile.in
--- arj-3.10.22.orig/gnu/makefile.in	2004-04-17 13:28:06.000000000 +0200
+++ arj-3.10.22/gnu/makefile.in	2017-10-15 07:20:50.861997816 +0200
@@ -16,6 +16,7 @@ REGISTER = @REGISTER@
 
 # Programs
 
+CC = @CC@
 INSTALL = @INSTALL@
 INSTALL_DIR = @INSTALL@ -d -m 755
 INSTALL_DATA = @INSTALL_DATA@
@@ -170,7 +171,6 @@ all:	prepare timestamp
 
 do-all:	\
 	$(ARJ_DIR)/arj$x \
-	$(ARJCRYPT_DIR)/arjcrypt$d \
 	$(REARJ_DIR)/rearj$x \
 	$(REGISTER_DIR)/$(REGISTER)$x \
 	$(ARJDISP_DIR)/arjdisp$x \
@@ -350,7 +350,7 @@ $(ARJ_DIR)/arj$x: $(ARJ_OBJS) \
 	$(TOOLS_DIR)/join $(ARJ_DIR)/arj$x $(ARJSFXV_DIR)/arjsfxv$x
 	$(TOOLS_DIR)/join $(ARJ_DIR)/arj$x $(SFXSTUB_DIR)/sfxstub$x
 	rm -f $(BASEDIR)/help.arj
-	$(ARJ_DIR)/arj$x a $(BASEDIR)/help.arj -+ -t1f -2e -e -jm -jh65535 -jt $(RES_DIR)/$(LOCALE)/arj?.txt
+	$(NATIVE_ARJ) a $(BASEDIR)/help.arj -+ -t1f -2e -e -jm -jh65535 -jt $(RES_DIR)/$(LOCALE)/arj?.txt
 	$(TOOLS_DIR)/join $(ARJ_DIR)/arj$x $(BASEDIR)/help.arj
 	$(TOOLS_DIR)/postproc $@
 
@@ -450,7 +450,7 @@ install: all
 	$(INSTALL_DIR) $(DESTDIR)$(pkglibdir)
 
 	$(INSTALL_PROGRAM) $(ARJ_DIR)/arj$x $(DESTDIR)$(bindir)
-	$(INSTALL_PROGRAM) $(ARJCRYPT_DIR)/arjcrypt$d $(DESTDIR)$(pkglibdir)
+#	$(INSTALL_PROGRAM) $(ARJCRYPT_DIR)/arjcrypt$d $(DESTDIR)$(pkglibdir)
 	$(INSTALL_PROGRAM) $(REARJ_DIR)/rearj$x $(DESTDIR)$(bindir)
 	$(INSTALL_PROGRAM) $(ARJDISP_DIR)/arjdisp$x $(DESTDIR)$(bindir)
 	$(INSTALL_PROGRAM) $(REGISTER_DIR)/$(REGISTER)$x $(DESTDIR)$(bindir)
diff -rupN arj-3.10.22.orig/integr.c arj-3.10.22/integr.c
--- arj-3.10.22.orig/integr.c	2005-06-21 20:27:18.000000000 +0200
+++ arj-3.10.22/integr.c	2017-10-15 07:14:26.154006187 +0200
@@ -5,7 +5,7 @@
  *
  */
 
-static const char intergrity_identifier[] = {
+const char intergrity_identifier[] = {
 	0xB0, 0x03, 0xB0, 0x02, 0xB0, 0x03,
 	0xB0, 0x04, 0xB0, 0x05,
 	0x90, 0x90, 0x90, 0x90,
--- arj-3.10.22/msgbind.c.orig	2017-10-15 07:47:18.365963270 +0200
+++ arj-3.10.22/msgbind.c	2017-10-15 07:49:00.485961048 +0200
@@ -56,8 +56,7 @@ struct pool
 
 /* A Q&D emulation of the strupr() and strlwr() for those who don't have it */
 
-#ifndef HAVE_STRLWR
-static char *strlwr(char *str)
+static char *my_strlwr(char *str)
 {
  char *p;
 
@@ -65,10 +64,8 @@ static char *strlwr(char *str)
   *p=tolower(*p);
  return(str);
 }
-#endif
 
-#ifndef HAVE_STRUPR
-static char *strupr(char *str)
+static char *my_strupr(char *str)
 {
  char *p;
 
@@ -76,7 +73,6 @@ static char *strupr(char *str)
   *p=toupper(*p);
  return(str);
 }
-#endif
 
 /* Remove given characters */
 
@@ -339,11 +335,11 @@ char *get_msg(FILE *resfile, char *targe
      read_brackets(resfile, locales, sizeof(locales), 1)==NULL||
      read_brackets(resfile, msgtype, sizeof(msgtype), 0)==NULL)
    return(NULL);
-  strlwr(targets);
-  strlwr(systems);
-  strlwr(bindings);
-  strlwr(locales);
-  strupr(msgtype);
+  my_strlwr(targets);
+  my_strlwr(systems);
+  my_strlwr(bindings);
+  my_strlwr(locales);
+  my_strupr(msgtype);
 
   locale_offset=is_in_brackets(locales, locale);
   if(is_in_brackets(targets, target)&&is_in_brackets(systems, c_system)&&(bindings[0]=='\0'||strchr(bindings, binding)!=NULL)&&locale_offset)
@@ -381,7 +377,7 @@ void put_hdr(FILE *file, char *name, cha
  cur_unixtime=time(NULL);
  stm=localtime(&cur_unixtime);
  strcpyn(buf, name, sizeof(buf));
- strupr(buf);
+ my_strupr(buf);
  fprintf(file, "/*\n"
           " * %-29s, %04u/%02u/%02u\n"
           " * ---------------------------------------------------------------------------\n"
@@ -389,7 +385,7 @@ void put_hdr(FILE *file, char *name, cha
           " * %s.\n"
           " * To rebuild the language resources, run MSGBIND.\n"
           " *\n"
-          " */\n\n", buf, (unsigned int)stm->tm_year+1900, stm->tm_mon+1, stm->tm_mday, strupr(src));
+          " */\n\n", buf, (unsigned int)stm->tm_year+1900, stm->tm_mon+1, stm->tm_mday, my_strupr(src));
 }
 
 /* And so we begin... */
@@ -479,9 +475,9 @@ int main(int argc, char **argv)
  strcat(target_f, ".c");
  strcat(target_h, ".h");
  strcpyn(locale, argv[5], sizeof(locale));
- strlwr(target);
- strlwr(c_system);
- strlwr(locale);
+ my_strlwr(target);
+ my_strlwr(c_system);
+ my_strlwr(locale);
  /* Block out all signals, since this transaction is mission-critical */
  signal(SIGINT, SIG_IGN);
  #ifndef NO_TERM_HDL
--- arj-3.10.22/fardata.c.orig	2017-10-15 08:04:43.913708864 +0200
+++ arj-3.10.22/fardata.c	2017-10-15 08:04:58.333708550 +0200
@@ -190,7 +190,7 @@
 
 /* Length-limited strlen() */
 
-static int strnlen(const char FAR *s, int count)
+static int my_strnlen(const char FAR *s, int count)
 {
  const char FAR *sc;
 
@@ -569,7 +569,7 @@
     if(!s)
      s="(null)";
 #endif
-    len=strnlen(s, precision);
+    len=my_strnlen(s, precision);
     if(!(flags&LEFT))
     {
      while(len<field_width--)
