diff -rup libxml2-2.10.3.orig/SAX2.c libxml2-2.10.3/SAX2.c
--- libxml2-2.10.3.orig/SAX2.c	2022-10-14 14:22:16.000000000 +0200
+++ libxml2-2.10.3/SAX2.c	2023-03-02 07:57:05.096496681 +0100
@@ -2207,7 +2207,6 @@ xmlSAX2StartElementNs(void *ctx,
     xmlNodePtr parent;
     xmlNsPtr last = NULL, ns;
     const xmlChar *uri, *pref;
-    xmlChar *lname = NULL;
     int i, j;
 
     if (ctx == NULL) return;
@@ -2227,20 +2226,6 @@ xmlSAX2StartElementNs(void *ctx,
     }
 
     /*
-     * Take care of the rare case of an undefined namespace prefix
-     */
-    if ((prefix != NULL) && (URI == NULL)) {
-        if (ctxt->dictNames) {
-	    const xmlChar *fullname;
-
-	    fullname = xmlDictQLookup(ctxt->dict, prefix, localname);
-	    if (fullname != NULL)
-	        localname = fullname;
-	} else {
-	    lname = xmlBuildQName(localname, prefix, NULL, 0);
-	}
-    }
-    /*
      * allocate the node
      */
     if (ctxt->freeElems != NULL) {
@@ -2254,10 +2239,7 @@ xmlSAX2StartElementNs(void *ctx,
 	if (ctxt->dictNames)
 	    ret->name = localname;
 	else {
-	    if (lname == NULL)
-		ret->name = xmlStrdup(localname);
-	    else
-	        ret->name = lname;
+	    ret->name = xmlStrdup(localname);
 	    if (ret->name == NULL) {
 	        xmlSAX2ErrMemory(ctxt, "xmlSAX2StartElementNs");
 		return;
@@ -2269,11 +2251,8 @@ xmlSAX2StartElementNs(void *ctx,
 	if (ctxt->dictNames)
 	    ret = xmlNewDocNodeEatName(ctxt->myDoc, NULL,
 	                               (xmlChar *) localname, NULL);
-	else if (lname == NULL)
-	    ret = xmlNewDocNode(ctxt->myDoc, NULL, localname, NULL);
 	else
-	    ret = xmlNewDocNodeEatName(ctxt->myDoc, NULL,
-	                               (xmlChar *) lname, NULL);
+	    ret = xmlNewDocNode(ctxt->myDoc, NULL, localname, NULL);
 	if (ret == NULL) {
 	    xmlSAX2ErrMemory(ctxt, "xmlSAX2StartElementNs");
 	    return;
@@ -2384,31 +2363,6 @@ xmlSAX2StartElementNs(void *ctx,
      */
     if (nb_attributes > 0) {
         for (j = 0,i = 0;i < nb_attributes;i++,j+=5) {
-	    /*
-	     * Handle the rare case of an undefined attribute prefix
-	     */
-	    if ((attributes[j+1] != NULL) && (attributes[j+2] == NULL)) {
-		if (ctxt->dictNames) {
-		    const xmlChar *fullname;
-
-		    fullname = xmlDictQLookup(ctxt->dict, attributes[j+1],
-		                              attributes[j]);
-		    if (fullname != NULL) {
-			xmlSAX2AttributeNs(ctxt, fullname, NULL,
-			                   attributes[j+3], attributes[j+4]);
-		        continue;
-		    }
-		} else {
-		    lname = xmlBuildQName(attributes[j], attributes[j+1],
-		                          NULL, 0);
-		    if (lname != NULL) {
-			xmlSAX2AttributeNs(ctxt, lname, NULL,
-			                   attributes[j+3], attributes[j+4]);
-			xmlFree(lname);
-		        continue;
-		    }
-		}
-	    }
 	    xmlSAX2AttributeNs(ctxt, attributes[j], attributes[j+1],
 			       attributes[j+3], attributes[j+4]);
 	}
diff -rup libxml2-2.10.3.orig/result/namespaces/err_7.xml libxml2-2.10.3/result/namespaces/err_7.xml
--- libxml2-2.10.3.orig/result/namespaces/err_7.xml	2022-08-02 15:06:29.000000000 +0200
+++ libxml2-2.10.3/result/namespaces/err_7.xml	2023-03-02 07:55:26.960656499 +0100
@@ -1,2 +1,2 @@
 <?xml version="1.0"?>
-<f:foo/>
+<foo/>
diff -rup libxml2-2.10.3.orig/result/xmlid/id_tst2.xml libxml2-2.10.3/result/xmlid/id_tst2.xml
--- libxml2-2.10.3.orig/result/xmlid/id_tst2.xml	2022-08-02 15:06:33.000000000 +0200
+++ libxml2-2.10.3/result/xmlid/id_tst2.xml	2023-03-02 07:55:26.960656499 +0100
@@ -1,6 +1,6 @@
 Object is a Node Set :
 Set contains 1 nodes:
-1  ELEMENT n:foo
+1  ELEMENT foo
     ATTRIBUTE id
       TEXT
         content=bar
diff -rup libxml2-2.10.3.orig/result/xmlid/id_tst3.xml libxml2-2.10.3/result/xmlid/id_tst3.xml
--- libxml2-2.10.3.orig/result/xmlid/id_tst3.xml	2022-08-02 15:06:33.000000000 +0200
+++ libxml2-2.10.3/result/xmlid/id_tst3.xml	2023-03-02 07:55:26.960656499 +0100
@@ -1,6 +1,6 @@
 Object is a Node Set :
 Set contains 1 nodes:
-1  ELEMENT f:o:o
+1  ELEMENT o:o
     ATTRIBUTE id
       TEXT
         content=bar
