Index: src/devices/grops/ps.cpp
===================================================================
--- a/src/devices/grops/ps.cpp.orig	2009-01-09 15:25:52.000000000 +0100
+++ b/src/devices/grops/ps.cpp	2010-08-23 15:29:15.848156800 +0200
@@ -833,7 +833,7 @@
 {
   if (next_encoding_index == 0)
     return;
-  char *done_encoding = new char[next_encoding_index];
+  char *done_encoding = new char[next_encoding_index +1];
   for (int i = 0; i < next_encoding_index; i++)
     done_encoding[i] = 0;
   for (font_pointer_list *f = font_list; f; f = f->next) {
