Index: groff-1.20.1/tmac/tty.tmac
===================================================================
--- groff-1.20.1.orig/tmac/tty.tmac	2009-01-09 15:25:52.000000000 +0100
+++ groff-1.20.1/tmac/tty.tmac	2011-06-08 11:06:28.611516653 +0200
@@ -65,8 +65,11 @@
 .fchar \[.i] i
 .fchar \[bq] ,
 .
-.if '\*[.T]'utf8' \
+.if '\*[.T]'utf8' \{\
 .  mso unicode.tmac
+.\" use ascii hyphenation characters - bnc#68385
+.  shc -
+.\}
 .
 .\" color definitions
 .defcolor black rgb #000000
