diff -rupN curl-7.56.0.orig/libcurl.pc.in curl-7.56.0/libcurl.pc.in
--- curl-7.56.0.orig/libcurl.pc.in	2017-09-06 14:44:27.000000000 +0200
+++ curl-7.56.0/libcurl.pc.in	2017-10-19 21:17:22.645858348 +0200
@@ -34,6 +34,6 @@ Name: libcurl
 URL: https://curl.haxx.se/
 Description: Library to transfer files with ftp, http, etc.
 Version: @CURLVERSION@
-Libs: -L${libdir} -lcurl
-Libs.private: @LIBCURL_LIBS@
+Libs: -L${libdir} -lcurl @LIBCURL_LIBS@
+Libs.private:
 Cflags: -I${includedir} @CPPFLAG_CURL_STATICLIB@
--- a/configure.ac.orig	2023-03-07 07:02:00.799284557 +0100
+++ b/configure.ac	2023-03-07 07:42:30.591324477 +0100
@@ -2523,10 +2523,12 @@
   AC_SEARCH_LIBS(psl_builtin, psl,
     [curl_psl_msg="yes";
      AC_DEFINE([USE_LIBPSL], [1], [PSL support enabled])
+     LIBS="$LIBS -lidn2 -lunistring -liconv"
      ],
     [curl_psl_msg="no      (libpsl not found)";
      AC_MSG_WARN([libpsl was not found])
-     ]
+     ],
+     [-lidn2 -lunistring -liconv]
   )
 fi
 AM_CONDITIONAL([USE_LIBPSL], [test "$curl_psl_msg" = "yes"])
