diff -rupN curl-7.56.0.orig/libcurl.pc.in curl-7.56.0/libcurl.pc.in
--- curl-7.56.0.orig/libcurl.pc.in	2017-09-06 14:44:27.000000000 +0200
+++ curl-7.56.0/libcurl.pc.in	2017-10-19 21:17:22.645858348 +0200
@@ -34,6 +34,6 @@ Name: libcurl
 URL: https://curl.haxx.se/
 Description: Library to transfer files with ftp, http, etc.
 Version: @CURLVERSION@
-Libs: -L${libdir} -lcurl
-Libs.private: @LIBCURL_LIBS@
+Libs: -L${libdir} -lcurl @LIBCURL_LIBS@
+Libs.private:
 Cflags: -I${includedir} @CPPFLAG_CURL_STATICLIB@
