--- a/src/globals.h.orig	2023-04-04 09:42:53.578354578 +0200
+++ b/src/globals.h	2023-04-04 09:43:17.310318881 +0200
@@ -241,6 +241,8 @@
 	RSC header and support struct
 *******************************************************************************/
 
+#ifndef __RSXHDR
+#define __RSXHDR
 typedef struct {		/* fichier ressource ‚tendu */
   UWORD rsh_vrsn;			/* should be 3														     */
   UWORD rsh_extvrsn;		/* not used, initialised to 'IN' for Interface */
@@ -262,6 +264,7 @@
   ULONG rsh_nimages;
   ULONG rsh_rssize;			/* total bytes in resource   */
 } RSXHDR;
+#endif
 
 typedef struct {
 	unsigned long	rlen;
