--- ./Configurations/unix-Makefile.tmpl.orig	2022-09-23 16:30:10.771929639 +0200
+++ ./Configurations/unix-Makefile.tmpl	2022-09-23 16:33:09.243923560 +0200
@@ -964,7 +964,7 @@
 	    echo 'Name: OpenSSL-libcrypto'; \
 	    echo 'Description: OpenSSL cryptography library'; \
 	    echo 'Version: '$(VERSION); \
-	    echo 'Libs: -L$${libdir} -lcrypto'; \
+	    echo 'Libs: -L$${libdir} -lcrypto $(LIB_EX_LIBS)'; \
 	    echo 'Libs.private: $(LIB_EX_LIBS)'; \
 	    echo 'Cflags: -I$${includedir}' ) > libcrypto.pc
 
