--- a/crypto/bn/bn_div.c.orig	2023-03-07 19:20:01.906002317 +0100
+++ b/crypto/bn/bn_div.c	2023-03-07 19:21:43.609851054 +0100
@@ -194,6 +194,22 @@
             q;                                  \
         })
 #    define REMAINDER_IS_ALREADY_CALCULATED
+#   elif defined(__mc68020__)
+   /* Same rationale as for i386.  On the 68020-60/CPU32 the divu
+    * instruction will take a 64 bit dividend in a register pair
+    * and overwrite that pair with the remainder:quotient.
+    *                                  hyc@highlandsun.com
+    */
+#    undef bn_div_words
+#    define bn_div_words(n0,n1,d0)                \
+        ({  asm volatile (                      \
+                "divu.l %4,%2:%3"               \
+                : "=d"(rem), "=d"(q)            \
+                : "0"(n0), "1"(n1), "d"(d0)     \
+                : "cc");                        \
+            q;                                  \
+        })
+#    define REMAINDER_IS_ALREADY_CALCULATED
 #   endif                       /* __<cpu> */
 #  endif                        /* __GNUC__ */
 # endif                         /* OPENSSL_NO_ASM */
