--- a/src/audio/mint/SDL_mintaudio_gsxb.h	2023-04-08 09:36:07.583895907 +0200
+++ b/src/audio/mint/SDL_mintaudio_gsxb.h	2023-04-08 09:37:41.615728697 +0200
@@ -95,10 +95,14 @@
 
 /* Extra xbios functions */
 
-#define NSoundcmd(mode,data,data2)	\
-	(long)trap_14_wwl((short)130,(short)(mode),(short)(data),(long)(data2))
-#define NSetinterrupt(src_inter,cause,inth_addr)	\
-	(long)trap_14_wwwl((short)135,(short)(src_inter),(short)(cause),	\
-		(long)(inth_addr))
+#ifndef NSoundcmd
+#define NSoundcmd(mode,data,data2)						\
+	(long)trap_14_wwwl((short)130,(short)(mode),(short)(data),(long)(data2))
+#endif
+#ifndef NSetinterrupt
+#define NSetinterrupt(src_inter,cause,inth_addr)			\
+		(long)trap_14_wwwl((short)135,(short)(src_inter),	\
+		(short)(cause),(long)(inth_addr))
+#endif
 
 #endif /* _SDL_mintaudio_gsxb_h */
