diff -rpuN libgpg-error-1.46.orig/src/estream.c libgpg-error-1.46/src/estream.c
--- libgpg-error-1.46.orig/src/estream.c	2022-08-02 14:53:50.000000000 +0200
+++ libgpg-error-1.46/src/estream.c	2023-03-07 09:02:13.848236752 +0100
@@ -52,14 +52,14 @@
  * OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifdef USE_ESTREAM_SUPPORT_H
-# include <estream-support.h>
-#endif
-
 #ifdef HAVE_CONFIG_H
 # include <config.h>
 #endif
 
+#ifdef USE_ESTREAM_SUPPORT_H
+# include <estream-support.h>
+#endif
+
 #if defined(_WIN32) && !defined(HAVE_W32_SYSTEM)
 # define HAVE_W32_SYSTEM 1
 #endif
diff -rpuN libgpg-error-1.46.orig/src/syscfg/lock-obj-pub.m68k-atari-mint.h libgpg-error-1.46/src/syscfg/lock-obj-pub.m68k-atari-mint.h
--- libgpg-error-1.46.orig/src/syscfg/lock-obj-pub.m68k-atari-mint.h	1970-01-01 01:00:00.000000000 +0100
+++ libgpg-error-1.46/src/syscfg/lock-obj-pub.m68k-atari-mint.h	2023-03-07 09:02:13.848236752 +0100
@@ -0,0 +1,23 @@
+## lock-obj-pub.m68k-atari-mint.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff -rpuN libgpg-error-1.46.orig/src/syscfg/lock-obj-pub.m68k-atari-mintelf.h libgpg-error-1.46/src/syscfg/lock-obj-pub.m68k-atari-mintelf.h
--- libgpg-error-1.46.orig/src/syscfg/lock-obj-pub.m68k-atari-mintelf.h	1970-01-01 01:00:00.000000000 +0100
+++ libgpg-error-1.46/src/syscfg/lock-obj-pub.m68k-atari-mintelf.h	2023-03-07 09:02:13.848236752 +0100
@@ -0,0 +1,23 @@
+## lock-obj-pub.m68k-atari-mint.h
+## File created by gen-posix-lock-obj - DO NOT EDIT
+## To be included by mkheader into gpg-error.h
+
+typedef struct
+{
+  long _vers;
+  union {
+    volatile char _priv[24];
+    long _x_align;
+    long *_xp_align;
+  } u;
+} gpgrt_lock_t;
+
+#define GPGRT_LOCK_INITIALIZER {1,{{0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0, \
+                                    0,0,0,0,0,0,0,0}}}
+##
+## Local Variables:
+## mode: c
+## buffer-read-only: t
+## End:
+##
diff -rpuN libgpg-error-1.46.orig/tests/t-poll.c libgpg-error-1.46/tests/t-poll.c
--- libgpg-error-1.46.orig/tests/t-poll.c	2022-01-25 18:56:13.000000000 +0100
+++ libgpg-error-1.46/tests/t-poll.c	2023-03-07 09:02:13.848236752 +0100
@@ -438,9 +438,11 @@ main (int argc, char **argv)
   if (es_set_nonblock (test_stderr, 1))
     fail ("error setting test_stderr to nonblock: %s\n", strerror (errno));
 
+#if defined(_WIN32) || defined(USE_POSIX_THREADS)
   launch_thread (producer_thread, &peer_stdin );
   launch_thread (consumer_thread, &peer_stdout);
   launch_thread (consumer_thread, &peer_stderr);
+#endif
   test_poll ();
   show ("Waiting for threads to terminate...\n");
   es_fclose (test_stdin);
