From 9a16bddd517334c627797ec6f00059a76bfaf1c9 Mon Sep 17 00:00:00 2001
From: Thorsten Otto <admin@tho-otto.de>
Date: Thu, 14 Mar 2024 15:28:10 +0100
Subject: [PATCH 09/40] Remove unneeded include of <dlfcn.h>

---
 common/paths_posix.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/common/paths_posix.cpp b/common/paths_posix.cpp
index f45d187..030fee1 100644
--- a/common/paths_posix.cpp
+++ b/common/paths_posix.cpp
@@ -15,7 +15,6 @@
 #include <cstdlib>
 #include <cstring>
 #include <errno.h>
-#include <dlfcn.h>
 #include <pwd.h>
 #include <stdexcept>
 #include <sys/stat.h>
-- 
2.41.0

