diff -rup cairo-1.18.0.orig/src/cairo-ft-private.h cairo-1.18.0/src/cairo-ft-private.h
--- cairo-1.18.0.orig/src/cairo-ft-private.h	2023-09-23 16:18:32.000000000 +0200
+++ cairo-1.18.0/src/cairo-ft-private.h	2023-11-03 12:20:09.175246402 +0100
@@ -63,10 +63,10 @@ cairo_private cairo_status_t
 _cairo_ft_face_decompose_glyph_outline (FT_Face		     face,
 					cairo_path_fixed_t **pathp);
 
-#if HAVE_FT_SVG_DOCUMENT
-
 typedef struct FT_Color_ FT_Color;
 
+#if HAVE_FT_SVG_DOCUMENT
+
 cairo_private cairo_status_t
 _cairo_render_svg_glyph (const char           *svg_document,
                          unsigned long         first_glyph,
