diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdk.c gtk+-1.2.10/gdk/gdk.c
--- gtk+-1.2.10.orig/gdk/gdk.c	2001-03-10 04:32:30.000000000 +0100
+++ gtk+-1.2.10/gdk/gdk.c	2023-03-29 12:18:01.295936914 +0200
@@ -963,7 +963,7 @@ gdk_set_sm_client_id (const gchar* sm_cl
       XChangeProperty (gdk_display, gdk_leader_window,
 	   	       gdk_atom_intern ("SM_CLIENT_ID", FALSE),
 		       XA_STRING, 8, PropModeReplace,
-		       sm_client_id, strlen(sm_client_id));
+		       (const unsigned char *)sm_client_id, strlen(sm_client_id));
     }
   else
      XDeleteProperty (gdk_display, gdk_leader_window,
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkcc.c gtk+-1.2.10/gdk/gdkcc.c
--- gtk+-1.2.10.orig/gdk/gdkcc.c	1999-09-19 04:19:26.000000000 +0200
+++ gtk+-1.2.10/gdk/gdkcc.c	2023-03-29 12:18:01.295936914 +0200
@@ -147,11 +147,12 @@ query_colors (GdkColorContext *cc)
 }
 
 static void
-init_bw (GdkColorContext *cc)
+init_bw (GdkColorContext *cc, const char *where)
 {
   GdkColor color;
   
-  g_warning ("init_bw: failed to allocate colors, falling back to black and white");
+  if (where)
+    g_warning ("%s: failed to allocate colors, falling back to black and white", where);
   
   cc->mode = GDK_CC_MODE_BW;
   
@@ -207,7 +208,7 @@ init_gray (GdkColorContext *cc)
 	    {
 	      g_free (cc->clut);
 	      cc->clut = NULL;
-	      init_bw (cc);
+	      init_bw (cc, "init_gray");
 	      g_free (cstart);
 	      return;
 	    }
@@ -431,7 +432,7 @@ init_direct_color (GdkColorContext *cc)
 	    {
 	      g_free (cc->clut);
 	      cc->clut = NULL;
-	      init_bw (cc);
+	      init_bw (cc, "init_direct_color");
 	      break;
 	    }
 	}
@@ -567,7 +568,7 @@ gdk_color_context_new (GdkVisual   *visu
 			       "GDK_VISUAL_GRAYSCALE"));
 	  
 	  if (GDK_VISUAL_XVISUAL (cc->visual)->map_entries == 2)
-	    init_bw (cc);
+	    init_bw (cc, NULL);
 	  else
 	    init_gray (cc);
 	  
@@ -642,7 +643,7 @@ gdk_color_context_new_mono (GdkVisual
   cc->mode = GDK_CC_MODE_UNDEFINED;
   cc->need_to_free_colormap = FALSE;
   
-  init_bw (cc);
+  init_bw (cc, NULL);
   
   return (GdkColorContext *) cc;
 }
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkfont.c gtk+-1.2.10/gdk/gdkfont.c
--- gtk+-1.2.10.orig/gdk/gdkfont.c	2001-04-02 04:31:25.000000000 +0200
+++ gtk+-1.2.10/gdk/gdkfont.c	2023-03-29 12:21:26.979735250 +0200
@@ -845,14 +845,11 @@ _gdk_font_wc_to_glyphs (GdkFont        *
 			gchar         **result,
 			gint           *result_length)
 {
-  XFontStruct *xfont;
   GdkFontPrivate *font_private = (GdkFontPrivate*) font;
 
   g_return_val_if_fail (font != NULL, FALSE);
   g_return_val_if_fail (font->type == GDK_FONT_FONT, FALSE);
 
-  xfont = (XFontStruct *) font_private->xfont;
-
   if (FONT_IS_8BIT (font_private))
     {
       /* 8-bit font, assume that we are in a 8-bit locale,
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkgc.c gtk+-1.2.10/gdk/gdkgc.c
--- gtk+-1.2.10.orig/gdk/gdkgc.c	2000-01-27 22:10:44.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkgc.c	2023-03-29 12:18:01.295936914 +0200
@@ -832,7 +832,7 @@ gdk_gc_set_dashes (GdkGC      *gc,
 
   private = (GdkGCPrivate*) gc;
 
-  XSetDashes (private->xdisplay, private->xgc, dash_offset, dash_list, n);
+  XSetDashes (private->xdisplay, private->xgc, dash_offset, (const char *)dash_list, n);
 }
 
 void
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkim.c gtk+-1.2.10/gdk/gdkim.c
--- gtk+-1.2.10.orig/gdk/gdkim.c	2001-04-02 04:22:18.000000000 +0200
+++ gtk+-1.2.10/gdk/gdkim.c	2023-03-29 12:20:10.899809842 +0200
@@ -1581,7 +1581,7 @@ _gdk_wcstombs_len (const GdkWChar *src,
 	   * We must copy the string into an area allocated by glib, because
 	   * the string 'tpr.value' must be freed by XFree().
 	   */
-	  mbstr = g_strdup(tpr.value);
+	  mbstr = g_strdup((char *)tpr.value);
 	  XFree (tpr.value);
 	}
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkimage.c gtk+-1.2.10/gdk/gdkimage.c
--- gtk+-1.2.10.orig/gdk/gdkimage.c	2001-02-20 17:42:20.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkimage.c	2023-03-29 12:18:01.299936911 +0200
@@ -59,6 +59,7 @@ static void gdk_image_put_normal (GdkDra
 				  gint         ydest,
 				  gint         width,
 				  gint         height);
+#ifdef USE_SHM
 static void gdk_image_put_shared (GdkDrawable *drawable,
 				  GdkGC       *gc,
 				  GdkImage    *image,
@@ -68,6 +69,7 @@ static void gdk_image_put_shared (GdkDra
 				  gint         ydest,
 				  gint         width,
 				  gint         height);
+#endif
 
 
 static GList *image_list = NULL;
@@ -480,6 +482,7 @@ gdk_image_put_normal (GdkDrawable *drawa
 	     xsrc, ysrc, xdest, ydest, width, height);
 }
 
+#ifdef USE_SHM
 static void
 gdk_image_put_shared (GdkDrawable *drawable,
 		      GdkGC       *gc,
@@ -491,7 +494,6 @@ gdk_image_put_shared (GdkDrawable *drawa
 		      gint         width,
 		      gint         height)
 {
-#ifdef USE_SHM
   GdkWindowPrivate *drawable_private;
   GdkImagePrivate *image_private;
   GdkGCPrivate *gc_private;
@@ -511,7 +513,5 @@ gdk_image_put_shared (GdkDrawable *drawa
   XShmPutImage (drawable_private->xdisplay, drawable_private->xwindow,
 		gc_private->xgc, image_private->ximage,
 		xsrc, ysrc, xdest, ydest, width, height, False);
-#else /* USE_SHM */
-  g_error ("trying to draw shared memory image when gdk was compiled without shared memory support");
-#endif /* USE_SHM */
 }
+#endif /* USE_SHM */
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkinput.c gtk+-1.2.10/gdk/gdkinput.c
--- gtk+-1.2.10.orig/gdk/gdkinput.c	2000-01-27 22:10:44.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkinput.c	2023-03-29 12:18:01.299936911 +0200
@@ -46,12 +46,12 @@ static GdkDevicePrivate *gdk_input_find_
 
 /* Incorporate the specific routines depending on compilation options */
 
-static const GdkAxisUse gdk_input_core_axes[] = { GDK_AXIS_X, GDK_AXIS_Y };
+static GdkAxisUse gdk_input_core_axes[] = { GDK_AXIS_X, GDK_AXIS_Y };
 
 static const GdkDeviceInfo gdk_input_core_info =
 {
   GDK_CORE_POINTER,
-  "Core Pointer",
+  (gchar *)"Core Pointer",
   GDK_SOURCE_MOUSE,
   GDK_MODE_SCREEN,
   TRUE,
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkinputcommon.h gtk+-1.2.10/gdk/gdkinputcommon.h
--- gtk+-1.2.10.orig/gdk/gdkinputcommon.h	1999-11-12 22:27:10.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkinputcommon.h	2023-03-29 12:18:01.299936911 +0200
@@ -26,6 +26,8 @@
 
 #if defined(XINPUT_GXI) || defined(XINPUT_XFREE)
 
+#include <string.h>
+
 /* Forward declarations */
 static void gdk_input_get_root_relative_geometry (Display *dpy, Window w, 
 						  int *x_ret, int *y_ret,
@@ -463,7 +465,7 @@ gdk_input_common_init(gint include_core)
       XFreeDeviceList(devices);
     }
 
-  gdk_input_devices = g_list_append (gdk_input_devices, &gdk_input_core_info);
+  gdk_input_devices = g_list_append (gdk_input_devices, (void *)&gdk_input_core_info);
 
   return TRUE;
 }
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkpixmap.c gtk+-1.2.10/gdk/gdkpixmap.c
--- gtk+-1.2.10.orig/gdk/gdkpixmap.c	2000-01-22 18:43:44.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkpixmap.c	2023-03-29 12:18:01.299936911 +0200
@@ -445,12 +445,11 @@ _gdk_pixmap_create_from_xpm (GdkWindow
   GdkVisual *visual;
   GdkGC *gc = NULL;
   GdkColor tmp_color;
-  gint width, height, num_cols, cpp, n, ns, cnt, xcnt, ycnt, wbytes;
+  gint width, height, num_cols, cpp, n, cnt, xcnt, ycnt, wbytes;
   gchar *buffer, pixel_str[32];
   gchar *name_buf;
   _GdkPixmapColor *color = NULL, *fallbackcolor = NULL;
   _GdkPixmapColor *colors = NULL;
-  gulong index;
   GHashTable *color_hash = NULL;
   _GdkPixmapInfo *color_info = NULL;
   
@@ -541,7 +540,6 @@ _gdk_pixmap_create_from_xpm (GdkWindow
 	fallbackcolor = color;
     }
   
-  index = 0;
   image = gdk_image_new (GDK_IMAGE_FASTEST, visual, width, height);
   
   if (mask)
@@ -579,7 +577,6 @@ _gdk_pixmap_create_from_xpm (GdkWindow
 	{
 	  strncpy (pixel_str, &buffer[n], cpp);
 	  pixel_str[cpp] = 0;
-	  ns = 0;
 	  
 	  color = g_hash_table_lookup (color_hash, pixel_str);
 	  
@@ -766,7 +763,8 @@ gdk_pixmap_foreign_new (guint32 anid)
   GdkWindowPrivate *private;
   Pixmap xpixmap;
   Window root_return;
-  unsigned int x_ret, y_ret, w_ret, h_ret, bw_ret, depth_ret;
+  int x_ret, y_ret;
+  unsigned int w_ret, h_ret, bw_ret, depth_ret;
 
   /* check to make sure we were passed something at
      least a little sane */
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkrgb.c gtk+-1.2.10/gdk/gdkrgb.c
--- gtk+-1.2.10.orig/gdk/gdkrgb.c	2001-02-19 18:11:02.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkrgb.c	2023-03-29 12:18:01.299936911 +0200
@@ -533,7 +533,9 @@ gdk_rgb_set_gray_cmap (GdkColormap *cmap
 {
   gint i;
   GdkColor color;
+#ifdef VERBOSE
   gint status;
+#endif
   gulong pixels[256];
   gint r, g, b, gray;
 
@@ -543,7 +545,10 @@ gdk_rgb_set_gray_cmap (GdkColormap *cmap
       color.red = i * 257;
       color.green = i * 257;
       color.blue = i * 257;
-      status = gdk_color_alloc (cmap, &color);
+#ifdef VERBOSE
+      status =
+#endif
+      gdk_color_alloc (cmap, &color);
       pixels[i] = color.pixel;
 #ifdef VERBOSE
       g_print ("allocating pixel %d, %x %x %x, result %d\n",
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkselection.c gtk+-1.2.10/gdk/gdkselection.c
--- gtk+-1.2.10.orig/gdk/gdkselection.c	2001-03-15 20:09:36.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkselection.c	2023-03-29 12:28:20.227295275 +0200
@@ -278,7 +278,7 @@ gdk_text_property_to_text_list (GdkAtom
     {
       gint sanitized_text_length = length;
       
-      property.value = sanitized_text = sanitize_ctext (text, &sanitized_text_length);
+      property.value = (guchar *)(sanitized_text = sanitize_ctext ((char *)text, &sanitized_text_length));
       property.nitems = sanitized_text_length;
     }
   else
@@ -336,10 +336,10 @@ gdk_string_to_compound_text (const gchar
     *format = property.format;
 
   sanitized_text_length = property.nitems;
-  sanitized_text = sanitize_ctext (property.value, &sanitized_text_length);
+  sanitized_text = sanitize_ctext ((char *)property.value, &sanitized_text_length);
 
   if (ctext)
-    *ctext = sanitized_text;
+    *ctext = (guchar *)sanitized_text;
   else
     g_free (sanitized_text);
   
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkvisual.c gtk+-1.2.10/gdk/gdkvisual.c
--- gtk+-1.2.10.orig/gdk/gdkvisual.c	1999-02-24 08:33:17.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkvisual.c	2023-03-29 12:18:01.295936914 +0200
@@ -219,8 +219,8 @@ gdk_visual_init (void)
 #ifdef G_ENABLE_DEBUG 
   if (gdk_debug_flags & GDK_DEBUG_MISC)
     for (i = 0; i < nvisuals; i++)
-      g_message ("visual: %s: %d",
-		 visual_names[visuals[i].visual.type],
+      g_message ("visual[%d]: %s: %d",
+		 i, visual_names[visuals[i].visual.type],
 		 visuals[i].visual.depth);
 #endif /* G_ENABLE_DEBUG */
 
@@ -253,8 +253,21 @@ gdk_visual_init (void)
 	}
     }
 
+#ifdef G_ENABLE_DEBUG 
+  if (gdk_debug_flags & GDK_DEBUG_MISC)
+  {
+  	g_message ("navailable_depths: %d", navailable_depths);
+    for (i = 0; i < navailable_depths; i++)
+      g_message ("  depth[%d]: %d", i, available_depths[i]);
+  	g_message ("navailable_types: %d", navailable_types);
+    for (i = 0; i < navailable_types; i++)
+      g_message ("  type[%d]: %d %s", i, available_types[i], visual_names[available_types[i]]);
+    g_message("system_visual: %s", system_visual ? visual_names[system_visual->visual.type] : "none");
+  }
+#endif /* G_ENABLE_DEBUG */
+
   for (i = 0; i < nvisuals; i++)
-    gdk_visual_add ((GdkVisual*) &visuals[i]);
+    gdk_visual_add (&visuals[i].visual);
 
   if (npossible_types == 0)
     g_error ("unable to find a usable visual type");
@@ -287,13 +300,13 @@ gdk_visual_get_best_type (void)
 GdkVisual*
 gdk_visual_get_system (void)
 {
-  return ((GdkVisual*) system_visual);
+  return &system_visual->visual;
 }
 
 GdkVisual*
 gdk_visual_get_best (void)
 {
-  return ((GdkVisual*) &(visuals[0]));
+  return &visuals[0].visual;
 }
 
 GdkVisual*
@@ -306,7 +319,7 @@ gdk_visual_get_best_with_depth (gint dep
   for (i = 0; i < nvisuals; i++)
     if (depth == visuals[i].visual.depth)
       {
-	return_val = (GdkVisual*) &(visuals[i]);
+	return_val = &visuals[i].visual;
 	break;
       }
 
@@ -323,7 +336,7 @@ gdk_visual_get_best_with_type (GdkVisual
   for (i = 0; i < nvisuals; i++)
     if (visual_type == visuals[i].visual.type)
       {
-	return_val = (GdkVisual*) &(visuals[i]);
+	return_val = &visuals[i].visual;
 	break;
       }
 
@@ -342,7 +355,7 @@ gdk_visual_get_best_with_both (gint
     if ((depth == visuals[i].visual.depth) &&
 	(visual_type == visuals[i].visual.type))
       {
-	return_val = (GdkVisual*) &(visuals[i]);
+	return_val = &visuals[i].visual;
 	break;
       }
 
@@ -398,7 +411,7 @@ gdkx_visual_get (VisualID xvisualid)
 
   for (i = 0; i < nvisuals; i++)
     if (xvisualid == visuals[i].xvisual->visualid)
-      return (GdkVisual*) &visuals[i];
+      return &visuals[i].visual;
 
   return NULL;
 }
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gdk/gdkwindow.c gtk+-1.2.10/gdk/gdkwindow.c
--- gtk+-1.2.10.orig/gdk/gdkwindow.c	2001-02-22 23:02:11.000000000 +0100
+++ gtk+-1.2.10/gdk/gdkwindow.c	2023-03-29 12:18:01.299936911 +0200
@@ -1201,7 +1201,7 @@ gdk_window_set_role (GdkWindow   *window
   if (role)
     XChangeProperty (private->xdisplay, private->xwindow,
 		     gdk_atom_intern ("WM_WINDOW_ROLE", FALSE), XA_STRING,
-		     8, PropModeReplace, role, strlen (role));
+		     8, PropModeReplace, (const unsigned char *)role, strlen (role));
   else
     XDeleteProperty (private->xdisplay, private->xwindow,
 		     gdk_atom_intern ("WM_WINDOW_ROLE", FALSE));
@@ -2078,7 +2078,7 @@ gdk_window_set_icon_name (GdkWindow   *w
   if (window_private->destroyed)
     return;
   res = XmbTextListToTextProperty (window_private->xdisplay,
-				   &name, 1, XStdICCTextStyle,
+				   (char **)&name, 1, XStdICCTextStyle,
                                	   &property);
   if (res < 0)
     {
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkaccellabel.c gtk+-1.2.10/gtk/gtkaccellabel.c
--- gtk+-1.2.10.orig/gtk/gtkaccellabel.c	1999-02-24 08:33:28.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkaccellabel.c	2023-03-29 12:18:01.299936911 +0200
@@ -89,14 +89,10 @@ gtk_accel_label_class_init (GtkAccelLabe
 {
   GtkObjectClass *object_class;
   GtkWidgetClass *widget_class;
-  GtkMiscClass *misc_class;
-  GtkLabelClass *label_class;
   
   accel_label_class = class;
   object_class = (GtkObjectClass*) class;
   widget_class = (GtkWidgetClass*) class;
-  misc_class = (GtkMiscClass*) class;
-  label_class = (GtkLabelClass*) class;
   
   parent_class = gtk_type_class (gtk_label_get_type ());
   
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkalignment.c gtk+-1.2.10/gtk/gtkalignment.c
--- gtk+-1.2.10.orig/gtk/gtkalignment.c	1999-02-24 08:33:32.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkalignment.c	2023-03-29 12:18:01.299936911 +0200
@@ -234,14 +234,12 @@ static void
 gtk_alignment_size_request (GtkWidget      *widget,
 			    GtkRequisition *requisition)
 {
-  GtkAlignment *alignment;
   GtkBin *bin;
 
   g_return_if_fail (widget != NULL);
   g_return_if_fail (GTK_IS_ALIGNMENT (widget));
   g_return_if_fail (requisition != NULL);
 
-  alignment = GTK_ALIGNMENT (widget);
   bin = GTK_BIN (widget);
 
   requisition->width = GTK_CONTAINER (widget)->border_width * 2;
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkbbox.c gtk+-1.2.10/gtk/gtkbbox.c
--- gtk+-1.2.10.orig/gtk/gtkbbox.c	1999-02-24 08:33:41.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkbbox.c	2023-03-29 12:18:01.299936911 +0200
@@ -65,9 +65,7 @@ gtk_button_box_get_type (void)
 static void
 gtk_button_box_class_init (GtkButtonBoxClass *class)
 {
-  GtkWidgetClass *widget_class;
-
-  widget_class = (GtkWidgetClass*) class;
+  (void)class;
 }
 
 static void
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkbin.c gtk+-1.2.10/gtk/gtkbin.c
--- gtk+-1.2.10.orig/gtk/gtkbin.c	1999-05-06 04:59:00.000000000 +0200
+++ gtk+-1.2.10/gtk/gtkbin.c	2023-03-29 12:18:01.299936911 +0200
@@ -77,11 +77,9 @@ gtk_bin_get_type (void)
 static void
 gtk_bin_class_init (GtkBinClass *class)
 {
-  GtkObjectClass *object_class;
   GtkWidgetClass *widget_class;
   GtkContainerClass *container_class;
 
-  object_class = (GtkObjectClass*) class;
   widget_class = (GtkWidgetClass*) class;
   container_class = (GtkContainerClass*) class;
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkbindings.c gtk+-1.2.10/gtk/gtkbindings.c
--- gtk+-1.2.10.orig/gtk/gtkbindings.c	1999-05-06 04:59:01.000000000 +0200
+++ gtk+-1.2.10/gtk/gtkbindings.c	2023-03-29 12:18:01.299936911 +0200
@@ -1296,7 +1296,7 @@ gtk_binding_parse_signal (GScanner
 	  else
 	    done = TRUE;
 	  break;
-	case '-':
+	case G_TOKEN_MINUS:
 	  if (!need_arg)
 	    done = TRUE;
 	  else if (negate)
@@ -1307,14 +1307,14 @@ gtk_binding_parse_signal (GScanner
 	  else
 	    negate = TRUE;
 	  break;
-	case ',':
+	case G_TOKEN_COMMA:
 	  seen_comma = TRUE;
 	  if (need_arg)
 	    done = TRUE;
 	  else
 	    need_arg = TRUE;
 	  break;
-	case ')':
+	case G_TOKEN_RIGHT_PAREN:
 	  if (!(need_arg && seen_comma) && !negate)
 	    {
 	      args = g_slist_reverse (args);
@@ -1360,7 +1360,7 @@ gtk_binding_parse_bind (GScanner       *
   g_return_val_if_fail (scanner != NULL, G_TOKEN_ERROR);
   
   g_scanner_get_next_token (scanner);
-  if (scanner->token != GTK_RC_TOKEN_BIND)
+  if (scanner->token != (GTokenType)GTK_RC_TOKEN_BIND)
     return GTK_RC_TOKEN_BIND;
   g_scanner_get_next_token (scanner);
   if (scanner->token != G_TOKEN_STRING)
@@ -1411,7 +1411,7 @@ gtk_binding_parse_binding (GScanner
   g_return_val_if_fail (scanner != NULL, G_TOKEN_ERROR);
 
   g_scanner_get_next_token (scanner);
-  if (scanner->token != GTK_RC_TOKEN_BINDING)
+  if (scanner->token != (GTokenType)GTK_RC_TOKEN_BINDING)
     return GTK_RC_TOKEN_BINDING;
   g_scanner_get_next_token (scanner);
   if (scanner->token != G_TOKEN_STRING)
@@ -1433,7 +1433,7 @@ gtk_binding_parse_binding (GScanner
   g_scanner_peek_next_token (scanner);
   while (scanner->next_token != '}')
     {
-      switch (scanner->next_token)
+      switch ((int)scanner->next_token)
 	{
 	  guint expected_token;
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkcalendar.c gtk+-1.2.10/gtk/gtkcalendar.c
--- gtk+-1.2.10.orig/gtk/gtkcalendar.c	2001-03-15 19:41:40.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkcalendar.c	2023-03-29 12:24:08.315577068 +0200
@@ -214,7 +214,7 @@ enum {
   LAST_SIGNAL
 };
 
-static gint gtk_calendar_signals[LAST_SIGNAL] = { 0 };
+static guint gtk_calendar_signals[LAST_SIGNAL] = { 0 };
 
 static GtkWidgetClass *parent_class = NULL;
 
@@ -241,8 +241,8 @@ struct _GtkCalendarPrivateData
 
   guint min_day_width;
   guint max_day_char_width;
-  guint max_day_char_ascent;
-  guint max_day_char_descent;
+  gint max_day_char_ascent;
+  gint max_day_char_descent;
   guint max_label_char_ascent;
   guint max_label_char_descent;
   guint max_week_char_width;
@@ -752,13 +752,10 @@ static void
 gtk_calendar_set_year_next (GtkCalendar *calendar)
 {
   gint month_len;
-  GtkWidget *widget;
   
   g_return_if_fail (calendar != NULL);
   g_return_if_fail (GTK_IS_WIDGET (calendar));
   
-  widget = GTK_WIDGET (calendar);
-  
   gtk_calendar_freeze (calendar);
   
   calendar->year++;
@@ -1429,17 +1426,12 @@ gtk_calendar_size_allocate (GtkWidget
 static void
 gtk_calendar_draw_focus (GtkWidget *widget)
 {
-  GtkCalendar *calendar;
-  GtkCalendarPrivateData *private_data;
   gint width, height;
   gint x, y;
   
   g_return_if_fail (widget != NULL);
   g_return_if_fail (GTK_IS_CALENDAR (widget));
 
-  calendar = GTK_CALENDAR (widget);
-  private_data = GTK_CALENDAR_PRIVATE_DATA (widget);
-  
   if (GTK_WIDGET_DRAWABLE (widget))
     {
       x = 0;
@@ -1462,14 +1454,12 @@ static gint
 gtk_calendar_expose (GtkWidget	    *widget,
 		     GdkEventExpose *event)
 {
-  GtkCalendar *calendar;
   GtkCalendarPrivateData *private_data;
 
   g_return_val_if_fail (widget != NULL, FALSE);
   g_return_val_if_fail (GTK_IS_CALENDAR (widget), FALSE);
   g_return_val_if_fail (event != NULL, FALSE);
   
-  calendar = GTK_CALENDAR (widget);
   private_data = GTK_CALENDAR_PRIVATE_DATA (widget);
   
   if (GTK_WIDGET_DRAWABLE (widget))
@@ -1509,14 +1499,12 @@ static void
 gtk_calendar_paint (GtkWidget	 *widget,
 		    GdkRectangle *area)
 {
-  GtkCalendar *calendar;
   GtkCalendarPrivateData *private_data;
 
   g_return_if_fail (widget != NULL);
   g_return_if_fail (widget->window != NULL);
   g_return_if_fail (GTK_IS_CALENDAR (widget));
   
-  calendar = GTK_CALENDAR (widget);
   private_data = GTK_CALENDAR_PRIVATE_DATA (widget);
   
   if (private_data->main_win != NULL)
@@ -1540,8 +1528,8 @@ gtk_calendar_paint_header (GtkWidget *wi
   GtkCalendar *calendar;
   GdkGC *gc;
   char buffer[255];
-  int y, y_arrow;
-  gint header_width, cal_height;
+  int y;
+  gint header_width;
   gint str_width;
   gint max_month_width;
   gint max_year_width;
@@ -1562,7 +1550,6 @@ gtk_calendar_paint_header (GtkWidget *wi
   gdk_window_clear (private_data->header_win);
   
   header_width = widget->allocation.width - 4;
-  cal_height = widget->allocation.height;
   
   max_month_width = private_data->max_month_width;
   max_year_width = private_data->max_year_width;
@@ -1577,7 +1564,6 @@ gtk_calendar_paint_header (GtkWidget *wi
   y = private_data->header_h - (private_data->header_h 
 				- HEADER_FONT (widget)->ascent
 				+ HEADER_FONT (widget)->descent) / 2;
-  y_arrow = (private_data->header_h - 9) / 2;
   
   /* Draw year and its arrows */
   sprintf (buffer, "%d", calendar->year);
@@ -1625,7 +1611,6 @@ gtk_calendar_paint_day_names (GtkWidget
   char buffer[255];
   int day,i;
   int day_width, cal_width;
-  gint cal_height;
   int day_wid_sep;
   int str_width;
   GtkCalendarPrivateData *private_data;
@@ -1655,7 +1640,6 @@ gtk_calendar_paint_day_names (GtkWidget
   
   day_width = private_data->day_width;
   cal_width = widget->allocation.width;
-  cal_height = widget->allocation.height;
   day_wid_sep = day_width + DAY_XSEP;
   
   /*
@@ -1702,7 +1686,9 @@ gtk_calendar_paint_week_numbers (GtkWidg
 {
   GtkCalendar *calendar;
   GdkGC *gc;
-  gint row, week = 0, year;
+  gint row;
+  N_int year;
+  N_int week = 0;
   gint x_loc;
   char buffer[3];
   gint y_baseline, day_height;
@@ -1828,7 +1814,6 @@ gtk_calendar_paint_day (GtkWidget *widge
   gint x_loc;
   gint y_top;
   gint y_baseline;
-  gint day_xspace;
   GtkCalendarPrivateData *private_data;
   
   g_return_if_fail (widget != NULL);
@@ -1850,8 +1835,6 @@ gtk_calendar_paint_day (GtkWidget *widge
   
   day_height = row_height (calendar);
   
-  day_xspace = private_data->day_width - private_data->max_day_char_width*2;
-  
   day = calendar->day[row][col];
   
   x_left = left_x_for_column (calendar, col);
@@ -1928,7 +1911,6 @@ gtk_calendar_paint_day (GtkWidget *widge
 static void
 gtk_calendar_paint_main (GtkWidget *widget)
 {
-  GtkCalendar *calendar;
   GtkCalendarPrivateData *private_data;
   gint row, col;
   
@@ -1936,7 +1918,6 @@ gtk_calendar_paint_main (GtkWidget *widg
   g_return_if_fail (widget->window != NULL);
   g_return_if_fail (GTK_IS_CALENDAR (widget));
   
-  calendar = GTK_CALENDAR (widget);
   private_data = GTK_CALENDAR_PRIVATE_DATA (widget);
 
   if (private_data->freeze_count)
@@ -2311,7 +2292,6 @@ gtk_calendar_button_press (GtkWidget	  *
 {
   GtkCalendar *calendar;
   GtkCalendarPrivateData *private_data;
-  gint x, y;
   
   g_return_val_if_fail (widget != NULL, FALSE);
   g_return_val_if_fail (GTK_IS_CALENDAR (widget), FALSE);
@@ -2320,9 +2300,6 @@ gtk_calendar_button_press (GtkWidget	  *
   calendar = GTK_CALENDAR (widget);
   private_data = GTK_CALENDAR_PRIVATE_DATA (widget);
   
-  x = (gint) (event->x);
-  y = (gint) (event->y);
-  
   if (event->window == private_data->main_win)
     gtk_calendar_main_button (widget, event);
   
@@ -2390,13 +2367,11 @@ static gint
 gtk_calendar_enter_notify (GtkWidget	    *widget,
 			   GdkEventCrossing *event)
 {
-  GtkCalendar *calendar;
   GtkCalendarPrivateData *private_data;
   
   g_return_val_if_fail (widget != NULL, FALSE);
   g_return_val_if_fail (event != NULL, FALSE);
   
-  calendar = GTK_CALENDAR (widget);
   private_data = GTK_CALENDAR_PRIVATE_DATA (widget);
   
   if (event->window == private_data->arrow_win[ARROW_MONTH_LEFT])
@@ -2565,14 +2540,12 @@ gtk_calendar_thaw (GtkCalendar *calendar
 static void
 gtk_calendar_set_background (GtkWidget *widget)
 {
-  GtkCalendar *calendar;
   GtkCalendarPrivateData *private_data;
   gint i;
   
   g_return_if_fail (widget != NULL);
   g_return_if_fail (GTK_IS_CALENDAR (widget));
   
-  calendar = GTK_CALENDAR (widget);
   private_data = GTK_CALENDAR_PRIVATE_DATA (widget);
 
   if (GTK_WIDGET_REALIZED (widget))
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkcheckbutton.c gtk+-1.2.10/gtk/gtkcheckbutton.c
--- gtk+-1.2.10.orig/gtk/gtkcheckbutton.c	2001-02-12 22:37:27.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkcheckbutton.c	2023-03-29 12:25:01.723524704 +0200
@@ -165,7 +165,6 @@ static void
 gtk_check_button_draw (GtkWidget    *widget,
 		       GdkRectangle *area)
 {
-  GtkCheckButton *check_button;
   GtkToggleButton *toggle_button;
   GtkBin *bin;
   GdkRectangle child_area;
@@ -174,7 +173,6 @@ gtk_check_button_draw (GtkWidget    *wid
   g_return_if_fail (GTK_IS_CHECK_BUTTON (widget));
   g_return_if_fail (area != NULL);
   
-  check_button = GTK_CHECK_BUTTON (widget);
   toggle_button = GTK_TOGGLE_BUTTON (widget);
   bin = GTK_BIN (widget);
   
@@ -316,7 +314,6 @@ static gint
 gtk_check_button_expose (GtkWidget      *widget,
 			 GdkEventExpose *event)
 {
-  GtkCheckButton *check_button;
   GtkToggleButton *toggle_button;
   GtkBin *bin;
   GdkEventExpose child_event;
@@ -325,7 +322,6 @@ gtk_check_button_expose (GtkWidget
   g_return_val_if_fail (GTK_IS_CHECK_BUTTON (widget), FALSE);
   g_return_val_if_fail (event != NULL, FALSE);
   
-  check_button = GTK_CHECK_BUTTON (widget);
   toggle_button = GTK_TOGGLE_BUTTON (widget);
   bin = GTK_BIN (widget);
   
@@ -371,7 +367,6 @@ gtk_real_check_button_draw_indicator (Gt
 				      GdkRectangle   *area)
 {
   GtkWidget *widget;
-  GtkToggleButton *toggle_button;
   GtkStateType state_type;
   GtkShadowType shadow_type;
   GdkRectangle restrict_area;
@@ -386,7 +381,6 @@ gtk_real_check_button_draw_indicator (Gt
   g_return_if_fail (GTK_IS_CHECK_BUTTON (check_button));
   
   widget = GTK_WIDGET (check_button);
-  toggle_button = GTK_TOGGLE_BUTTON (check_button);
   
   if (GTK_WIDGET_DRAWABLE (check_button))
     {
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkclist.c gtk+-1.2.10/gtk/gtkclist.c
--- gtk+-1.2.10.orig/gtk/gtkclist.c	2001-03-01 01:14:09.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkclist.c	2023-03-29 12:18:01.303936906 +0200
@@ -1879,7 +1879,7 @@ abort_column_resize (GtkCList *clist)
   if (GTK_CLIST_ADD_MODE(clist))
     {
       gdk_gc_set_line_attributes (clist->xor_gc, 1, GDK_LINE_ON_OFF_DASH, 0,0);
-      gdk_gc_set_dashes (clist->xor_gc, 0, "\4\4", 2);
+      gdk_gc_set_dashes (clist->xor_gc, 0, (gint8 *)"\4\4", 2);
     }
 }
 
@@ -2763,7 +2763,7 @@ static void
 real_remove_row (GtkCList *clist,
 		 gint      row)
 {
-  gint was_visible, was_selected;
+  gint was_visible;
   GList *list;
   GtkCListRow *clist_row;
 
@@ -2775,7 +2775,6 @@ real_remove_row (GtkCList *clist,
     return;
 
   was_visible = (gtk_clist_row_is_visible (clist, row) != GTK_VISIBILITY_NONE);
-  was_selected = 0;
 
   /* get the row we're going to delete */
   list = ROW_ELEMENT (clist, row);
@@ -3657,7 +3656,7 @@ toggle_add_mode (GtkCList *clist)
       GTK_CLIST_SET_FLAG (clist, CLIST_ADD_MODE);
       gdk_gc_set_line_attributes (clist->xor_gc, 1,
 				  GDK_LINE_ON_OFF_DASH, 0, 0);
-      gdk_gc_set_dashes (clist->xor_gc, 0, "\4\4", 2);
+      gdk_gc_set_dashes (clist->xor_gc, 0, (gint8 *)"\4\4", 2);
     }
   else
     {
@@ -5197,7 +5196,7 @@ gtk_clist_button_release (GtkWidget
 	{
 	  gdk_gc_set_line_attributes (clist->xor_gc, 1,
 				      GDK_LINE_ON_OFF_DASH, 0, 0);
-	  gdk_gc_set_dashes (clist->xor_gc, 0, "\4\4", 2);
+	  gdk_gc_set_dashes (clist->xor_gc, 0, (gint8 *)"\4\4", 2);
 	}
 
       width = new_column_width (clist, i, &x);
@@ -6164,6 +6163,7 @@ vadjustment_changed (GtkAdjustment *adju
   g_return_if_fail (data != NULL);
 
   clist = GTK_CLIST (data);
+  (void)clist;
 }
 
 static void
@@ -6176,6 +6176,7 @@ hadjustment_changed (GtkAdjustment *adju
   g_return_if_fail (data != NULL);
 
   clist = GTK_CLIST (data);
+  (void)clist;
 }
 
 static void
@@ -6578,7 +6579,6 @@ gtk_clist_focus (GtkContainer     *conta
 {
   GtkCList *clist;
   GtkWidget *focus_child;
-  gint old_row;
 
   g_return_val_if_fail (container != NULL, FALSE);
   g_return_val_if_fail (GTK_IS_CLIST (container), FALSE);
@@ -6588,7 +6588,6 @@ gtk_clist_focus (GtkContainer     *conta
   
   clist = GTK_CLIST (container);
   focus_child = container->focus_child;
-  old_row = clist->focus_row;
 
   switch (direction)
     {
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkcolorsel.c gtk+-1.2.10/gtk/gtkcolorsel.c
--- gtk+-1.2.10.orig/gtk/gtkcolorsel.c	1999-08-17 21:49:06.000000000 +0200
+++ gtk+-1.2.10/gtk/gtkcolorsel.c	2023-03-29 12:18:01.303936906 +0200
@@ -257,11 +257,9 @@ gtk_color_selection_class_init (GtkColor
 {
   GtkObjectClass *object_class;
   GtkWidgetClass *widget_class;
-  GtkContainerClass *container_class;
 
   object_class = (GtkObjectClass*) klass;
   widget_class = (GtkWidgetClass*) klass;
-  container_class = (GtkContainerClass*) klass;
 
   color_selection_parent_class = gtk_type_class (GTK_TYPE_VBOX);
   
@@ -1083,14 +1081,13 @@ gtk_color_selection_update_wheel (GtkCol
                                   gint               y)
 {
   gdouble wid, heig;
-  gint res;
 
   gtk_color_selection_draw_wheel_marker (colorsel);
 
   wid = (gdouble) (colorsel->wheel_area->allocation.width) / 2.0;
   heig = (gdouble) (colorsel->wheel_area->allocation.height) / 2.0;
 
-  res = gtk_color_selection_eval_wheel (x, y, wid, heig, &colorsel->values[HUE],
+  gtk_color_selection_eval_wheel (x, y, wid, heig, &colorsel->values[HUE],
                                         &colorsel->values[SATURATION]);
 
   HSV_TO_RGB ();
@@ -1686,10 +1683,6 @@ gtk_color_selection_dialog_get_type (voi
 static void
 gtk_color_selection_dialog_class_init (GtkColorSelectionDialogClass *klass)
 {
-  GtkObjectClass *object_class;
-
-  object_class = (GtkObjectClass*) klass;
-
   color_selection_dialog_parent_class = gtk_type_class (GTK_TYPE_WINDOW);
 }
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkctree.c gtk+-1.2.10/gtk/gtkctree.c
--- gtk+-1.2.10.orig/gtk/gtkctree.c	2001-02-19 22:08:18.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkctree.c	2023-03-29 12:18:01.303936906 +0200
@@ -730,7 +730,7 @@ gtk_ctree_realize (GtkWidget *widget)
     {
       gdk_gc_set_line_attributes (ctree->lines_gc, 1, 
 				  GDK_LINE_ON_OFF_DASH, None, None);
-      gdk_gc_set_dashes (ctree->lines_gc, 0, "\1\1", 2);
+      gdk_gc_set_dashes (ctree->lines_gc, 0, (gint8 *)"\1\1", 2);
     }
 }
 
@@ -1606,7 +1606,6 @@ draw_row (GtkCList     *clist,
 {
   GtkWidget *widget;
   GtkCTree  *ctree;
-  GdkRectangle *rect;
   GdkRectangle *crect;
   GdkRectangle row_rectangle;
   GdkRectangle cell_rectangle; 
@@ -1668,7 +1667,6 @@ draw_row (GtkCList     *clist,
   /* draw the cell borders */
   if (area)
     {
-      rect = &intersect_rectangle;
       crect = &intersect_rectangle;
 
       if (gdk_rectangle_intersect (area, &cell_rectangle, crect))
@@ -1678,7 +1676,6 @@ draw_row (GtkCList     *clist,
     }
   else
     {
-      rect = &clip_rectangle;
       crect = &cell_rectangle;
 
       gdk_draw_rectangle (clist->clist_window,
@@ -2509,7 +2506,6 @@ real_tree_expand (GtkCTree     *ctree,
   GtkCTreeNode *work;
   GtkRequisition requisition;
   gboolean visible;
-  gint level;
 
   g_return_if_fail (ctree != NULL);
   g_return_if_fail (GTK_IS_CTREE (ctree));
@@ -2522,7 +2518,6 @@ real_tree_expand (GtkCTree     *ctree,
   GTK_CLIST_CLASS_FW (clist)->resync_selection (clist, NULL);
 
   GTK_CTREE_ROW (node)->expanded = TRUE;
-  level = GTK_CTREE_ROW (node)->level;
 
   visible = gtk_ctree_is_viewable (ctree, node);
   /* get cell width if tree_column is auto resized */
@@ -3518,7 +3513,6 @@ ctree_is_hot_spot (GtkCTree     *ctree,
 {
   GtkCTreeRow *tree_row;
   GtkCList *clist;
-  GtkCellPixText *cell;
   gint xl;
   gint yu;
   
@@ -3534,8 +3528,6 @@ ctree_is_hot_spot (GtkCTree     *ctree,
 
   tree_row = GTK_CTREE_ROW (node);
 
-  cell = GTK_CELL_PIXTEXT(tree_row->row.cell[ctree->tree_column]);
-
   yu = (ROW_TOP_YPIXEL (clist, row) + (clist->row_height - PM_SIZE) / 2 -
 	(clist->row_height - 1) % 2);
 
@@ -3871,9 +3863,6 @@ gtk_ctree_remove_node (GtkCTree     *ctr
 
   if (node)
     {
-      gboolean visible;
-
-      visible = gtk_ctree_is_viewable (ctree, node);
       gtk_ctree_unlink (ctree, node, TRUE);
       gtk_ctree_post_recursive (ctree, node, GTK_CTREE_FUNC (tree_delete),
 				NULL);
@@ -5394,7 +5383,7 @@ gtk_ctree_set_line_style (GtkCTree
 	  if (GTK_WIDGET_REALIZED (ctree))
 	    gdk_gc_set_line_attributes (ctree->lines_gc, 1, 
 					GDK_LINE_ON_OFF_DASH, None, None);
-	  gdk_gc_set_dashes (ctree->lines_gc, 0, "\1\1", 2);
+	  gdk_gc_set_dashes (ctree->lines_gc, 0, (gint8 *)"\1\1", 2);
 	  break;
 	case GTK_CTREE_LINES_TABBED:
 	  if (GTK_WIDGET_REALIZED (ctree))
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkcurve.c gtk+-1.2.10/gtk/gtkcurve.c
--- gtk+-1.2.10.orig/gtk/gtkcurve.c	1999-06-09 11:07:28.000000000 +0200
+++ gtk+-1.2.10/gtk/gtkcurve.c	2023-03-29 12:18:01.303936906 +0200
@@ -378,7 +378,6 @@ gtk_curve_graph_events (GtkWidget *widge
 {
   GdkCursorType new_type = c->cursor_type;
   gint i, src, dst, leftbound, rightbound;
-  GdkEventButton *bevent;
   GdkEventMotion *mevent;
   GtkWidget *w;
   gint tx, ty;
@@ -431,7 +430,6 @@ gtk_curve_graph_events (GtkWidget *widge
     case GDK_BUTTON_PRESS:
       gtk_grab_add (widget);
 
-      bevent = (GdkEventButton *) event;
       new_type = GDK_TCROSS;
 
       switch (c->curve_type)
@@ -735,7 +733,7 @@ gtk_curve_reset (GtkCurve *c)
 void
 gtk_curve_set_gamma (GtkCurve *c, gfloat gamma)
 {
-  gfloat x, one_over_gamma, height, one_over_width;
+  gfloat x, one_over_gamma, height;
   GtkCurveType old_type;
   gint i;
 
@@ -749,7 +747,6 @@ gtk_curve_set_gamma (GtkCurve *c, gfloat
     one_over_gamma = 1.0;
   else
     one_over_gamma = 1.0 / gamma;
-  one_over_width = 1.0 / (c->num_points - 1);
   height = c->height;
   for (i = 0; i < c->num_points; ++i)
     {
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkdnd.c gtk+-1.2.10/gtk/gtkdnd.c
--- gtk+-1.2.10.orig/gtk/gtkdnd.c	2001-03-24 00:24:31.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkdnd.c	2023-03-29 12:18:01.303936906 +0200
@@ -28,6 +28,7 @@
 
 #include "gdk/gdkx.h"
 #include "gdk/gdkkeysyms.h"
+#include <stdlib.h>
 
 #include "gtkdnd.h"
 #include "gtkinvisible.h"
@@ -252,70 +253,70 @@ static gint gtk_drag_abort_timeout
 
 #define action_ask_width 16
 #define action_ask_height 16
-static const guchar action_ask_bits[] = {
+static const char action_ask_bits[] = {
   0x00, 0x00, 0xfe, 0x7f, 0xfe, 0x1f, 0x06, 0xc0, 0x76, 0xf8, 0xb6, 0xf7, 
   0xd6, 0xec, 0x66, 0xdb, 0x66, 0xdb, 0x96, 0xec, 0x76, 0xf7, 0x76, 0xfb, 
   0xf6, 0xfc, 0x72, 0xfb, 0x7a, 0xfb, 0xf8, 0xfc, };
 
 #define action_ask_mask_width 16
 #define action_ask_mask_height 16
-static const guchar action_ask_mask_bits[] = {
+static const char action_ask_mask_bits[] = {
   0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x3f, 0x8f, 0x07, 0xcf, 0x0f, 
   0xef, 0x1f, 0xff, 0x3c, 0xff, 0x3c, 0x6f, 0x1f, 0x8f, 0x0f, 0x8f, 0x07, 
   0x0f, 0x03, 0x8f, 0x07, 0x87, 0x07, 0x07, 0x03, };
 
 #define action_copy_width 16
 #define action_copy_height 16
-static const guchar action_copy_bits[] = {
+static const char action_copy_bits[] = {
   0x00, 0x00, 0xfe, 0x7f, 0xfe, 0x1f, 0x06, 0xc0, 0x76, 0xfb, 0x76, 0xfb, 
   0x76, 0xfb, 0x06, 0x83, 0xf6, 0xbf, 0xf6, 0xbf, 0x06, 0x83, 0x76, 0xfb, 
   0x76, 0xfb, 0x72, 0xfb, 0x7a, 0xf8, 0xf8, 0xff, };
 
 #define action_copy_mask_width 16
 #define action_copy_mask_height 16
-static const guchar action_copy_mask_bits[] = {
+static const char action_copy_mask_bits[] = {
   0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x3f, 0x8f, 0x07, 0x8f, 0x07, 
   0x8f, 0x07, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0xff, 0x7f, 0x8f, 0x07, 
   0x8f, 0x07, 0x8f, 0x07, 0x87, 0x07, 0x07, 0x00, };
 
 #define action_move_width 16
 #define action_move_height 16
-static const guchar action_move_bits[] = {
+static const char action_move_bits[] = {
   0x00, 0x00, 0xfe, 0x7f, 0xfe, 0x1f, 0x06, 0xc0, 0x96, 0xff, 0x26, 0xff, 
   0xc6, 0xf8, 0xd6, 0xe3, 0x96, 0x8f, 0xb6, 0xbf, 0x36, 0xc3, 0x76, 0xfb, 
   0x76, 0xfa, 0xf2, 0xfa, 0xfa, 0xf8, 0xf8, 0xff, };
 
 #define action_move_mask_width 16
 #define action_move_mask_height 16
-static const guchar action_move_mask_bits[] = {
+static const char action_move_mask_bits[] = {
   0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x3f, 0x6f, 0x00, 0xff, 0x00, 
   0xff, 0x07, 0xef, 0x1f, 0xef, 0x7f, 0xcf, 0x7f, 0xcf, 0x3f, 0x8f, 0x07, 
   0x8f, 0x07, 0x0f, 0x07, 0x07, 0x07, 0x07, 0x00, };
 
 #define action_link_width 16
 #define action_link_height 16
-static const guchar action_link_bits[] = {
+static const char action_link_bits[] = {
   0x00, 0x00, 0xfe, 0x7f, 0xfe, 0x1f, 0x06, 0xc0, 0x36, 0xf8, 0xd6, 0xf7, 
   0x66, 0xec, 0xa6, 0xe8, 0x26, 0xdf, 0xe6, 0xbd, 0xd6, 0xa7, 0xb6, 0xa8, 
   0xb6, 0xb1, 0x72, 0xdf, 0xfa, 0xe0, 0xf8, 0xff, };
 
 #define action_link_mask_width 16
 #define action_link_mask_height 16
-static const guchar action_link_mask_bits[] = {
+static const char action_link_mask_bits[] = {
   0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x3f, 0xcf, 0x07, 0xef, 0x0f, 
   0xff, 0x1f, 0x7f, 0x1f, 0xff, 0x3f, 0xff, 0x7f, 0xef, 0x7f, 0xcf, 0x77, 
   0xcf, 0x7f, 0x8f, 0x3f, 0x07, 0x1f, 0x07, 0x00, };
 
 #define action_none_width 16
 #define action_none_height 16
-static const guchar action_none_bits[] = {
+static const char action_none_bits[] = {
   0x00, 0x00, 0xfe, 0x7f, 0xfe, 0x1f, 0x06, 0xc0, 0xf6, 0xff, 0xf6, 0xff, 
   0xf6, 0xff, 0xf6, 0xff, 0xf6, 0xff, 0xf6, 0xff, 0xf6, 0xff, 0xf6, 0xff, 
   0xf6, 0xff, 0xf2, 0xff, 0xfa, 0xff, 0xf8, 0xff, };
 
 #define action_none_mask_width 16
 #define action_none_mask_height 16
-static const guchar action_none_mask_bits[] = {
+static const char action_none_mask_bits[] = {
   0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x3f, 0x0f, 0x00, 0x0f, 0x00, 
   0x0f, 0x00, 0x0f, 0x00, 0x0f, 0x00, 0x0f, 0x00, 0x0f, 0x00, 0x0f, 0x00, 
   0x0f, 0x00, 0x0f, 0x00, 0x07, 0x00, 0x07, 0x00, };
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkeditable.c gtk+-1.2.10/gtk/gtkeditable.c
--- gtk+-1.2.10.orig/gtk/gtkeditable.c	2001-03-15 19:41:40.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkeditable.c	2023-03-29 12:18:01.303936906 +0200
@@ -426,6 +426,7 @@ gtk_editable_insert_text (GtkEditable *e
   gtk_widget_ref (GTK_WIDGET (editable));
 
   klass = GTK_EDITABLE_CLASS (GTK_OBJECT (editable)->klass);
+  (void)klass;
 
   if (new_text_length <= 64)
     text = buf;
@@ -456,7 +457,8 @@ gtk_editable_delete_text (GtkEditable *e
   gtk_widget_ref (GTK_WIDGET (editable));
 
   klass = GTK_EDITABLE_CLASS (GTK_OBJECT (editable)->klass);
-
+  (void)klass;
+  
   gtk_signal_emit (GTK_OBJECT (editable), editable_signals[DELETE_TEXT], start_pos, end_pos);
   gtk_signal_emit (GTK_OBJECT (editable), editable_signals[CHANGED]);
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkentry.c gtk+-1.2.10/gtk/gtkentry.c
--- gtk+-1.2.10.orig/gtk/gtkentry.c	2001-04-02 04:14:54.000000000 +0200
+++ gtk+-1.2.10/gtk/gtkentry.c	2023-03-29 12:18:01.307936903 +0200
@@ -1743,7 +1743,7 @@ gtk_entry_insert_text (GtkEditable *edit
 
   if (new_text_length < 0)
     {
-      new_text_nt = (gchar *)new_text;
+      new_text_nt = (guchar *)new_text;
       new_text_length = strlen (new_text);
       if (new_text_length <= 0) return;
     }
@@ -1754,7 +1754,7 @@ gtk_entry_insert_text (GtkEditable *edit
   else
     {
       /* make a null-terminated copy of new_text */
-      new_text_nt = g_new (gchar, new_text_length + 1);
+      new_text_nt = g_new (guchar, new_text_length + 1);
       memcpy (new_text_nt, new_text, new_text_length);
       new_text_nt[new_text_length] = 0;
     }
@@ -1772,7 +1772,7 @@ gtk_entry_insert_text (GtkEditable *edit
   /* Convert to wide characters */
   insertion_text = g_new (GdkWChar, new_text_length);
   if (entry->use_wchar)
-    insertion_length = gdk_mbstowcs (insertion_text, new_text_nt,
+    insertion_length = gdk_mbstowcs (insertion_text, (gchar *)new_text_nt,
 				     new_text_length);
   else
     for (insertion_length=0; new_text_nt[insertion_length]; insertion_length++)
@@ -1971,14 +1971,14 @@ gtk_entry_get_chars      (GtkEditable
 	    gtk_entry_grow_text(entry);
 	  ch = entry->text[end_pos];
 	  entry->text[end_pos] = 0;
-	  mbstr = gdk_wcstombs (entry->text + start_pos);
+	  mbstr = (guchar *)gdk_wcstombs (entry->text + start_pos);
 	  entry->text[end_pos] = ch;
 	  return (gchar *)mbstr;
 	}
       else
 	{
 	  gint i;
-	  mbstr = g_new (gchar, end_pos - start_pos + 1);
+	  mbstr = g_new (guchar, end_pos - start_pos + 1);
 	  for (i=0; i<end_pos-start_pos; i++)
 	    mbstr[i] = entry->text[start_pos + i];
 	  mbstr[i] = 0;
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkfilesel.c gtk+-1.2.10/gtk/gtkfilesel.c
--- gtk+-1.2.10.orig/gtk/gtkfilesel.c	2001-02-16 05:36:19.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkfilesel.c	2023-03-29 12:18:01.307936903 +0200
@@ -1311,7 +1311,6 @@ gtk_file_selection_populate (GtkFileSele
   CompletionState *cmpl_state;
   PossibleCompletion* poss;
   gchar* filename;
-  gint row;
   gchar* rem_path = rel_path;
   gchar* sel_text;
   gchar* text[2];
@@ -1344,10 +1343,10 @@ gtk_file_selection_populate (GtkFileSele
   /* Set the dir_list to include ./ and ../ */
   text[1] = NULL;
   text[0] = "./";
-  row = gtk_clist_append (GTK_CLIST (fs->dir_list), text);
+  gtk_clist_append (GTK_CLIST (fs->dir_list), text);
 
   text[0] = "../";
-  row = gtk_clist_append (GTK_CLIST (fs->dir_list), text);
+  gtk_clist_append (GTK_CLIST (fs->dir_list), text);
 
   /*reset the max widths of the lists*/
   dir_list_width = gdk_string_width(fs->dir_list->style->font,"../");
@@ -1372,7 +1371,7 @@ gtk_file_selection_populate (GtkFileSele
 		{
 		  int width = gdk_string_width(fs->dir_list->style->font,
 					       filename);
-		  row = gtk_clist_append (GTK_CLIST (fs->dir_list), text);
+		  gtk_clist_append (GTK_CLIST (fs->dir_list), text);
 		  if(width > dir_list_width)
 		    {
 		      dir_list_width = width;
@@ -1385,7 +1384,7 @@ gtk_file_selection_populate (GtkFileSele
 	    {
 	      int width = gdk_string_width(fs->file_list->style->font,
 				           filename);
-	      row = gtk_clist_append (GTK_CLIST (fs->file_list), text);
+	      gtk_clist_append (GTK_CLIST (fs->file_list), text);
 	      if(width > file_list_width)
 	        {
 	          file_list_width = width;
@@ -2340,12 +2339,11 @@ static PossibleCompletion*
 attempt_homedir_completion(gchar* text_to_complete,
 			   CompletionState *cmpl_state)
 {
-  gint index, length;
+  gint index;
 
   if (!cmpl_state->user_dir_name_buffer &&
       !get_pwdb(cmpl_state))
     return NULL;
-  length = strlen(text_to_complete) - 1;
 
   cmpl_state->user_completion_index += 1;
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkfixed.c gtk+-1.2.10/gtk/gtkfixed.c
--- gtk+-1.2.10.orig/gtk/gtkfixed.c	1999-05-06 04:59:04.000000000 +0200
+++ gtk+-1.2.10/gtk/gtkfixed.c	2023-03-29 12:18:01.307936903 +0200
@@ -83,11 +83,9 @@ gtk_fixed_get_type (void)
 static void
 gtk_fixed_class_init (GtkFixedClass *class)
 {
-  GtkObjectClass *object_class;
   GtkWidgetClass *widget_class;
   GtkContainerClass *container_class;
 
-  object_class = (GtkObjectClass*) class;
   widget_class = (GtkWidgetClass*) class;
   container_class = (GtkContainerClass*) class;
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkfontsel.c gtk+-1.2.10/gtk/gtkfontsel.c
--- gtk+-1.2.10.orig/gtk/gtkfontsel.c	2023-03-27 19:17:56.000356912 +0200
+++ gtk+-1.2.10/gtk/gtkfontsel.c	2023-03-29 12:18:01.307936903 +0200
@@ -955,7 +955,6 @@ gtk_font_selection_expose_list (GtkWidge
 				gpointer		 data)
 {
   GtkFontSelection *fontsel;
-  FontInfo *font_info;
   GList *selection;
   gint index;
   
@@ -964,8 +963,6 @@ gtk_font_selection_expose_list (GtkWidge
 #endif
   fontsel = GTK_FONT_SELECTION(data);
   
-  font_info = fontsel_info->font_info;
-      
   /* Try to scroll the font family clist to the selected item */
   selection = GTK_CLIST(fontsel->font_clist)->selection;
   if (selection)
@@ -3434,10 +3431,6 @@ gtk_font_selection_dialog_get_type (void
 static void
 gtk_font_selection_dialog_class_init (GtkFontSelectionDialogClass *klass)
 {
-  GtkObjectClass *object_class;
-  
-  object_class = (GtkObjectClass*) klass;
-  
   font_selection_dialog_parent_class = gtk_type_class (GTK_TYPE_WINDOW);
 }
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkhandlebox.c gtk+-1.2.10/gtk/gtkhandlebox.c
--- gtk+-1.2.10.orig/gtk/gtkhandlebox.c	2001-02-23 01:53:52.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkhandlebox.c	2023-03-29 12:18:01.307936903 +0200
@@ -296,6 +296,7 @@ gtk_handle_box_destroy (GtkObject *objec
   g_return_if_fail (GTK_IS_HANDLE_BOX (object));
 
   hb = GTK_HANDLE_BOX (object);
+  (void)hb;
 
   if (GTK_OBJECT_CLASS (parent_class)->destroy)
     (* GTK_OBJECT_CLASS (parent_class)->destroy) (object);
@@ -574,11 +575,9 @@ gtk_handle_box_size_allocate (GtkWidget
 
   if (bin->child && GTK_WIDGET_VISIBLE (bin->child))
     {
-      GtkWidget *child;
       GtkAllocation child_allocation;
       guint border_width;
 
-      child = bin->child;
       border_width = GTK_CONTAINER (widget)->border_width;
 
       child_allocation.x = border_width;
@@ -746,8 +745,8 @@ gtk_handle_box_paint (GtkWidget      *wi
 {
   GtkBin *bin;
   GtkHandleBox *hb;
-  guint width;
-  guint height;
+  gint width;
+  gint height;
   GdkRectangle rect;
   GdkRectangle dest;
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkhbbox.c gtk+-1.2.10/gtk/gtkhbbox.c
--- gtk+-1.2.10.orig/gtk/gtkhbbox.c	1999-02-24 08:34:45.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkhbbox.c	2023-03-29 12:18:01.307936903 +0200
@@ -212,6 +212,7 @@ gtk_hbutton_box_size_allocate (GtkWidget
 
   box = GTK_BUTTON_BOX (widget);
   hbox = GTK_HBUTTON_BOX (widget);
+  (void)hbox;
   spacing = box->spacing != GTK_BUTTONBOX_DEFAULT
 	  ? box->spacing : default_spacing;
   layout = box->layout_style != GTK_BUTTONBOX_DEFAULT_STYLE
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkhruler.c gtk+-1.2.10/gtk/gtkhruler.c
--- gtk+-1.2.10.orig/gtk/gtkhruler.c	1999-02-24 08:34:51.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkhruler.c	2023-03-29 12:18:01.307936903 +0200
@@ -134,7 +134,7 @@ static void
 gtk_hruler_draw_ticks (GtkRuler *ruler)
 {
   GtkWidget *widget;
-  GdkGC *gc, *bg_gc;
+  GdkGC *gc;
   GdkFont *font;
   gint i;
   gint width, height;
@@ -160,7 +160,6 @@ gtk_hruler_draw_ticks (GtkRuler *ruler)
   widget = GTK_WIDGET (ruler);
 
   gc = widget->style->fg_gc[GTK_STATE_NORMAL];
-  bg_gc = widget->style->bg_gc[GTK_STATE_NORMAL];
   font = widget->style->font;
 
   xthickness = widget->style->klass->xthickness;
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkhscale.c gtk+-1.2.10/gtk/gtkhscale.c
--- gtk+-1.2.10.orig/gtk/gtkhscale.c	2001-02-12 21:36:37.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkhscale.c	2023-03-29 12:18:01.307936903 +0200
@@ -305,6 +305,7 @@ gtk_hscale_clear_background (GtkRange
   g_return_if_fail (range != NULL);
   
   widget = GTK_WIDGET (range);
+  (void)widget;
   
   gtk_hscale_pos_background (GTK_HSCALE (range), &x, &y, &width, &height);
   
@@ -359,7 +360,6 @@ gtk_hscale_size_allocate (GtkWidget
                           GtkAllocation *allocation)
 {
   GtkRange *range;
-  GtkScale *scale;
   gint width, height;
   gint x, y;
   
@@ -371,8 +371,7 @@ gtk_hscale_size_allocate (GtkWidget
   if (GTK_WIDGET_REALIZED (widget))
     {
       range = GTK_RANGE (widget);
-      scale = GTK_SCALE (widget);
-      
+
       gtk_hscale_pos_trough (GTK_HSCALE (widget), &x, &y, &width, &height);
       
       gdk_window_move_resize (range->trough, 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkitemfactory.c gtk+-1.2.10/gtk/gtkitemfactory.c
--- gtk+-1.2.10.orig/gtk/gtkitemfactory.c	2001-03-02 18:38:35.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkitemfactory.c	2023-03-29 12:28:46.035264395 +0200
@@ -219,6 +219,7 @@ gtk_item_factory_init (GtkItemFactory
   GtkObject *object;
 
   object = GTK_OBJECT (ifactory);
+  (void)object;
 
   ifactory->path = NULL;
   ifactory->accel_group = NULL;
@@ -1311,6 +1312,7 @@ gtk_item_factory_delete_item (GtkItemFac
   g_return_if_fail (path != NULL);
 
   class = GTK_ITEM_FACTORY_CLASS (GTK_OBJECT (ifactory)->klass);
+  (void)class;
 
   widget = gtk_item_factory_get_widget (ifactory, path);
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtklabel.c gtk+-1.2.10/gtk/gtklabel.c
--- gtk+-1.2.10.orig/gtk/gtklabel.c	2001-04-02 05:12:38.000000000 +0200
+++ gtk+-1.2.10/gtk/gtklabel.c	2023-03-29 12:18:01.307936903 +0200
@@ -714,13 +714,10 @@ gtk_label_finalize_lines_wrap (GtkLabel
 			       gint            max_line_width)
 {
   GtkLabelWord *word, *line, *next_line;
-  GtkWidget *widget;
-  gchar *ptrn;
   gint x, y, space, extra_width, add_space, baseline_skip;
   
   g_return_if_fail (label->wrap);
   
-  ptrn = label->pattern;
   y = 0;
   baseline_skip = (GTK_WIDGET (label)->style->font->ascent +
 		   GTK_WIDGET (label)->style->font->descent + 1);
@@ -766,7 +763,6 @@ gtk_label_finalize_lines_wrap (GtkLabel
     }
   
   label->max_width = max_line_width;
-  widget = GTK_WIDGET (label);
   requisition->width = max_line_width + 2 * label->misc.xpad;
   requisition->height = y + 2 * label->misc.ypad + 1;
 }
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkmenu.c gtk+-1.2.10/gtk/gtkmenu.c
--- gtk+-1.2.10.orig/gtk/gtkmenu.c	2001-02-22 23:07:54.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkmenu.c	2023-03-29 12:26:08.499452887 +0200
@@ -465,7 +465,6 @@ gtk_menu_popup (GtkMenu		    *menu,
 		guint		     button,
 		guint32		     activate_time)
 {
-  GtkWidget *widget;
   GtkWidget *xgrab_shell;
   GtkWidget *parent;
   GdkEvent *current_event;
@@ -474,7 +473,6 @@ gtk_menu_popup (GtkMenu		    *menu,
   g_return_if_fail (menu != NULL);
   g_return_if_fail (GTK_IS_MENU (menu));
   
-  widget = GTK_WIDGET (menu);
   menu_shell = GTK_MENU_SHELL (menu);
   
   menu_shell->parent_menu_shell = parent_menu_shell;
@@ -1040,14 +1038,12 @@ gtk_menu_expose (GtkWidget	*widget,
   GtkWidget *child;
   GdkEventExpose child_event;
   GList *children;
-  GtkMenu *menu;
   
   g_return_val_if_fail (widget != NULL, FALSE);
   g_return_val_if_fail (GTK_IS_MENU (widget), FALSE);
   g_return_val_if_fail (event != NULL, FALSE);
 
   menu_shell = GTK_MENU_SHELL (widget);
-  menu = GTK_MENU (widget);
   
   if (GTK_WIDGET_DRAWABLE (widget))
     {
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkmisc.c gtk+-1.2.10/gtk/gtkmisc.c
--- gtk+-1.2.10.orig/gtk/gtkmisc.c	1999-02-24 08:35:32.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkmisc.c	2023-03-29 12:18:06.071932232 +0200
@@ -237,6 +237,7 @@ gtk_misc_realize (GtkWidget *widget)
 
   GTK_WIDGET_SET_FLAGS (widget, GTK_REALIZED);
   misc = GTK_MISC (widget);
+  (void)misc;
 
   if (GTK_WIDGET_NO_WINDOW (widget))
     {
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtknotebook.c gtk+-1.2.10/gtk/gtknotebook.c
--- gtk+-1.2.10.orig/gtk/gtknotebook.c	2001-01-31 00:27:26.000000000 +0100
+++ gtk+-1.2.10/gtk/gtknotebook.c	2023-03-29 12:18:06.071932232 +0200
@@ -2601,7 +2601,6 @@ static void
 gtk_notebook_pages_allocate (GtkNotebook   *notebook,
 			     GtkAllocation *allocation)
 {
-  GtkWidget    *widget;
   GtkContainer *container;
   GtkNotebookPage *page = NULL;
   GtkAllocation child_allocation;
@@ -2620,7 +2619,6 @@ gtk_notebook_pages_allocate (GtkNotebook
   if (!notebook->show_tabs || !notebook->children || !notebook->cur_page)
     return;
 
-  widget = GTK_WIDGET (notebook);
   container = GTK_CONTAINER (notebook);
 
   child_allocation.x = container->border_width;
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkoptionmenu.c gtk+-1.2.10/gtk/gtkoptionmenu.c
--- gtk+-1.2.10.orig/gtk/gtkoptionmenu.c	2001-02-12 21:36:37.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkoptionmenu.c	2023-03-29 12:18:06.071932232 +0200
@@ -122,12 +122,10 @@ gtk_option_menu_class_init (GtkOptionMen
 {
   GtkObjectClass *object_class;
   GtkWidgetClass *widget_class;
-  GtkButtonClass *button_class;
   GtkContainerClass *container_class;
 
   object_class = (GtkObjectClass*) class;
   widget_class = (GtkWidgetClass*) class;
-  button_class = (GtkButtonClass*) class;
   container_class = (GtkContainerClass*) class;
 
   parent_class = gtk_type_class (gtk_button_get_type ());
@@ -453,7 +451,6 @@ gtk_option_menu_expose (GtkWidget      *
 {
   GtkWidget *child;
   GdkEventExpose child_event;
-  gint remove_child;
 
   g_return_val_if_fail (widget != NULL, FALSE);
   g_return_val_if_fail (GTK_IS_OPTION_MENU (widget), FALSE);
@@ -498,7 +495,7 @@ gtk_option_menu_expose (GtkWidget      *
       if (remove_child)
 	gtk_option_menu_remove_contents (GTK_OPTION_MENU (widget));
 #else
-      remove_child = FALSE;
+      /* remove_child = FALSE; */
       child = GTK_BIN (widget)->child;
       child_event = *event;
       if (child && GTK_WIDGET_NO_WINDOW (child) &&
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkpacker.c gtk+-1.2.10/gtk/gtkpacker.c
--- gtk+-1.2.10.orig/gtk/gtkpacker.c	1999-05-06 04:59:09.000000000 +0200
+++ gtk+-1.2.10/gtk/gtkpacker.c	2023-03-29 12:18:06.071932232 +0200
@@ -967,8 +967,6 @@ gtk_packer_size_request (GtkWidget
   GtkContainer *container;
   GtkPackerChild *child;
   GList *children;
-  gint nvis_vert_children;
-  gint nvis_horz_children;
   gint width, height;
   gint maxWidth, maxHeight;
   
@@ -981,8 +979,6 @@ gtk_packer_size_request (GtkWidget
 
   requisition->width = 0;
   requisition->height = 0;
-  nvis_vert_children = 0;
-  nvis_horz_children = 0;
   
   width = height = maxWidth = maxHeight = 0;
   
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkpreview.c gtk+-1.2.10/gtk/gtkpreview.c
--- gtk+-1.2.10.orig/gtk/gtkpreview.c	1999-08-27 07:48:39.000000000 +0200
+++ gtk+-1.2.10/gtk/gtkpreview.c	2023-03-29 12:18:06.071932232 +0200
@@ -247,7 +247,8 @@ gtk_preview_put (GtkPreview   *preview,
     return;
 
   widget = GTK_WIDGET (preview);
-
+  (void)widget;
+  
   r1.x = 0;
   r1.y = 0;
   r1.width = preview->buffer_width;
@@ -446,7 +447,7 @@ gtk_preview_finalize (GtkObject *object)
   preview = GTK_PREVIEW (object);
   if (preview->buffer)
     g_free (preview->buffer);
-  preview->type = (GtkPreviewType) -1;
+  preview->type = GTK_PREVIEW_GRAYSCALE;
 
   (* GTK_OBJECT_CLASS (parent_class)->finalize) (object);
 }
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkprogressbar.c gtk+-1.2.10/gtk/gtkprogressbar.c
--- gtk+-1.2.10.orig/gtk/gtkprogressbar.c	2001-02-12 21:29:36.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkprogressbar.c	2023-03-29 12:18:06.071932232 +0200
@@ -417,7 +417,6 @@ gtk_progress_bar_act_mode_enter (GtkProg
 {
   GtkProgressBar *pbar;
   GtkWidget *widget;
-  gint size;
 
   pbar = GTK_PROGRESS_BAR (progress);
   widget = GTK_WIDGET (progress);
@@ -427,8 +426,6 @@ gtk_progress_bar_act_mode_enter (GtkProg
   if (pbar->orientation == GTK_PROGRESS_LEFT_TO_RIGHT ||
       pbar->orientation == GTK_PROGRESS_RIGHT_TO_LEFT)
     {
-      size = MAX (2, widget->allocation.width / pbar->activity_blocks);
-
       if (pbar->orientation == GTK_PROGRESS_LEFT_TO_RIGHT)
 	{
 	  pbar->activity_pos = widget->style->klass->xthickness;
@@ -444,8 +441,6 @@ gtk_progress_bar_act_mode_enter (GtkProg
     }
   else
     {
-      size = MAX (2, widget->allocation.height / pbar->activity_blocks);
-
       if (pbar->orientation == GTK_PROGRESS_TOP_TO_BOTTOM)
 	{
 	  pbar->activity_pos = widget->style->klass->ythickness;
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkradiobutton.c gtk+-1.2.10/gtk/gtkradiobutton.c
--- gtk+-1.2.10.orig/gtk/gtkradiobutton.c	2001-03-15 19:41:40.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkradiobutton.c	2023-03-29 12:18:06.071932232 +0200
@@ -149,6 +149,7 @@ gtk_radio_button_get_arg (GtkObject
   GtkRadioButton *radio_button;
 
   radio_button = GTK_RADIO_BUTTON (object);
+  (void)radio_button;
 
   switch (arg_id)
     {
@@ -371,7 +372,6 @@ gtk_radio_button_draw_indicator (GtkChec
 				 GdkRectangle   *area)
 {
   GtkWidget *widget;
-  GtkButton *button;
   GtkToggleButton *toggle_button;
   GtkStateType state_type;
   GtkShadowType shadow_type;
@@ -386,8 +386,8 @@ gtk_radio_button_draw_indicator (GtkChec
   if (GTK_WIDGET_VISIBLE (check_button) && GTK_WIDGET_MAPPED (check_button))
     {
       widget = GTK_WIDGET (check_button);
-      button = GTK_BUTTON (check_button);
       toggle_button = GTK_TOGGLE_BUTTON (check_button);
+      (void)toggle_button;
 
       state_type = GTK_WIDGET_STATE (widget);
       if ((state_type != GTK_STATE_NORMAL) &&
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkrc.c gtk+-1.2.10/gtk/gtkrc.c
--- gtk+-1.2.10.orig/gtk/gtkrc.c	2001-03-15 19:41:40.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkrc.c	2023-03-29 12:18:06.071932232 +0200
@@ -1057,8 +1057,8 @@ gtk_rc_parse_any (const gchar  *input_na
 		      if (msg)
 			msg = g_strconcat ("e.g. `", msg, "'", NULL);
 		    }
-		  if (scanner->token > GTK_RC_TOKEN_INVALID &&
-		      scanner->token < GTK_RC_TOKEN_LAST)
+		  if (scanner->token > (GTokenType)GTK_RC_TOKEN_INVALID &&
+		      scanner->token < (GTokenType)GTK_RC_TOKEN_LAST)
 		    {
 		      symbol_name = "???";
 		      for (i = 0; i < n_symbols; i++)
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkselection.c gtk+-1.2.10/gtk/gtkselection.c
--- gtk+-1.2.10.orig/gtk/gtkselection.c	2000-08-18 23:36:34.000000000 +0200
+++ gtk+-1.2.10/gtk/gtkselection.c	2023-03-29 12:18:06.071932232 +0200
@@ -710,7 +710,7 @@ gtk_selection_data_set (GtkSelectionData
       if (length < 0)
 	selection_data->data = NULL;
       else
-	selection_data->data = g_strdup("");
+	selection_data->data = (unsigned char *)g_strdup("");
     }
   
   selection_data->length = length;
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtksignal.c gtk+-1.2.10/gtk/gtksignal.c
--- gtk+-1.2.10.orig/gtk/gtksignal.c	2001-02-28 21:36:41.000000000 +0100
+++ gtk+-1.2.10/gtk/gtksignal.c	2023-03-29 12:18:06.071932232 +0200
@@ -1730,6 +1730,7 @@ gtk_signal_connect_by_type (GtkObject
   g_return_val_if_fail (object->klass != NULL, 0);
   
   signal = LOOKUP_SIGNAL_ID (signal_id);
+  (void)signal;
 
   /* Search through the signals for this object and make
    *  sure the one we are adding is valid. We need to perform
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtksocket.c gtk+-1.2.10/gtk/gtksocket.c
--- gtk+-1.2.10.orig/gtk/gtksocket.c	2001-02-28 21:36:57.000000000 +0100
+++ gtk+-1.2.10/gtk/gtksocket.c	2023-03-29 12:18:06.075932227 +0200
@@ -89,11 +89,9 @@ gtk_socket_get_type ()
 static void
 gtk_socket_class_init (GtkSocketClass *class)
 {
-  GtkObjectClass *object_class;
   GtkWidgetClass *widget_class;
   GtkContainerClass *container_class;
 
-  object_class = (GtkObjectClass*) class;
   widget_class = (GtkWidgetClass*) class;
   container_class = (GtkContainerClass*) class;
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkstatusbar.c gtk+-1.2.10/gtk/gtkstatusbar.c
--- gtk+-1.2.10.orig/gtk/gtkstatusbar.c	1999-02-24 08:36:24.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkstatusbar.c	2023-03-29 12:18:06.075932227 +0200
@@ -78,12 +78,8 @@ static void
 gtk_statusbar_class_init (GtkStatusbarClass *class)
 {
   GtkObjectClass *object_class;
-  GtkWidgetClass *widget_class;
-  GtkContainerClass *container_class;
 
   object_class = (GtkObjectClass *) class;
-  widget_class = (GtkWidgetClass *) class;
-  container_class = (GtkContainerClass *) class;
 
   parent_class = gtk_type_class (gtk_hbox_get_type ());
 
@@ -351,6 +347,7 @@ gtk_statusbar_finalize (GtkObject *objec
   g_return_if_fail (GTK_IS_STATUSBAR (object));
 
   statusbar = GTK_STATUSBAR (object);
+  (void)statusbar;
 
   GTK_OBJECT_CLASS (parent_class)->finalize (object);
 }
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkstyle.c gtk+-1.2.10/gtk/gtkstyle.c
--- gtk+-1.2.10.orig/gtk/gtkstyle.c	2001-02-20 17:46:58.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkstyle.c	2023-03-29 12:18:06.075932227 +0200
@@ -3106,7 +3106,7 @@ gtk_default_draw_focus (GtkStyle      *s
   if (detail && !strcmp (detail, "add-mode"))
     {
       gdk_gc_set_line_attributes (style->black_gc, 1, GDK_LINE_ON_OFF_DASH, 0, 0);
-      gdk_gc_set_dashes (style->black_gc, 0, "\4\4", 2);
+      gdk_gc_set_dashes (style->black_gc, 0, (gint8 *)"\4\4", 2);
       
       gdk_draw_rectangle (window,
                           style->black_gc, FALSE,
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtktearoffmenuitem.c gtk+-1.2.10/gtk/gtktearoffmenuitem.c
--- gtk+-1.2.10.orig/gtk/gtktearoffmenuitem.c	1999-02-24 08:36:31.000000000 +0100
+++ gtk+-1.2.10/gtk/gtktearoffmenuitem.c	2023-03-29 12:18:06.075932227 +0200
@@ -78,11 +78,9 @@ gtk_tearoff_menu_item_new (void)
 static void
 gtk_tearoff_menu_item_class_init (GtkTearoffMenuItemClass *klass)
 {
-  GtkObjectClass *object_class;
   GtkWidgetClass *widget_class;
   GtkMenuItemClass *menu_item_class;
 
-  object_class = (GtkObjectClass*) klass;
   widget_class = (GtkWidgetClass*) klass;
   menu_item_class = (GtkMenuItemClass*) klass;
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtktext.c gtk+-1.2.10/gtk/gtktext.c
--- gtk+-1.2.10.orig/gtk/gtktext.c	2001-03-15 21:15:12.000000000 +0100
+++ gtk+-1.2.10/gtk/gtktext.c	2023-03-29 12:19:13.363866255 +0200
@@ -1070,7 +1070,7 @@ gint
 gtk_text_forward_delete (GtkText *text,
 			 guint    nchars)
 {
-  guint old_lines, old_height;
+  guint old_lines = 0, old_height = 0;
   GtkEditable *editable = GTK_EDITABLE (text);
   gboolean frozen = FALSE;
   
@@ -1191,7 +1191,7 @@ gtk_text_get_chars (GtkEditable   *edita
       guchar ch;
       ch = text->text.ch[end_pos];
       text->text.ch[end_pos] = 0;
-      retval = g_strdup (text->text.ch + start_pos);
+      retval = g_strdup ((gchar *)(text->text.ch + start_pos));
       text->text.ch[end_pos] = ch;
     }
 
@@ -5042,14 +5042,14 @@ draw_line (GtkText* text,
 						 buffer.wc, len);
 	      else
 		pixel_width = gdk_text_width (gc_values.font,
-					      buffer.ch, len);
+					      (gchar *)buffer.ch, len);
 	    }
 	  else
 	    {
 	      if (text->use_wchar)
 		pixel_width = gdk_text_width_wc (font, buffer.wc, len);
 	      else
-		pixel_width = gdk_text_width (font, buffer.ch, len);
+		pixel_width = gdk_text_width (font, (gchar *)buffer.ch, len);
 	    }
 	  
 	  draw_bg_rect (text, &mark, running_offset, pixel_start_height,
@@ -5081,7 +5081,7 @@ draw_line (GtkText* text,
 			   fg_gc,
 			   running_offset,
 			   pixel_height,
-			   buffer.ch,
+			   (gchar *)buffer.ch,
 			   len);
 	  
 	  running_offset += pixel_width;
@@ -5249,6 +5249,7 @@ drawn_cursor_max (GtkText* text)
   g_assert(text->cursor_mark.property);
   
   font = MARK_CURRENT_FONT(text, &text->cursor_mark);
+  (void)font;
   
   return text->cursor_pos_y - text->cursor_char_offset;
 }
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtktogglebutton.c gtk+-1.2.10/gtk/gtktogglebutton.c
--- gtk+-1.2.10.orig/gtk/gtktogglebutton.c	2000-12-03 17:48:51.000000000 +0100
+++ gtk+-1.2.10/gtk/gtktogglebutton.c	2023-03-29 12:18:06.075932227 +0200
@@ -106,12 +106,10 @@ gtk_toggle_button_class_init (GtkToggleB
 {
   GtkObjectClass *object_class;
   GtkWidgetClass *widget_class;
-  GtkContainerClass *container_class;
   GtkButtonClass *button_class;
 
   object_class = (GtkObjectClass*) class;
   widget_class = (GtkWidgetClass*) class;
-  container_class = (GtkContainerClass*) class;
   button_class = (GtkButtonClass*) class;
 
   parent_class = gtk_type_class (GTK_TYPE_BUTTON);
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtktree.c gtk+-1.2.10/gtk/gtktree.c
--- gtk+-1.2.10.orig/gtk/gtktree.c	2001-02-26 20:20:07.000000000 +0100
+++ gtk+-1.2.10/gtk/gtktree.c	2023-03-29 12:18:06.075932227 +0200
@@ -332,7 +332,9 @@ gtk_tree_button_release (GtkWidget
   
   tree = GTK_TREE (widget);
   item = gtk_get_event_widget ((GdkEvent*) event);
-  
+  (void)tree;
+  (void)item;
+    
   return TRUE;
 }
 
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtktreeitem.c gtk+-1.2.10/gtk/gtktreeitem.c
--- gtk+-1.2.10.orig/gtk/gtktreeitem.c	2001-02-16 21:55:47.000000000 +0100
+++ gtk+-1.2.10/gtk/gtktreeitem.c	2023-03-29 12:18:06.075932227 +0200
@@ -642,7 +642,6 @@ static void
 gtk_tree_item_paint (GtkWidget    *widget,
 		     GdkRectangle *area)
 {
-  GtkBin *bin;
   GdkRectangle child_area, item_area;
   GtkTreeItem* tree_item;
 
@@ -658,7 +657,6 @@ gtk_tree_item_paint (GtkWidget    *widge
    */
   if (GTK_WIDGET_DRAWABLE (widget))
     {
-      bin = GTK_BIN (widget);
       tree_item = GTK_TREE_ITEM(widget);
 
       if (widget->state == GTK_STATE_NORMAL)
@@ -825,6 +823,7 @@ gtk_real_tree_item_select (GtkItem *item
   g_return_if_fail (GTK_IS_TREE_ITEM (item));
 
   tree_item = GTK_TREE_ITEM (item);
+  (void)tree_item;
   widget = GTK_WIDGET (item);
 
   gtk_widget_set_state (GTK_WIDGET (item), GTK_STATE_SELECTED);
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkvbbox.c gtk+-1.2.10/gtk/gtkvbbox.c
--- gtk+-1.2.10.orig/gtk/gtkvbbox.c	1999-02-24 08:36:53.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkvbbox.c	2023-03-29 12:18:06.075932227 +0200
@@ -215,6 +215,7 @@ gtk_vbutton_box_size_allocate (GtkWidget
 
   box = GTK_BUTTON_BOX (widget);
   hbox = GTK_VBUTTON_BOX (widget);
+  (void)hbox;
   spacing = box->spacing != GTK_BUTTONBOX_DEFAULT
 	  ? box->spacing : default_spacing;
   layout = box->layout_style != GTK_BUTTONBOX_DEFAULT_STYLE
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkviewport.c gtk+-1.2.10/gtk/gtkviewport.c
--- gtk+-1.2.10.orig/gtk/gtkviewport.c	2001-01-30 17:37:35.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkviewport.c	2023-03-29 12:18:06.075932227 +0200
@@ -652,6 +652,7 @@ gtk_viewport_size_request (GtkWidget
   g_return_if_fail (requisition != NULL);
 
   viewport = GTK_VIEWPORT (widget);
+  (void)viewport;
   bin = GTK_BIN (widget);
 
   requisition->width = (GTK_CONTAINER (widget)->border_width +
@@ -791,6 +792,7 @@ gtk_viewport_adjustment_changed (GtkAdju
   g_return_if_fail (GTK_IS_VIEWPORT (data));
 
   viewport = GTK_VIEWPORT (data);
+  (void)viewport;
 }
 
 static void
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkvruler.c gtk+-1.2.10/gtk/gtkvruler.c
--- gtk+-1.2.10.orig/gtk/gtkvruler.c	1999-02-24 08:37:01.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkvruler.c	2023-03-29 12:18:06.075932227 +0200
@@ -134,7 +134,7 @@ static void
 gtk_vruler_draw_ticks (GtkRuler *ruler)
 {
   GtkWidget *widget;
-  GdkGC *gc, *bg_gc;
+  GdkGC *gc;
   GdkFont *font;
   gint i, j;
   gint width, height;
@@ -161,7 +161,6 @@ gtk_vruler_draw_ticks (GtkRuler *ruler)
   widget = GTK_WIDGET (ruler);
 
   gc = widget->style->fg_gc[GTK_STATE_NORMAL];
-  bg_gc = widget->style->bg_gc[GTK_STATE_NORMAL];
   font = widget->style->font;
   xthickness = widget->style->klass->xthickness;
   ythickness = widget->style->klass->ythickness;
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkvscale.c gtk+-1.2.10/gtk/gtkvscale.c
--- gtk+-1.2.10.orig/gtk/gtkvscale.c	2001-02-12 21:36:37.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkvscale.c	2023-03-29 12:18:06.075932227 +0200
@@ -300,14 +300,12 @@ static void
 gtk_vscale_clear_background (GtkRange    *range)
 {
   GtkWidget *widget;
-  GtkScale *scale;
   gint x, y, width, height;
   
   g_return_if_fail (range != NULL);
   g_return_if_fail (GTK_IS_SCALE (range));
   
   widget = GTK_WIDGET (range);
-  scale = GTK_SCALE (range);
   
   gtk_vscale_pos_background (GTK_VSCALE (widget), &x, &y, &width, &height);
   
@@ -362,7 +360,6 @@ gtk_vscale_size_allocate (GtkWidget
                           GtkAllocation *allocation)
 {
   GtkRange *range;
-  GtkScale *scale;
   gint width, height;
   gint x, y;
   
@@ -374,7 +371,6 @@ gtk_vscale_size_allocate (GtkWidget
   if (GTK_WIDGET_REALIZED (widget))
     {
       range = GTK_RANGE (widget);
-      scale = GTK_SCALE (widget);
       
       gtk_vscale_pos_trough (GTK_VSCALE (widget), &x, &y, &width, &height);
       
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/gtkwindow.c gtk+-1.2.10/gtk/gtkwindow.c
--- gtk+-1.2.10.orig/gtk/gtkwindow.c	2001-03-10 00:39:16.000000000 +0100
+++ gtk+-1.2.10/gtk/gtkwindow.c	2023-03-29 12:18:06.079932224 +0200
@@ -876,7 +876,8 @@ gtk_window_show (GtkWidget *widget)
       GtkWindowGeometryInfo *info = gtk_window_get_geometry_info (window, TRUE);
       GtkAllocation allocation = { 0, 0 };
       GdkGeometry new_geometry;
-      guint width, height, new_flags;
+      guint width, height;
+      guint new_flags;
 
       /* determine default size to initially show the window with */
       gtk_widget_size_request (widget, NULL);
@@ -891,7 +892,7 @@ gtk_window_show (GtkWidget *widget)
       gtk_window_constrain_size (window,
 				 &new_geometry, new_flags,
 				 width, height,
-				 &width, &height);
+				 (gint *)&width, (gint *)&height);
 
       /* and allocate the window */
       allocation.width  = width;
@@ -1615,7 +1616,7 @@ gtk_window_move_resize (GtkWindow *windo
   saved_last_info = info->last;
 
   gtk_widget_size_request (widget, NULL);
-  gtk_window_compute_default_size (window, &new_width, &new_height);
+  gtk_window_compute_default_size (window, (guint *)&new_width, (guint *)&new_height);
   
   if (info->last.width < 0 ||
       info->last.width != new_width ||
@@ -2010,7 +2011,6 @@ gtk_window_compute_hints (GtkWindow   *w
 {
   GtkWidget *widget;
   GtkWidgetAuxInfo *aux_info;
-  gint ux, uy;
   gint extra_width = 0;
   gint extra_height = 0;
   GtkWindowGeometryInfo *geometry_info;
@@ -2034,14 +2034,9 @@ gtk_window_compute_hints (GtkWindow   *w
       extra_height = widget->requisition.height - geometry_info->widget->requisition.height;
     }
   
-  ux = 0;
-  uy = 0;
-  
   aux_info = gtk_object_get_data (GTK_OBJECT (widget), "gtk-aux-info");
   if (aux_info && (aux_info->x != -1) && (aux_info->y != -1))
     {
-      ux = aux_info->x;
-      uy = aux_info->y;
       *new_flags |= GDK_HINT_POS;
     }
   
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/simple.c gtk+-1.2.10/gtk/simple.c
--- gtk+-1.2.10.orig/gtk/simple.c	1998-05-04 00:41:06.000000000 +0200
+++ gtk+-1.2.10/gtk/simple.c	2023-03-29 12:18:06.079932224 +0200
@@ -31,6 +31,7 @@ main (int argc, char *argv[])
 			   "GtkWidget::parent", window,
 			   "GtkWidget::visible", TRUE,
 			   NULL);
+  (void)button;			   
   gtk_widget_show (window);
 
   gtk_main ();
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/testdnd.c gtk+-1.2.10/gtk/testdnd.c
--- gtk+-1.2.10.orig/gtk/testdnd.c	1999-04-06 20:50:38.000000000 +0200
+++ gtk+-1.2.10/gtk/testdnd.c	2023-03-29 12:18:06.079932224 +0200
@@ -395,11 +395,11 @@ source_drag_data_get  (GtkWidget
   else if (info == TARGET_URL)
     gtk_selection_data_set (selection_data,
 			    selection_data->target,
-			    8, "file:///home/otaylor/images/weave.png", 37);
+			    8, (const guchar *)"file:///home/otaylor/images/weave.png", 37);
   else
     gtk_selection_data_set (selection_data,
 			    selection_data->target,
-			    8, "I'm Data!", 9);
+			    8, (const guchar *)"I'm Data!", 9);
 }
   
 /* The following is a rather elaborate example demonstrating/testing
diff -rup -x '*.o' -x '*.lo' -x Makefile -x Makefile.in gtk+-1.2.10.orig/gtk/testgtk.c gtk+-1.2.10/gtk/testgtk.c
--- gtk+-1.2.10.orig/gtk/testgtk.c	2001-02-26 21:18:44.000000000 +0100
+++ gtk+-1.2.10/gtk/testgtk.c	2023-03-29 12:18:06.079932224 +0200
@@ -3117,7 +3117,6 @@ cursor_expose_event (GtkWidget *widget,
 		     GdkEvent  *event,
 		     gpointer   user_data)
 {
-  GtkDrawingArea *darea;
   GdkDrawable *drawable;
   GdkGC *black_gc;
   GdkGC *gray_gc;
@@ -3128,7 +3127,6 @@ cursor_expose_event (GtkWidget *widget,
   g_return_val_if_fail (widget != NULL, TRUE);
   g_return_val_if_fail (GTK_IS_DRAWING_AREA (widget), TRUE);
 
-  darea = GTK_DRAWING_AREA (widget);
   drawable = widget->window;
   white_gc = widget->style->white_gc;
   gray_gc = widget->style->bg_gc[GTK_STATE_NORMAL];
@@ -6560,13 +6558,11 @@ shape_create_icon (char     *xpm_file,
   GtkWidget *pixmap;
   GtkWidget *fixed;
   CursorOffset* icon_pos;
-  GdkGC* gc;
   GdkBitmap *gdk_pixmap_mask;
   GdkPixmap *gdk_pixmap;
   GtkStyle *style;
 
   style = gtk_widget_get_default_style ();
-  gc = style->black_gc;	
 
   /*
    * GDK_WINDOW_TOPLEVEL works also, giving you a title border
@@ -6690,7 +6686,7 @@ create_wmhints (void)
       gtk_widget_realize (window);
       
       circles = gdk_bitmap_create_from_data (window->window,
-					     circles_bits,
+					     (const char *)circles_bits,
 					     circles_width,
 					     circles_height);
       gdk_window_set_icon (window->window, NULL,
@@ -7542,7 +7538,6 @@ scroll_test_adjustment_changed (GtkAdjus
   gint source_min = (int)adj->value - scroll_test_pos;
   gint source_max = source_min + widget->allocation.height;
   gint dest_min = 0;
-  gint dest_max = widget->allocation.height;
   GdkRectangle rect;
   GdkEvent *event;
 
@@ -7573,7 +7568,6 @@ scroll_test_adjustment_changed (GtkAdjus
       rect.height = widget->allocation.height - rect.y;
 
       source_max = widget->allocation.height;
-      dest_max = rect.y;
     }
 
   if (source_min != source_max)
