--- gtk+-1.2.8/gtk/gtkstyle.c.~	Mon Mar 13 01:23:26 2000
+++ gtk+-1.2.8/gtk/gtkstyle.c	Mon Dec 18 17:00:18 2000
@@ -398,7 +398,7 @@
   if (!default_font)
     {
       default_font =
-	gdk_font_load ("-adobe-helvetica-medium-r-normal--*-120-*-*-*-*-iso8859-1");
+	gdk_font_load ("-*-helvetica-medium-r-normal--*-120-*-*-*-*-iso8859-1");
       if (!default_font)
 	default_font = gdk_font_load ("fixed");
       if (!default_font)
--- gtk+-1.2.8/gtk/testgtk.c.~	Wed Mar  1 06:00:34 2000
+++ gtk+-1.2.8/gtk/testgtk.c	Mon Dec 18 17:00:50 2000
@@ -4077,7 +4077,7 @@
       
       gdk_font_unref (style->font);
       style->font =
-	gdk_font_load ("-adobe-helvetica-bold-r-*-*-*-140-*-*-*-*-*-*");
+	gdk_font_load ("-*-helvetica-bold-r-*-*-*-140-*-*-*-*-*-*");
 
       for (i = 0; i < 10; i++)
 	{
@@ -5640,7 +5640,7 @@
 
       gtk_text_freeze (GTK_TEXT (text));
 
-      font = gdk_font_load ("-adobe-courier-medium-r-normal--*-120-*-*-*-*-*-*");
+      font = gdk_font_load ("-*-courier-medium-r-normal--*-120-*-*-*-*-*-*");
 
       for (i=0; i<ntext_colors; i++)
 	{
