--- a/configure.orig	2023-08-23 16:29:35.730921765 +0200
+++ b/configure	2023-08-23 16:29:35.770921737 +0200
@@ -8884,6 +8884,7 @@
 fi;
 echo "$as_me:8885: result: $with_overwrite" >&5
 echo "${ECHO_T}$with_overwrite" >&6
+WITH_OVERWRITE="$with_overwrite"
 
 echo "$as_me:8888: checking if external terminfo-database is used" >&5
 echo $ECHO_N "checking if external terminfo-database is used... $ECHO_C" >&6
--- a/configure.in.orig	2023-07-22 19:20:35.000000000 +0200
+++ b/configure.in	2023-08-23 16:29:35.770921737 +0200
@@ -587,6 +587,7 @@
 	[with_overwrite=$enableval],
 	[if test "$prefix" = "/usr" ; then with_overwrite=yes; else with_overwrite=no; fi])
 AC_MSG_RESULT($with_overwrite)
+WITH_OVERWRITE=$with_overwrite
 
 AC_MSG_CHECKING(if external terminfo-database is used)
 AC_ARG_ENABLE(database,
