diff -rupN ncurses-6.0.orig/ncurses/tinfo/lib_data.c ncurses-6.0/ncurses/tinfo/lib_data.c
--- ncurses-6.0.orig/ncurses/tinfo/lib_data.c	2017-08-04 10:59:48.000000000 +0200
+++ ncurses-6.0/ncurses/tinfo/lib_data.c	2017-10-10 18:17:07.114837726 +0200
@@ -270,7 +270,7 @@ NCURSES_EXPORT_VAR(NCURSES_PRESCREEN) _n
     NULL,			/* _tputs_trace */
 #endif
 #endif
-    FALSE,			/* use_tioctl */
+    TRUE,			/* use_tioctl */
 };
 /* *INDENT-ON* */
 
