--- ncurses-6.0/configure.in.orig	2017-10-23 01:35:26.157940765 +0200
+++ ncurses-6.0/configure.in	2017-10-23 01:35:36.725940535 +0200
@@ -112,8 +112,6 @@
 		AC_MSG_CHECKING(if $CXX works)
 
 		save_CPPFLAGS="$CPPFLAGS"
-		eval cf_includedir=${includedir}
-		CPPFLAGS="$CPPFLAGS -I${cf_includedir}"
 		AC_TRY_COMPILE([
 #include <stdlib.h>
 #include <stdio.h>
--- ncurses-6.0/configure.orig	2017-10-23 01:37:09.329938520 +0200
+++ ncurses-6.0/configure	2017-10-23 01:37:17.617938339 +0200
@@ -3515,8 +3515,6 @@
 echo $ECHO_N "checking if $CXX works... $ECHO_C" >&6
 
 		save_CPPFLAGS="$CPPFLAGS"
-		eval cf_includedir=${includedir}
-		CPPFLAGS="$CPPFLAGS -I${cf_includedir}"
 
 cat >"conftest.$ac_ext" <<_ACEOF
 #line 3522 "configure"
