diff -rupN ncurses-6.0.orig/ncurses/tinfo/lib_data.c ncurses-6.0/ncurses/tinfo/lib_data.c
--- ncurses-6.0.orig/ncurses/tinfo/lib_data.c	2017-08-04 10:59:48.000000000 +0200
+++ ncurses-6.0/ncurses/tinfo/lib_data.c	2017-10-10 18:17:07.114837726 +0200
@@ -258,7 +258,7 @@ NCURSES_EXPORT_VAR(NCURSES_PRESCREEN) _n
 #endif
     },
     NULL,			/* saved_tty */
-    FALSE,			/* use_tioctl */
+    TRUE,			/* use_tioctl */
     0,				/* _outch */
 #ifndef USE_SP_RIPOFF
     RIPOFF_0s,			/* ripoff */
