--- a/misc/terminfo.src.orig	2023-08-25 09:25:26.395549749 +0200
+++ b/misc/terminfo.src	2023-08-25 09:25:57.882194771 +0200
@@ -857,7 +857,7 @@
 	kf15=\Et, kf16=\Eu, kf17=\Ev, kf18=\Ew, kf19=\Ex, kf2=\EOQ,
 	kf20=\Ey, kf3=\EOR, kf4=\EOS, kf5=\EOT, kf6=\EOU, kf7=\EOV,
 	kf8=\EOW, kf9=\EOX, khlp=\EH, khome=\E\EE, kich1=\EI,
-	knp=\Eb, kpp=\E\Ea, kund=\EK, ll=\E[24H, nel=\EE,
+	knp=\Eb, kpp=\E\Ea, kund=\EK, nel=\EE,
 	oc=\E[30;47m, op=\E[30;47m, rc=\E8, rev=\E[7m, ri=\EM,
 	rmacs=^O, rmcup=\E[?7h, rmir=\Ei, rmkx=\E[?1l\E>,
 	rmso=\E[m, rmul=\E[m,
