--- lua-5.3.4/src/luaconf.h.orig	2018-03-15 19:55:01.885056438 +0100
+++ lua-5.3.4/src/luaconf.h	2018-03-15 20:00:04.540456157 +0100
@@ -47,10 +47,15 @@
 /*
 ** By default, Lua on Windows use (some) specific Windows features
 */
-#if !defined(LUA_USE_C89) && defined(_WIN32) && !defined(_WIN32_WCE)
+#if defined(_WIN32) && !defined(_WIN32_WCE)
 #define LUA_USE_WINDOWS  /* enable goodies for regular Windows */
 #endif
-
+#if defined(__linux__)
+#define LUA_USE_LINUX
+#endif
+#if defined(__MINT__)
+#define LUA_USE_MINT
+#endif
 
 #if defined(LUA_USE_WINDOWS)
 #define LUA_DL_DLL	/* enable support for DLL */
@@ -62,6 +67,7 @@
 #define LUA_USE_POSIX
 #define LUA_USE_DLOPEN		/* needs an extra library: -ldl */
 #define LUA_USE_READLINE	/* needs some extra libraries */
+#define LUA_COMPAT_MODULE	/* needed for lots of packages, like rpm */
 #endif
 
 
@@ -71,6 +77,10 @@
 #define LUA_USE_READLINE	/* needs an extra library: -lreadline */
 #endif
 
+#if defined(LUA_USE_MINT)
+#define LUA_USE_C89 1			/* needed because lack of support of some functions */
+#define LUA_COMPAT_MODULE 1	/* needed for lots of packages, like rpm */
+#endif
 
 /*
 @@ LUA_C89_NUMBERS ensures that Lua uses the largest types available for
