--- a/src/libFLAC++/flac++.pc.in.orig	2019-03-19 17:49:13.147009549 +0100
+++ a/src/libFLAC++/flac++.pc.in	2019-03-19 17:49:28.135009223 +0100
@@ -7,5 +7,5 @@
 Description: Free Lossless Audio Codec Library (C++ API)
 Version: @VERSION@
 Requires: flac
-Libs: -L${libdir} -lFLAC++
+Libs: -L${libdir} -lFLAC++ -logg -lm
 Cflags: -I${includedir}
--- a/src/libFLAC/flac.pc.in.orig	2019-03-19 17:51:25.563006668 +0100
+++ b/src/libFLAC/flac.pc.in	2019-03-19 17:51:32.767006511 +0100
@@ -7,6 +7,6 @@
 Description: Free Lossless Audio Codec Library
 Version: @VERSION@
 Requires.private: @OGG_PACKAGE@
-Libs: -L${libdir} -lFLAC
+Libs: -L${libdir} -lFLAC -logg -lm
 Libs.private: -lm
 Cflags: -I${includedir}
