--- xpdf-0.91/T1-1.0.1/xglyph/Makefile.in.orig	Sun Jan 23 08:04:36 2000
+++ xpdf-0.91/T1-1.0.1/xglyph/Makefile.in	Fri Dec 29 22:46:04 2000
@@ -74,7 +74,7 @@
 xglyph: $(OBJS) $(T1LIBX) $(T1LIB)
 	$(LIBTOOL) --mode=link \
 		$(CC) -o xglyph $(LDFLAGS) $(OBJS) $(T1LIBX) $(T1LIB) \
-			        $(X_LIBS) $(XAWLIB) $(XLIB) $(LDLIBS)
+			        $(X_LIBS) $(XAWLIB) $(XPM_LIB) $(XLIB) $(LDLIBS)
 
 .SUFFIXES: .lo
 .c.lo:
