diff -rup libxmp-4.6.0.orig/src/depackers/depacker.c libxmp-4.6.0.new/src/depackers/depacker.c
--- libxmp-4.6.0.orig/src/depackers/depacker.c	2023-06-15 22:11:10.000000000 +0200
+++ libxmp-4.6.0.new/src/depackers/depacker.c	2023-12-22 03:20:39.756012310 +0100
@@ -201,7 +201,7 @@ static int execute_command(const char *
 
 static int decrunch_command(HIO_HANDLE *h, const char * const cmd[], char **temp)
 {
-#if defined __ANDROID__ || defined __native_client__
+#if defined __ANDROID__ || defined __native_client__ || defined(__MINT__)
 	/* Don't use external helpers in android */
 	return 0;
 #else
