--- a/config.sub	2016-10-12 12:30:03.000000000 +0200
+++ b/config.sub	2020-08-01 09:18:35.007276436 +0200
@@ -235,6 +235,10 @@
 	-psos*)
 		os=-psos
 		;;
+	-mintelf | -mintelf[0-9]*)
+		basic_machine=m68k-atari
+		os=-mintelf
+		;;
 	-mint | -mint[0-9]*)
 		basic_machine=m68k-atari
 		os=-mint
@@ -832,6 +836,10 @@
 	miniframe)
 		basic_machine=m68000-convergent
 		;;
+	*mintelf | -mintelf[0-9]*)
+		basic_machine=m68k-atari
+		os=-mintelf
+		;;
 	*mint | -mint[0-9]* | *MiNT | *MiNT[0-9]*)
 		basic_machine=m68k-atari
 		os=-mint
@@ -1525,6 +1533,9 @@
 	-xenix)
 		os=-xenix
 		;;
+	-*mintelf | -mintelf[0-9]*)
+		os=-mintelf
+		;;
 	-*mint | -mint[0-9]* | -*MiNT | -MiNT[0-9]*)
 		os=-mint
 		;;
@@ -1801,6 +1812,9 @@
 			-mpw* | -macos*)
 				vendor=apple
 				;;
+			-*mintelf | -mintelf[0-9]*)
+				vendor=atari
+				;;
 			-*mint | -mint[0-9]* | -*MiNT | -MiNT[0-9]*)
 				vendor=atari
 				;;
