--- chkconfig-1.0.6/leveldb.c.nommap	Thu Apr  8 20:19:13 1999
+++ chkconfig-1.0.6/leveldb.c	Sun Sep 19 20:03:29 1999
@@ -5,7 +5,9 @@
 #include <glob.h>
 #include <libintl.h> 
 #include <locale.h> 
+#ifndef __MINT__
 #include <sys/mman.h>
+#endif
 #include <sys/stat.h>
 #include <stdlib.h>
 #include <stdio.h>
@@ -58,11 +60,23 @@
     if ((fd = open(filename, O_RDONLY)) < 0) return -1;
     fstat(fd, &sb);
 
+#ifndef __MINT__
     bufstart = mmap(NULL, sb.st_size, PROT_READ, MAP_SHARED, fd, 0);
     if (bufstart == ((caddr_t) -1)) {
 	close(fd);	
 	return -1;
     }
+#else
+    bufstart = malloc (sb.st_size);
+    if (bufstart == NULL) {
+    	close (fd);
+    	return -1;
+    }
+    if (read (fd, bufstart, sb.st_size) != sb.st_size) {
+    	close (fd);
+    	return -1;
+    }
+#endif
 
     bufstop = bufstart + sb.st_size;
     close(fd);
@@ -90,7 +104,9 @@
 	    while (isspace(*start) && start < end) start++;
 	    if (start == end || !strncmp(start, "true", 4)) {
 		if (serv.desc) free(serv.desc);
+#ifndef __MINT__
 		munmap(bufstart, sb.st_size);
+#endif
 		return 1;
 	    }
 	}
@@ -100,7 +116,9 @@
 	    while (isspace(*start) && start < end) start++;
 	    if (start == end) {
 		if (serv.desc) free(serv.desc);
+#ifndef __MINT__
 		munmap(bufstart, sb.st_size);
+#endif
 		return 1;
 	    }
 
@@ -108,7 +126,9 @@
 			&serv.sPriority, &serv.kPriority, &overflow) != 4) ||
 		 overflow != '\n') {
 		if (serv.desc) free(serv.desc);
+#ifndef __MINT__
 		munmap(bufstart, sb.st_size);
+#endif
 		return 1;
 	    }
 
@@ -118,7 +138,9 @@
 		serv.levels = parseLevels(levelbuf, 0);
 	    if (serv.levels == -1) {
 		if (serv.desc) free(serv.desc);
+#ifndef __MINT__
 		munmap(bufstart, sb.st_size);
+#endif
 		return 1;
 	    }
 	} else if (!strncmp(start, "description", 11)) {
@@ -129,7 +151,9 @@
             if (!english) {
 		if (*start != '(') {
 		    if (serv.desc) free(serv.desc);
+#ifndef __MINT__
 		    munmap(bufstart, sb.st_size);
+#endif
 		    return 1;
 		}
 
@@ -138,7 +162,9 @@
 
 		if (final_parenthesis == NULL || final_parenthesis - start > 5) {
 		    if (serv.desc) free(serv.desc);
+#ifndef __MINT__
 		    munmap(bufstart, sb.st_size);
+#endif
 		    return 1;
 		}
 
@@ -148,7 +174,9 @@
 
 	    while (isspace(*start) && start < end) start++;
 	    if (start == end) {
+#ifndef __MINT__
 		munmap(bufstart, sb.st_size);
+#endif
 		return 1;
 	    }
           {
@@ -164,7 +192,9 @@
 		
 		while (isspace(*start) && start < bufstop) start++;
 		if (start == bufstop || *start != '#') {
+#ifndef __MINT__
 		    munmap(bufstart, sb.st_size);
+#endif
 		    return 1;
 		}
 
@@ -172,7 +202,9 @@
 
 		while (isspace(*start) && start < bufstop) start++;
 		if (start == bufstop) {
+#ifndef __MINT__
 		    munmap(bufstart, sb.st_size);
+#endif
 		    return 1;
 		}
 
@@ -213,7 +245,9 @@
 	}
     }
 
+#ifndef __MINT__
     munmap(bufstart, sb.st_size);
+#endif
 
     if (!serv.desc) {
       if (english_desc)
