--- mpg123-1.25.10/libout123.pc.in.orig	2019-03-24 13:55:59.845545651 +0100
+++ mpg123-1.25.10/libout123.pc.in	2019-03-24 13:56:14.905545323 +0100
@@ -7,5 +7,5 @@
 Description: A streaming audio output API derived from mpg123
 Requires: 
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lout123 
+Libs: -L${libdir} -lout123 -lm
 Cflags: -I${includedir} 
--- mpg123-1.25.10/libmpg123.pc.in.orig	2019-03-24 13:55:27.173546362 +0100
+++ mpg123-1.25.10/libmpg123.pc.in	2019-03-24 13:55:47.129545928 +0100
@@ -7,5 +7,5 @@
 Description: An optimised MPEG Audio decoder
 Requires: 
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lmpg123 
+Libs: -L${libdir} -lmpg123 -lm
 Cflags: -I${includedir} 
