diff -rup flatzebra-0.1.7.orig/configure.ac flatzebra-0.1.7/configure.ac
--- flatzebra-0.1.7.orig/configure.ac	2019-11-12 03:23:34.000000000 +0100
+++ flatzebra-0.1.7/configure.ac	2024-05-14 08:33:44.564740819 +0200
@@ -43,28 +43,8 @@ AM_PROG_LIBTOOL
 # Use C mode for SDL tests:
 AC_LANG_C
 
-AM_PATH_SDL(1.2.4)
-
-# Remove any -rpath option from SDL_LIBS, so as not to hardcode
-# the library path into the executable file:
-#
-
-SDL_LIBS=`echo $SDL_LIBS | sed ['s:-Wl,-rpath,[^ ]*: :g]'`
-
-# AM_PATH_SDL does not stop if SDL is not found; therefore:
-if test "_$SDL_LIBS" = _; then
-	AC_MSG_ERROR([SDL not found])
-fi
-
 # We also need the SDL_image and SDL_mixer libraries:
-AM_TRY_LINK_SDL_IMAGE
-TRY_LINK_SDL_MIXER
-
-# This will insert SDL_image and SDL_mixer in the dependency_libs field
-# of the installed .la file. The flatzebra shared library will also
-# specify SDL_image and SDL_mixer as being needed.
-#
-SDL_LIBS="$SDL_LIBS -lSDL_image -lSDL_mixer"
+SDL_LIBS=`pkg-config --libs sdl SDL_mixer SDL_image`
 
 # Back to C++ mode after SDL tests.
 AC_LANG_CPLUSPLUS
diff -rup flatzebra-0.1.7.orig/src/flatzebra/GameEngine.h flatzebra-0.1.7/src/flatzebra/GameEngine.h
--- flatzebra-0.1.7.orig/src/flatzebra/GameEngine.h	2017-08-31 00:53:13.000000000 +0200
+++ flatzebra-0.1.7/src/flatzebra/GameEngine.h	2024-05-14 08:36:23.087414509 +0200
@@ -36,8 +36,8 @@
     @sarrazip 20010324
 */
 
-#include <SDL_types.h>
-#include <SDL_keysym.h>
+#include <SDL/SDL_types.h>
+#include <SDL/SDL_keysym.h>
 
 #include <string>
 
diff -rup flatzebra-0.1.7.orig/src/flatzebra/Joystick.h flatzebra-0.1.7/src/flatzebra/Joystick.h
--- flatzebra-0.1.7.orig/src/flatzebra/Joystick.h	2012-07-14 05:44:21.000000000 +0200
+++ flatzebra-0.1.7/src/flatzebra/Joystick.h	2024-05-14 08:35:29.520970740 +0200
@@ -23,7 +23,7 @@
 #ifndef _H_Joystick
 #define _H_Joystick
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #include <vector>
 
diff -rup flatzebra-0.1.7.orig/src/flatzebra/KeyState.h flatzebra-0.1.7/src/flatzebra/KeyState.h
--- flatzebra-0.1.7.orig/src/flatzebra/KeyState.h	2010-06-05 22:50:42.000000000 +0200
+++ flatzebra-0.1.7/src/flatzebra/KeyState.h	2024-05-14 08:36:23.090747828 +0200
@@ -23,7 +23,7 @@
 #ifndef _H_KeyState
 #define _H_KeyState
 
-#include <SDL_keysym.h>
+#include <SDL/SDL_keysym.h>
 
 
 namespace flatzebra {
diff -rup flatzebra-0.1.7.orig/src/flatzebra/PixmapArray.h flatzebra-0.1.7/src/flatzebra/PixmapArray.h
--- flatzebra-0.1.7.orig/src/flatzebra/PixmapArray.h	2014-02-04 05:39:31.000000000 +0100
+++ flatzebra-0.1.7/src/flatzebra/PixmapArray.h	2024-05-14 08:35:29.520970740 +0200
@@ -25,8 +25,8 @@
 
 #include <flatzebra/Couple.h>
 
-#include <SDL.h>
-#include <SDL_image.h>
+#include <SDL/SDL.h>
+#include <SDL/SDL_image.h>
 
 #include <assert.h>
 #include <vector>
diff -rup flatzebra-0.1.7.orig/src/flatzebra/SoundMixer.h flatzebra-0.1.7/src/flatzebra/SoundMixer.h
--- flatzebra-0.1.7.orig/src/flatzebra/SoundMixer.h	2017-08-31 00:53:13.000000000 +0200
+++ flatzebra-0.1.7/src/flatzebra/SoundMixer.h	2024-05-14 08:36:23.090747828 +0200
@@ -23,7 +23,7 @@
 #ifndef _H_SoundMixer
 #define _H_SoundMixer
 
-#include <SDL_mixer.h>
+#include <SDL/SDL_mixer.h>
 
 #include <string>
 
